---
title: "Final EGRET v1 Banovich application - Data Analysis"
author: "Deborah Weighill"
date: "July 28, 2020"
output:
  html_notebook:
    toc: true
    theme: cosmo
    df_print: paged
---

# Overview
Here we do some analysis on the banovich results for the "application". We start from the "master difference table" that we created.



# Load libraries
```{r}
library(dplyr)
library(preprocessCore)
library(ggplot2)
library(tidyr)
library(ggpubr)
library(reshape2)
library(jcolors)
library(ALPACA)
library(gridExtra)
library(grid)
library(gridExtra)
pallet <- c(    "#FFBF00",  "#870E75"  ,  "#0BB19F" )
```

# load differences table differences table

```{r}
load("../outputs/diff_table_with_gwas_banovich_allModels_smart1_07042020.RData")
```

# Load gwas
```{r}

# load gene annotation to assign gene id to gene name
nameGeneMap <- read.table("../../final_egret_v1/annotation/geneID_name_map.txt", header = FALSE)
colnames(nameGeneMap) <- c("gene","name")

RA <- read.table("../gwas/RA_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(RA) <- c("RAtrait","name")
RA$gene <- nameGeneMap$gene[match(RA$name, nameGeneMap$name)]

CD <- read.table("../gwas/CD_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(CD) <- c("CDtrait","name")
CD$gene <- nameGeneMap$gene[match(CD$name, nameGeneMap$name)]

CAD <- read.table("../gwas/CAD_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(CAD) <- c("CADtrait","name")
CAD$gene <- nameGeneMap$gene[match(CAD$name, nameGeneMap$name)]

BMI <- read.table("../gwas/BMI_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(BMI) <- c("BMItrait","name")
BMI$gene <- nameGeneMap$gene[match(BMI$name, nameGeneMap$name)]

MS <- read.table("../gwas/MS_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(MS) <- c("MStrait","name")
MS$gene <- nameGeneMap$gene[match(MS$name, nameGeneMap$name)]

MI <- read.table("../gwas/MI_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(MI) <- c("MItrait","name")
MI$gene <- nameGeneMap$gene[match(MI$name, nameGeneMap$name)]
```

# Hard threshold edge differences, aggregate by tf and gene
```{r}
diff_table_hard_thresh <- diff_table[,c(c(5:123),c(125:243),c(245:363))]
diff_table_hard_thresh[diff_table_hard_thresh < 0.35] <- 0
diff_table_hard_thresh <- cbind(diff_table[,c(2,3)],diff_table_hard_thresh)
tf_degrees_hard_thresh <- aggregate(diff_table_hard_thresh[,c(3:359)], by = list(diff_table_hard_thresh$tf), FUN = sum) 
gene_degrees_hard_thresh <- aggregate(diff_table_hard_thresh[,c(3:359)], by = list(diff_table_hard_thresh$gene), FUN = sum) 
colnames(tf_degrees_hard_thresh)[1] <- "tf"
colnames(gene_degrees_hard_thresh)[1] <- "gene"
head(tf_degrees_hard_thresh)
head(gene_degrees_hard_thresh)
tf_degrees_hard_thresh$GWAS_CD_tf <- ifelse(tf_degrees_hard_thresh$tf %in% CD$name, 1, 0)
gene_degrees_hard_thresh$GWAS_CD_gene <- ifelse(gene_degrees_hard_thresh$gene %in% CD$gene, 1, 0)
tf_degrees_hard_thresh$GWAS_CAD_tf <- ifelse(tf_degrees_hard_thresh$tf %in% CAD$name, 1, 0)
gene_degrees_hard_thresh$GWAS_CAD_gene <- ifelse(gene_degrees_hard_thresh$gene %in% CAD$gene, 1, 0)
tf_degrees_hard_thresh$GWAS_MI_tf <- ifelse(tf_degrees_hard_thresh$tf %in% MI$name, 1, 0)
gene_degrees_hard_thresh$GWAS_MI_gene <- ifelse(gene_degrees_hard_thresh$gene %in% MI$gene, 1, 0)
tf_degrees_hard_thresh$GWAS_MS_tf <- ifelse(tf_degrees_hard_thresh$tf %in% MS$name, 1, 0)
gene_degrees_hard_thresh$GWAS_MS_gene <- ifelse(gene_degrees_hard_thresh$gene %in% MS$gene, 1, 0)
tf_degrees_hard_thresh$GWAS_RA_tf <- ifelse(tf_degrees_hard_thresh$tf %in% RA$name, 1, 0)
gene_degrees_hard_thresh$GWAS_RA_gene <- ifelse(gene_degrees_hard_thresh$gene %in% RA$gene, 1, 0)
tf_degrees_hard_thresh$GWAS_BMI_tf <- ifelse(tf_degrees_hard_thresh$tf %in% BMI$name, 1, 0)
gene_degrees_hard_thresh$GWAS_BMI_gene <- ifelse(gene_degrees_hard_thresh$gene %in% BMI$gene, 1, 0)
save(tf_degrees_hard_thresh,file = "tf_degrees_hard_thresh_035.RData")
save(gene_degrees_hard_thresh,file = "gene_degrees_hard_thresh_035.RData")
load("tf_degrees_hard_thresh_035.RData")
load("gene_degrees_hard_thresh_035.RData")
```


plot distributions

```{r}
tf_degrees_hard_thresh_scaled <- scale(tf_degrees_hard_thresh[,c(2:358)], center = FALSE, scale = colSums(tf_degrees_hard_thresh[,c(2:358)]))
tf_degrees_hard_thresh_scaled <- cbind(tf_degrees_hard_thresh[,c(1)], tf_degrees_hard_thresh_scaled, tf_degrees_hard_thresh[,c(359:364)])
colnames(tf_degrees_hard_thresh_scaled)[1] <-"tf"

tf_degrees_hard_thresh_scaled2 <- scale(tf_degrees_hard_thresh_scaled[,c(2:358)], center = TRUE, scale = TRUE)
tf_degrees_hard_thresh_scaled2 <- cbind(tf_degrees_hard_thresh_scaled[,c(1)], tf_degrees_hard_thresh_scaled2, tf_degrees_hard_thresh_scaled[,c(359:364)])
colnames(tf_degrees_hard_thresh_scaled2)[1] <-"tf"

tf_degrees_hard_thresh_scaled3 <- scale(tf_degrees_hard_thresh_scaled2[,c(2:358)], center = FALSE, scale = colSums(tf_degrees_hard_thresh_scaled2[,c(2:358)]))
tf_degrees_hard_thresh_scaled3 <- cbind(tf_degrees_hard_thresh_scaled2[,c(1)], tf_degrees_hard_thresh_scaled3, tf_degrees_hard_thresh_scaled2[,c(359:364)])
colnames(tf_degrees_hard_thresh_scaled3)[1] <-"tf"

gene_degrees_hard_thresh_scaled <- scale(gene_degrees_hard_thresh[,c(2:358)], center = FALSE, scale = colSums(gene_degrees_hard_thresh[,c(2:358)]))
gene_degrees_hard_thresh_scaled <- cbind(gene_degrees_hard_thresh[,c(1)], gene_degrees_hard_thresh_scaled, gene_degrees_hard_thresh[,c(359:364)])
colnames(gene_degrees_hard_thresh_scaled)[1] <-"gene"

#tf_degrees_hard_thresh_qq <- normalize.quantiles(as.matrix(tf_degrees_hard_thresh[,c(2:358)]))
#tf_degrees_hard_thresh_qq <- cbind(tf_degrees[,c(1)], tf_degrees_hard_thresh_qq, tf_degrees[,c(359:360)])
#colnames(tf_degrees_hard_thresh_qq)[1] <-"tf"

tf_degrees <- tf_degrees_hard_thresh
tf_degree_lcl <- melt(tf_degrees[,c(1,c(2:120))], id.vars = c("tf"))
tf_degree_lcl$tissue <- "LCL"
tf_degree_iPSC <- melt(tf_degrees[,c(1,c(121:239))], id.vars = c("tf"))
tf_degree_iPSC$tissue <- "iPSC"
tf_degree_iPSC_CM <- melt(tf_degrees[,c(1,c(240:358))], id.vars = c("tf"))
tf_degree_iPSC_CM$tissue <- "iPSC-CM"
tf_degrees_melted <- rbind(tf_degree_lcl,tf_degree_iPSC,tf_degree_iPSC_CM)
colnames(tf_degrees_melted) <- c("tf","individual","diff_out_degree","Tissue")

ggplot(tf_degrees_melted, aes(x=Tissue, y=diff_out_degree, fill=Tissue, col = Tissue)) + geom_violin() + theme_bw() + scale_fill_manual(values=pallet) + scale_color_manual(values=pallet) + labs(title="Distribution of TF Disruption Scores", x ="Cell type", y = "TF Disruption Score") + theme(legend.position="none")

ggplot(tf_degrees_melted, aes(x=diff_out_degree,  color=Tissue)) + geom_density() + theme_bw() + scale_color_manual(values=pallet) +labs(title=NULL, x ="Scaled TF disruption score", y = "Frequency") +  xlim(-1, 1)

ggplot(tf_degrees_melted, aes(x=diff_out_degree,  color=Tissue, fill = Tissue)) + geom_density(alpha = 0.1) + theme_bw() + scale_color_manual(values=pallet) + scale_fill_manual(values=pallet) +labs(title=NULL, x ="Scaled TF disruption score", y = "Frequency") +  xlim(0.1, 10)

tf_degrees <- tf_degrees_hard_thresh_scaled2
tf_degree_lcl <- melt(tf_degrees[,c(1,c(2:120))], id.vars = c("tf"))
tf_degree_lcl$tissue <- "LCL"
tf_degree_iPSC <- melt(tf_degrees[,c(1,c(121:239))], id.vars = c("tf"))
tf_degree_iPSC$tissue <- "iPSC"
tf_degree_iPSC_CM <- melt(tf_degrees[,c(1,c(240:358))], id.vars = c("tf"))
tf_degree_iPSC_CM$tissue <- "iPSC-CM"
tf_degrees_melted <- rbind(tf_degree_lcl,tf_degree_iPSC,tf_degree_iPSC_CM)
colnames(tf_degrees_melted) <- c("tf","individual","diff_out_degree","Tissue")

ggplot(tf_degrees_melted, aes(x=Tissue, y=diff_out_degree, fill=Tissue, col = Tissue)) + geom_violin() + theme_bw() + scale_fill_manual(values=pallet) + scale_color_manual(values=pallet) + labs(title="Distribution of TF Disruption Scores", x ="Cell type", y = "TF Disruption Score") + theme(legend.position="none")

ggplot(tf_degrees_melted, aes(x=diff_out_degree,  color=Tissue)) + geom_density() + theme_bw() + scale_color_manual(values=pallet) +labs(title=NULL, x ="Scaled TF disruption score", y = "Frequency") 

ggplot(tf_degrees_melted, aes(x=diff_out_degree,  color=Tissue, fill = Tissue)) + geom_density(alpha = 0.1) + theme_bw() + scale_color_manual(values=pallet) + scale_fill_manual(values=pallet) +labs(title=NULL, x ="Scaled TF disruption score", y = "Frequency") +  xlim(0.001, 10)



```



# Enrichment tests

## Hard threshold TF disruption scores

CAD
```{r}
tf_data <- tf_degrees_hard_thresh_scaled2
d = 8
individuals <- c(2:358)
CAD_enrich <- sapply(individuals, FUN = function(i){
  high_diff_gwas <- length(which((tf_data[,i] >d) & (tf_data$GWAS_CAD_tf == 1)))
  low_diff_gwas <- length(which((tf_data[,i] <=d) & (tf_data$GWAS_CAD_tf == 1)))
  high_diff_notGwas <- length(which((tf_data[,i] >d) & (tf_data$GWAS_CAD_tf == 0)))
  low_diff_notGwas <- length(which((tf_data[,i] <=d) & (tf_data$GWAS_CAD_tf == 0)))
  contingency_table <- matrix(c(high_diff_gwas,low_diff_gwas,high_diff_notGwas,low_diff_notGwas), nrow = 2,dimnames = list(Diff = c("High", "Low"),GWAS = c("yes", "no")))
  test <- fisher.test(contingency_table, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(CAD_enrich) <- colnames(tf_data)[individuals]

cad_enriched_indiv <- names(which(CAD_enrich < 0.05))
cad_enriched_indiv
cad_df <- as.data.frame(CAD_enrich)
colnames(cad_df) <- "pvalue"
cad_df$Tissue <- NA
cad_df$Tissue[1:119] <- "LCL"
cad_df$Tissue[120:238] <- "iPSC"
cad_df$Tissue[239:357] <- "iPSC-CM"
cad_df$Individual <- NA
cad_df$Individual <- c(c(1:119),c(1:119),c(1:119))

#pdf("banovich_CAD_enrichment.pdf",width = 5, height = 3.5)
ggplot(cad_df) + geom_point(shape = 19, aes(x = reorder(Individual, log10(pvalue)), y = -log10(pvalue), col = Tissue),) + theme_classic()+ scale_color_manual(values=pallet) + geom_abline(slope = 0, intercept = -log10(0.05), linetype = "dashed", col = "black") + xlab("Individual") + theme( axis.text.x=element_blank(), axis.ticks.x=element_blank())  + labs(y = "-log10(p-value)", title = "Enrichment of disrupted TFs in CAD genes",color = "Cell type") + annotate("text", x = 110, y = 1.4, label = "p = 0.05")
#dev.off()

```


CAD
```{r}
cad_tfs <- tf_degrees_hard_thresh[which(tf_degrees_hard_thresh$GWAS_CAD_tf == 1),1]
tf_data <- tf_degrees_hard_thresh_scaled2
individuals <- c(2:358)
CAD_enrich <- sapply(individuals, FUN = function(i){
  mylist <- as.vector(tf_degrees_hard_thresh[order(-as.numeric(as.vector(tf_degrees_hard_thresh[,i]))),1])
  test <- TestNodeRank(mylist, cad_tfs)
  return(test[3])
})
names(CAD_enrich) <- colnames(tf_data)[individuals]

cad_enriched_indiv <- names(which(CAD_enrich < 0.05))
cad_enriched_indiv
cad_df <- as.data.frame(CAD_enrich)
colnames(cad_df) <- "pvalue"
cad_df$Tissue <- NA
cad_df$Tissue[1:119] <- "LCL"
cad_df$Tissue[120:238] <- "iPSC"
cad_df$Tissue[239:357] <- "iPSC-CM"
cad_df$Individual <- NA
cad_df$Individual <- c(c(1:119),c(1:119),c(1:119))

#pdf("banovich_CAD_enrichment.pdf",width = 5, height = 3.5)
ggplot(cad_df) + geom_point(shape = 19, aes(x = reorder(Individual, log10(pvalue)), y = -log10(pvalue), col = Tissue),) + theme_classic()+ scale_color_manual(values=pallet) + geom_abline(slope = 0, intercept = -log10(0.05), linetype = "dashed", col = "black") + xlab("Individual") + theme( axis.text.x=element_blank(), axis.ticks.x=element_blank())  + labs(y = "-log10(p-value)", title = "Enrichment of disrupted TFs in CAD genes",color = "Cell type") + annotate("text", x = 110, y = 1.4, label = "p = 0.05")
#dev.off()

```


CD
```{r}
tf_data <- tf_degrees_hard_thresh
d = 8
individuals <- c(2:358)
CD_enrich <- sapply(individuals, FUN = function(i){
  high_diff_gwas <- length(which((tf_data[,i] >d) & (tf_data$GWAS_CD_tf == 1)))
  low_diff_gwas <- length(which((tf_data[,i] <=d) & (tf_data$GWAS_CD_tf == 1)))
  high_diff_notGwas <- length(which((tf_data[,i] >d) & (tf_data$GWAS_CD_tf == 0)))
  low_diff_notGwas <- length(which((tf_data[,i] <=d) & (tf_data$GWAS_CD_tf == 0)))
  contingency_table <- matrix(c(high_diff_gwas,low_diff_gwas,high_diff_notGwas,low_diff_notGwas), nrow = 2,dimnames = list(Diff = c("High", "Low"),GWAS = c("yes", "no")))
  test <- fisher.test(contingency_table, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(CD_enrich) <- colnames(tf_data)[individuals]
cd_enriched_indiv <- names(which(CD_enrich < 0.05))
cd_enriched_indiv

cd_df <- as.data.frame(CD_enrich)
colnames(cd_df) <- "pvalue"
cd_df$Tissue <- NA
cd_df$Tissue[1:119] <- "LCL"
cd_df$Tissue[120:238] <- "iPSC"
cd_df$Tissue[239:357] <- "iPSC-CM"
cd_df$Individual <- NA
cd_df$Individual <- c(c(1:119),c(1:119),c(1:119))

#pdf("banovich_CD_enrichment.pdf",width = 5, height = 3.5)
ggplot(cd_df) + geom_point(shape = 19, aes(x = reorder(Individual, log10(pvalue)), y = -log10(pvalue), col = Tissue),) + theme_classic()+ scale_color_manual(values=pallet) + geom_abline(slope = 0, intercept = -log10(0.05), linetype = "dashed", col = "black") + xlab("Individual") + theme( axis.text.x=element_blank(), axis.ticks.x=element_blank()) + labs(y = "-log10(p-value)", title = "Enrichment of disrupted TFs (d > 15) in CD genes",color = "Cell type") + annotate("text", x = 110, y = 1.4, label = "p = 0.05")

#dev.off()
```





# Some t tests and box plots
```{r}
tf_degrees <- tf_degrees_hard_thresh_scaled2
tf_degree_lcl <- melt(tf_degrees[,c(1,c(2:120))], id.vars = c("tf"))
tf_degree_lcl$tissue <- "LCL"
tf_degree_iPSC <- melt(tf_degrees[,c(1,c(121:239))], id.vars = c("tf"))
tf_degree_iPSC$tissue <- "iPSC"
tf_degree_iPSC_CM <- melt(tf_degrees[,c(1,c(240:358))], id.vars = c("tf"))
tf_degree_iPSC_CM$tissue <- "iPSC-CM"
tf_degrees_melted <- rbind(tf_degree_lcl,tf_degree_iPSC,tf_degree_iPSC_CM)
colnames(tf_degrees_melted) <- c("tf","individual","diff_out_degree","Tissue")

ggplot(tf_degrees_melted, aes(x=Tissue, y=diff_out_degree, fill=Tissue, col = Tissue)) + geom_violin() + theme_bw() + scale_fill_manual(values=pallet) + scale_color_manual(values=pallet) + labs(title="Distribution of TF Disruption Scores", x ="Cell type", y = "TF Disruption Score") + theme(legend.position="none")

ggplot(tf_degrees_melted, aes(x=diff_out_degree,  color=Tissue)) + geom_density() + theme_bw() + scale_color_manual(values=pallet) +labs(title=NULL, x ="Scaled TF disruption score", y = "Frequency") +  xlim(-0.5,0.5)

ggplot(tf_degrees_melted, aes(x=diff_out_degree,  color=Tissue)) + geom_density() + theme_bw() + scale_color_manual(values=pallet) +labs(title=NULL, x ="Scaled TF disruption score", y = "Frequency") +  xlim(-1,1)

ggplot(tf_degrees_melted, aes(x=diff_out_degree,  color=Tissue)) + geom_density() + theme_bw() + scale_color_manual(values=pallet) +labs(title=NULL, x ="Scaled TF disruption score", y = "Frequency") +  xlim(1,10)

ggplot(tf_degrees_melted, aes(x=diff_out_degree,  color=Tissue)) + geom_density() + theme_bw() + scale_color_manual(values=pallet) +labs(title=NULL, x ="Scaled TF disruption score", y = "Frequency") +  xlim(5,20)

ggplot(tf_degrees_melted, aes(x=diff_out_degree,  color=Tissue)) + geom_density() + theme_bw() + scale_color_manual(values=pallet) +labs(title=NULL, x ="Scaled TF disruption score", y = "Frequency") +  xlim(1,30)
```

```{r}
tf_degrees <- tf_degrees_hard_thresh_scaled2
colnames(tf_degree_iPSC_CM) <-  c("tf","individual","diff_out_degree","tissue")
tf_degree_iPSC_CM$CAD <- tf_degrees$GWAS_CAD_tf[match(tf_degree_iPSC_CM$tf, tf_degrees$tf)]

test_cad_in_cm <- t.test(tf_degree_iPSC_CM[which(tf_degree_iPSC_CM$CAD == 1),]$diff_out_degree,tf_degree_iPSC_CM[which(tf_degree_iPSC_CM$CAD == 0),]$diff_out_degree, alternative="greater")

mean0 <- mean(tf_degree_iPSC_CM[which(tf_degree_iPSC_CM$CAD == 0),]$diff_out_degree)
mean1 <- mean(tf_degree_iPSC_CM[which(tf_degree_iPSC_CM$CAD == 1),]$diff_out_degree)

p1 <- ggplot(tf_degree_iPSC_CM, aes(x = as.factor(CAD), y = diff_out_degree, group = as.factor(CAD), fill = as.factor(CAD))) + scale_fill_manual(values = pallet) + geom_violin() + theme_bw() + geom_point(x = as.factor(0), y = mean0, shape=18,size=5, color="#E41A1C")+ geom_point(x = as.factor(1), y = mean1, shape=18,size=5, color="#E41A1C") + labs(title = "Distribution of disruption scores for CAD and non-CAD TFs in iPSC-CMs",y = "TF disruption score", x = "CAD TF?") + theme(legend.position = "none")

p2 <- ggplot(tf_degree_iPSC_CM, aes(x = as.factor(CAD), y = diff_out_degree, group = as.factor(CAD), fill = as.factor(CAD))) + scale_fill_manual(values = pallet) + geom_violin() + theme_bw() + geom_point(x = as.factor(0), y = mean0, shape=18,size=5, color="#E41A1C")+ geom_point(x = as.factor(1), y = mean1, shape=18,size=5, color="#E41A1C")+ ylim(-1,1) + labs(title = "Distribution of disruption scores for CAD and non-CAD TFs in iPSC-CMs",y = "TF disruption score", x = "CAD TF?") + theme(legend.position = "none")

p3 <- grid.arrange(p1 + theme(legend.position="none"), p2 + theme(legend.position="none"), nrow=1, widths = c(3,3))
pdf("cad_1_0_ipsc_cm.pdf",width = 6, height = 3)
plot(p3)
dev.off()
plot(p3)

ggplot(tf_degree_iPSC_CM, aes(x = as.factor(CAD), y = diff_out_degree,group = as.factor(CAD))) + geom_jitter(position=position_jitter(0.2), size = 0.1) + theme_bw() + ylim(-1,1) + geom_point(aes(x = as.factor(0), y = mean0), shape=18,size=5, color="#E41A1C")+ geom_point(x = as.factor(1), y = mean1, shape=18,size=5, color="#E41A1C")


colnames(tf_degree_iPSC_CM) <-  c("tf","individual","diff_out_degree","tissue")
tf_degree_iPSC_CM$CD <- tf_degrees_hard_thresh$GWAS_CD_tf[match(tf_degree_iPSC_CM$tf, tf_degrees_hard_thresh$tf)]

test_cd_in_cm <- t.test(tf_degree_iPSC_CM[which(tf_degree_iPSC_CM$CD == 1),]$diff_out_degree,tf_degree_iPSC_CM[which(tf_degree_iPSC_CM$CD == 0),]$diff_out_degree, alternative="greater")

colnames(tf_degree_lcl) <-  c("tf","individual","diff_out_degree","tissue")
tf_degree_lcl$CD <- tf_degrees_hard_thresh$GWAS_CD_tf[match(tf_degree_lcl$tf, tf_degrees_hard_thresh$tf)]

test_cd_in_lcl <- t.test(tf_degree_lcl[which(tf_degree_lcl$CD == 1),]$diff_out_degree,tf_degree_lcl[which(tf_degree_lcl$CD == 0),]$diff_out_degree, alternative="greater")

colnames(tf_degree_lcl) <-  c("tf","individual","diff_out_degree","tissue")
tf_degree_lcl$CAD <- tf_degrees_hard_thresh$GWAS_CAD_tf[match(tf_degree_lcl$tf, tf_degrees_hard_thresh$tf)]

test_cad_in_lcl <- t.test(tf_degree_lcl[which(tf_degree_lcl$CAD == 1),]$diff_out_degree,tf_degree_lcl[which(tf_degree_lcl$CAD == 0),]$diff_out_degree, alternative="greater")

test_cad_in_lcl$p.value
test_cad_in_cm$p.value
test_cd_in_lcl$p.value
test_cd_in_cm$p.value
```

# Some basic-ass plots

```{r}
data <- tf_degrees_hard_thresh_scaled2
cad_tfs_degree_hard_thresh <- data[which(data$GWAS_CAD_tf == 1),c(1:358)]
cad_tfs_degree_hard_thresh_melted <- melt(cad_tfs_degree_hard_thresh)
colnames(cad_tfs_degree_hard_thresh_melted) <- c("tf","sample","tf_diff_degree")
cad_tfs_degree_hard_thresh_melted <- separate(cad_tfs_degree_hard_thresh_melted, sample, c("cellType","tag", "indiv"), "_", remove = TRUE)

p1 <- ggplot(cad_tfs_degree_hard_thresh_melted) + geom_point(shape = 19, aes(x = reorder(indiv, -tf_diff_degree), y = tf_diff_degree, col = cellType),) + theme_classic()+ scale_color_manual(values=pallet)+ xlab("Individual") + theme( axis.text.x=element_blank(), axis.ticks.x=element_blank())  + labs(y = "Scaled TF disruption score", title = "TF disruption scores for CAD genes",col = "Cell type",shape = "TF") 

p2 <- ggplot(cad_tfs_degree_hard_thresh_melted[cad_tfs_degree_hard_thresh_melted$tf_diff_degree>3,]) + geom_point(size = 3, aes(x = reorder(indiv, -tf_diff_degree), y = tf_diff_degree, col = cellType, size = cellType, shape = tf)) +  theme_classic()+ scale_color_manual(values=pallet)+ xlab("Individual") + theme( axis.text.x=element_blank(), axis.ticks.x=element_blank())  + labs(y = "Scaled TF disruption score", title = "TF disruption scores for top CAD genes",col = "Cell type",shape = "TF") 
#dev.off()

p3 <- grid.arrange(p1 + theme(legend.position="none"), p2 + theme(legend.position="right"), nrow=1, widths = c(3,4))
pdf("top_d_cad_genes.pdf",width = 8, height = 3)
plot(p3)
dev.off()
plot(p3)


# whcih individuals have the ipsc cm disruptins for ERG
indivs <- cad_tfs_degree_hard_thresh_melted[which((cad_tfs_degree_hard_thresh_melted$tf_diff_degree>5) & (cad_tfs_degree_hard_thresh_melted$cellType == "iPSC-CM")),c('indiv','tf_diff_degree')]
indivs

temp <- cad_tfs_degree_hard_thresh_melted[which((cad_tfs_degree_hard_thresh_melted$tf == "ERG") & (cad_tfs_degree_hard_thresh_melted$cellType == "iPSC-CM")),c('indiv','tf_diff_degree')]
```
This is the eQTL for ERG
ERG	ENSG00000159176	201476815	chr1	0.461

gene ENSG00000159176 is associated with some ehart disease
https://www.genecards.org/cgi-bin/carddisp.pl?gene=CSRP1

https://www-sciencedirect-com.ezp-prod1.hul.harvard.edu/science/article/pii/S1534580702003969

Let's look at expression
```{r}
indivs <- cad_tfs_degree_hard_thresh_melted[which((cad_tfs_degree_hard_thresh_melted$tf_diff_degree>1) & (cad_tfs_degree_hard_thresh_melted$cellType == "iPSC-CM")),c('indiv','tf_diff_degree')]
indivs
load("../inputs/LCL_expr_data_finalEgret_v1_banovich_06172020.RData")
expr_lcl <- expr_df
colnames(expr_lcl) <- sub("X","NA",colnames(expr_lcl))
load("../inputs/iPSC_expr_data_finalEgret_v1_banovich_06172020.RData")
expr_ipsc <- expr_df
load("../inputs/iPSC-CM_expr_data_finalEgret_v1_banovich_06172020.RData")
expr_ipscCM <- expr_qqnorm

genotypes <- read.table("../data/genotypes/genotypes", header = FALSE)
genotype_names <- as.data.frame(t(genotypes[c(10:128)]))
genotype_names$name <- paste0("NA",genotype_names$V1)
indivs_cad <- genotype_names[as.numeric(indivs$indiv),2]
indivs$name <- genotype_names[as.numeric(indivs$indiv),2]
temp$name <- genotype_names[as.numeric(temp$indiv),2]
intersection_gentypes <- colnames(expr_lcl)[which((colnames(expr_lcl) %in% colnames(expr_ipsc)) & (colnames(expr_lcl) %in% colnames(expr_ipscCM)))]

lcl_vector <- expr_lcl[which(rownames(expr_lcl) == "ENSG00000159176"),which(colnames(expr_lcl) %in% intersection_gentypes)]
ipsc_vector <- expr_ipsc[which(rownames(expr_ipsc) == "ENSG00000159176"),which(colnames(expr_ipsc) %in% intersection_gentypes)]
ipscCM_vector <- expr_ipscCM[which(rownames(expr_ipscCM) == "ENSG00000159176"),which(colnames(expr_ipscCM) %in% intersection_gentypes)]

which(intersection_gentypes %in% indivs_cad)

#lcl_vector$cellType <- "LCL"
#ipsc_vector$cellType <- "iPSC"
#ipscCM_vector$cellType <- "iPSC-CM"
matrix <- melt(ipscCM_vector)
matrix$indiv <- rownames(matrix)
colnames(matrix)[1] <- c("expr")
matrix$num <-  temp$indiv[match(matrix$indiv, temp$name)]
matrix$cad <- ifelse(matrix$indiv %in% indivs_cad, "cad","no")
matrix$diffscore <-  temp$tf_diff_degree[match(matrix$indiv, temp$name)]
ggplot(matrix) + geom_point(size = 3, aes(x = indiv, y = expr, col = diffscore, shape = cad)) +  theme_classic()+ xlab("Individual") 
```

```{r}
data <- tf_degrees_hard_thresh_scaled2
disease_test_tfs_degree_hard_thresh <- data[which(data$GWAS_CD   == 1),c(1:358)]
disease_test_tfs_degree_hard_thresh_melted <- melt(disease_test_tfs_degree_hard_thresh)
colnames(disease_test_tfs_degree_hard_thresh_melted) <- c("tf","sample","tf_diff_degree")
disease_test_tfs_degree_hard_thresh_melted <- separate(disease_test_tfs_degree_hard_thresh_melted, sample, c("cellType","tag", "indiv"), "_", remove = TRUE)

p1 <- ggplot(disease_test_tfs_degree_hard_thresh_melted) + geom_point(shape = 19, aes(x = reorder(indiv, -tf_diff_degree), y = tf_diff_degree, col = cellType),) + theme_classic()+ scale_color_manual(values=pallet)+ xlab("Individual") + theme( axis.text.x=element_blank(), axis.ticks.x=element_blank())  + labs(y = "Scaled TF disruption score", title = "TF disruption scores for CD genes",col = "Cell type",shape = "TF")

ggplot(disease_test_tfs_degree_hard_thresh_melted[disease_test_tfs_degree_hard_thresh_melted$tf_diff_degree > 2,]) + geom_point(shape = 19, aes(x = reorder(indiv, -tf_diff_degree), y = tf_diff_degree, col = cellType),) + theme_classic()+ scale_color_manual(values=pallet[3])+ xlab("Individual") + theme( axis.text.x=element_blank(), axis.ticks.x=element_blank())  + labs(y = "Scaled TF disruption score", title = "TF disruption scores for CD genes",col = "Cell type",shape = "TF") 
#dev.off()

p2 <- ggplot(disease_test_tfs_degree_hard_thresh_melted[disease_test_tfs_degree_hard_thresh_melted$tf_diff_degree >3,]) + geom_point(size = 3, aes(x = reorder(indiv, -tf_diff_degree), y = tf_diff_degree, col = cellType, shape = tf)) +  theme_classic()+ scale_color_manual(values=pallet[3])+xlab("Individual") + theme( axis.text.x=element_blank(), axis.ticks.x=element_blank())  + labs(y = "Scaled TF disruption score", title = "TF disruption scores for top CD genes",col = "Cell type",shape = "TF") 


p3 <- grid.arrange(p1 + theme(legend.position="right"), p2 + theme(legend.position="right"), nrow=1, widths = c(5,5))
pdf("top_d_cd_genes.pdf",width = 8, height = 3)
plot(p3)
dev.off()
plot(p3)

```




```{r}
gene_degrees <- gene_degrees_hard_thresh_scaled
gene_degree_lcl <- melt(gene_degrees[,c(1,c(2:120))], id.vars = c("gene"))
gene_degree_lcl$tissue <- "LCL"
gene_degree_iPSC <- melt(gene_degrees[,c(1,c(121:239))], id.vars = c("gene"))
gene_degree_iPSC$tissue <- "iPSC"
gene_degree_iPSC_CM <- melt(gene_degrees[,c(1,c(240:358))], id.vars = c("gene"))
gene_degree_iPSC_CM$tissue <- "iPSC-CM"
gene_degrees_melted <- rbind(gene_degree_lcl,gene_degree_iPSC,gene_degree_iPSC_CM)
colnames(gene_degrees_melted) <- c("gene","individual","diff_out_degree","Tissue")

ggplot(gene_degrees_melted, aes(x=Tissue, y=diff_out_degree, fill=Tissue, col = Tissue)) + geom_violin() + theme_bw() + scale_fill_manual(values=pallet) + scale_color_manual(values=pallet) + labs(title="Distribution of gene Disruption Scores", x ="Cell type", y = "gene Disruption Score") + theme(legend.position="none")

ggplot(gene_degrees_melted, aes(x=diff_out_degree,col = Tissue, fill=Tissue, group = Tissue)) + geom_density(alpha = 0.1, adjust = 0.25) + theme_bw() + scale_fill_manual(values=pallet) + scale_color_manual(values=pallet) +labs(title=NULL, x ="Scaled gene disruption score", y = "Frequency")

ggplot(gene_degrees_melted, aes(x=diff_out_degree,col = Tissue, fill=Tissue, group = Tissue)) + geom_density(alpha = 0.1, adjust = 0.5) + theme_bw() + scale_fill_manual(values=pallet) + scale_color_manual(values=pallet) +labs(title=NULL, x ="Scaled gene disruption score", y = "Frequency") +  xlim(0.1, 50)

```

```{r}
data <- gene_degrees_hard_thresh_scaled
cad_tfs_degree_hard_thresh <- data[which(data$GWAS_CAD_gene == 1),c(1:358)]
cad_tfs_degree_hard_thresh_melted <- melt(cad_tfs_degree_hard_thresh)
colnames(cad_tfs_degree_hard_thresh_melted) <- c("tf","sample","tf_diff_degree")
cad_tfs_degree_hard_thresh_melted <- separate(cad_tfs_degree_hard_thresh_melted, sample, c("cellType","tag", "indiv"), "_", remove = TRUE)

ggplot(cad_tfs_degree_hard_thresh_melted) + geom_point(shape = 19, aes(x = reorder(indiv, -tf_diff_degree), y = tf_diff_degree, col = cellType),) + theme_classic()+ scale_color_manual(values=pallet)+ xlab("Individual") #+ theme( axis.text.x=element_blank(), axis.ticks.x=element_blank())  + labs(y = "-log10(p-value)", title = "Enrichment of disrupted TFs in CAD genes",color = "Cell type") + annotate("text", x = 110, y = 1.4, label = "p = 0.05")
#dev.off()


ggplot(cad_tfs_degree_hard_thresh_melted[cad_tfs_degree_hard_thresh_melted$tf_diff_degree>2,]) + geom_point(shape = 19, aes(x = reorder(indiv, -tf_diff_degree), y = tf_diff_degree, col = cellType),) + theme_classic()+ scale_color_manual(values=pallet)+ xlab("Individual") #+ theme( axis.text.x=element_blank(), axis.ticks.x=element_blank())  + labs(y = "-log10(p-value)", title = "Enrichment of disrupted TFs in CAD genes",color = "Cell type") + annotate("text", x = 110, y = 1.4, label = "p = 0.05")
#dev.off()
```

```{r}
data <- gene_degrees_hard_thresh_scaled
cd_tfs_degree_hard_thresh <- data[which(data$GWAS_CD_gene == 1),c(1:358)]
cd_tfs_degree_hard_thresh_melted <- melt(cd_tfs_degree_hard_thresh)
colnames(cd_tfs_degree_hard_thresh_melted) <- c("tf","sample","tf_diff_degree")
cd_tfs_degree_hard_thresh_melted <- separate(cd_tfs_degree_hard_thresh_melted, sample, c("cellType","tag", "indiv"), "_", remove = TRUE)

ggplot(cad_tfs_degree_hard_thresh_melted) + geom_point(shape = 19, aes(x = reorder(indiv, -tf_diff_degree), y = tf_diff_degree, col = cellType),) + theme_classic()+ scale_color_manual(values=pallet)+ xlab("Individual") #+ theme( axis.text.x=element_blank(), axis.ticks.x=element_blank())  + labs(y = "-log10(p-value)", title = "Enrichment of disrupted TFs in CAD genes",color = "Cell type") + annotate("text", x = 110, y = 1.4, label = "p = 0.05")
#dev.off()

```

```{r}
data <- gene_degrees_hard_thresh_scaled
test_tfs_degree_hard_thresh <- data[which(data$gene=="ENSG00000177000"),c(1:358)]
cd_tfs_degree_hard_thresh_melted <- melt(cd_tfs_degree_hard_thresh)
colnames(cd_tfs_degree_hard_thresh_melted) <- c("tf","sample","tf_diff_degree")
cd_tfs_degree_hard_thresh_melted <- separate(cd_tfs_degree_hard_thresh_melted, sample, c("cellType","tag", "indiv"), "_", remove = TRUE)

ggplot(cad_tfs_degree_hard_thresh_melted) + geom_point(shape = 19, aes(x = reorder(indiv, -tf_diff_degree), y = tf_diff_degree, col = cellType),) + theme_classic()+ scale_color_manual(values=pallet)+ xlab("Individual") #+ theme( axis.text.x=element_blank(), axis.ticks.x=element_blank())  + labs(y = "-log10(p-value)", title = "Enrichment of disrupted TFs in CAD genes",color = "Cell type") + annotate("text", x = 110, y = 1.4, label = "p = 0.05")
#dev.off()

```