---
title: "Final EGRET v1 Banovich application - Data Analysis"
author: "Deborah Weighill"
date: "June 30, 2020"
output:
  html_notebook:
    toc: true
    theme: cosmo
    df_print: paged
---

# Overview
Here we do some analysis on the banovich results for the "application". We start from the "master difference table" that we created.



# Load the master table
```{r}
load("../outputs/diff_table_with_gwas_banovich_06302020.RData")
```


# TF/gene degree differences from PANDA
This section will answer the question: *"Which TFs overall are impacted the most for each individual and each tissue?"* This will be determined by calculating the **out-degree** of each TF in each individual **difference network**. In other words, for each individual EGRET network, we take *abs(Eij-Pij)* for each edge, and then we sum the "*difference edges"* around each tf, per individual. Similarly, we will dermine the **in-degree** of each gene in the difference network.

```{r}
tf_degrees <- aggregate(diff_table[,c(4:363)], by = list(diff_table$tf), FUN = sum) 
gene_degrees <- aggregate(diff_table[,c(4:363)], by = list(diff_table$gene), FUN = sum) 
colnames(tf_degrees)[1] <- "tf"
colnames(gene_degrees)[1] <- "gene"
head(tf_degrees)
head(gene_degrees)
```

Now load the gwas results each into a data frame

```{r}

# load gene annotation to assign gene id to gene name
nameGeneMap <- read.table("../../final_egret_v1/annotation/geneID_name_map.txt", header = FALSE)
colnames(nameGeneMap) <- c("gene","name")

RA <- read.table("../gwas/RA_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(RA) <- c("RAtrait","name")
RA$gene <- nameGeneMap$gene[match(RA$name, nameGeneMap$name)]

CD <- read.table("../gwas/CD_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(CD) <- c("CDtrait","name")
CD$gene <- nameGeneMap$gene[match(CD$name, nameGeneMap$name)]

CAD <- read.table("../gwas/CAD_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(CAD) <- c("CADtrait","name")
CAD$gene <- nameGeneMap$gene[match(CAD$name, nameGeneMap$name)]

BMI <- read.table("../gwas/BMI_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(BMI) <- c("BMItrait","name")
BMI$gene <- nameGeneMap$gene[match(BMI$name, nameGeneMap$name)]

MS <- read.table("../gwas/MS_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(MS) <- c("MStrait","name")
MS$gene <- nameGeneMap$gene[match(MS$name, nameGeneMap$name)]

MI <- read.table("../gwas/MI_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(MI) <- c("MItrait","name")
MI$gene <- nameGeneMap$gene[match(MI$name, nameGeneMap$name)]
```

Now load the difference table we made before, and merge in the GWAS traits
```{r}
tf_degrees$GWAS_RA_tf <- ifelse(tf_degrees$tf %in% RA$name, 1, 0)
gene_degrees$GWAS_RA_gene <- ifelse(gene_degrees$gene %in% RA$gene, 1, 0)

tf_degrees$GWAS_CD_tf <- ifelse(tf_degrees$tf %in% CD$name, 1, 0)
gene_degrees$GWAS_CD_gene <- ifelse(gene_degrees$gene %in% CD$gene, 1, 0)

tf_degrees$GWAS_CAD_tf <- ifelse(tf_degrees$tf %in% CAD$name, 1, 0)
gene_degrees$GWAS_CAD_gene <- ifelse(gene_degrees$gene %in% CAD$gene, 1, 0)

tf_degrees$GWAS_BMI_tf <- ifelse(tf_degrees$tf %in% BMI$name, 1, 0)
gene_degrees$GWAS_BMI_gene <- ifelse(gene_degrees$gene %in% BMI$gene, 1, 0)

tf_degrees$GWAS_MI_tf <- ifelse(tf_degrees$tf %in% MI$name, 1, 0)
gene_degrees$GWAS_MI_gene <- ifelse(gene_degrees$gene %in% MI$gene, 1, 0)

tf_degrees$GWAS_MS_tf <- ifelse(tf_degrees$tf %in% MS$name, 1, 0)
gene_degrees$GWAS_MS_gene <- ifelse(gene_degrees$gene %in% MS$gene, 1, 0)

save(tf_degrees, file = "../outputs/tf_diff_degree_table_banovich_06302020.RData")
save(gene_degrees, file = "../outputs/gene_diff_degree_table_banovich_06302020.RData")
load("../outputs/tf_diff_degree_table_banovich_06302020.RData")
#load("../outputs/")
```



We can now plot distributions of tf degree and gene degree in difference network for each tissue.
```{r}
library(ggplot2)
library(reshape2)
colnames(tf_degrees)[1] <- "tf"
tf_degree_lcl <- melt(tf_degrees[,c(1,c(3:121))], id.vars = c("tf"))
tf_degree_lcl$tissue <- "LCL"
tf_degree_iPSC <- melt(tf_degrees[,c(1,c(123:241))], id.vars = c("tf"))
tf_degree_iPSC$tissue <- "iPSC"
tf_degree_iPSC_CM <- melt(tf_degrees[,c(1,c(243:361))], id.vars = c("tf"))
tf_degree_iPSC_CM$tissue <- "iPSC-CM"
tf_degrees_melted <- rbind(tf_degree_lcl,tf_degree_iPSC,tf_degree_iPSC_CM)
colnames(tf_degrees_melted) <- c("tf","individual","diff_out_degree","tissue")
ggplot(tf_degrees_melted, aes(x=tissue, y=diff_out_degree, color=tissue)) + geom_boxplot() + theme_linedraw() + scale_color_manual(values=c("red", "blue","orange","cyan", "darkgreen", "violet")) + labs(title="Distribution of TF Disruption Scores", x ="Cell type", y = "TF Disruption Score") 

ggplot(tf_degrees_melted[which(tf_degrees_melted$tissue == "iPSC"),], aes(x=tissue, y=diff_out_degree, color=tissue)) + geom_violin() + theme_linedraw() + scale_color_manual(values=c("red")) + labs(title="Distribution of TF Disruption Scores", x = NULL, y = "TF Disruption Score") + theme(legend.position = "right")

ggplot(tf_degrees_melted[which(tf_degrees_melted$tissue == "iPSC-CM"),], aes(x=tissue, y=diff_out_degree, color=tissue)) + geom_violin() + theme_linedraw() + scale_color_manual(values=c("blue")) + labs(title="Distribution of TF Disruption Scores", x =NULL, y = "TF Disruption Score") + theme(legend.position = "right")

ggplot(tf_degrees_melted[which(tf_degrees_melted$tissue == "LCL"),], aes(x=tissue, y=diff_out_degree, color=tissue)) + geom_violin() + theme_linedraw() + scale_color_manual(values=c("orange")) + labs(title="Distribution of TF Disruption Scores", x =NULL, y = "TF Disruption Score") + theme(legend.position = "right")

ggplot(tf_degrees_melted[which(tf_degrees_melted$individual == "iPSC-CM_indiv_21"),], aes(x=tissue, y=diff_out_degree, color=tissue)) + geom_violin() + theme_linedraw() + scale_color_manual(values=c("blue")) + labs(title="Distribution of TF Disruption Scores", x =NULL, y = "TF Disruption Score") + theme(legend.position = "right")

```

```{r}
colnames(gene_degrees)[1] <- "gene"
gene_degree_lcl <- melt(gene_degrees[,c(1,c(3,121))], id.vars = c("gene"))
gene_degree_lcl$tissue <- "LCL"
gene_degree_iPSC <- melt(gene_degrees[,c(1,c(123,241))], id.vars = c("gene"))
gene_degree_iPSC$tissue <- "iPSC"
gene_degree_iPSC_CM <- melt(gene_degrees[,c(1,c(243,361))], id.vars = c("gene"))
gene_degree_iPSC_CM$tissue <- "iPSC-CM"
gene_degrees_melted <- rbind(gene_degree_lcl,gene_degree_iPSC,gene_degree_iPSC_CM)
colnames(gene_degrees_melted) <- c("gene","individual","diff_in_degree","tissue")
ggplot(gene_degrees_melted, aes(x=tissue, y=diff_in_degree, color=tissue)) + geom_boxplot() + theme_linedraw() + scale_color_manual(values=c("red", "blue","orange","cyan", "darkgreen", "violet")) + labs(title="Distribution of Gene Disruption Scores", x ="Cell type", y = "Gene Disruption Score")

```

# Do a freakin t-test - genes
```{r}
t_test_genes <- gene_degrees[,c(1,362,363,364,365, 366, 367)]
t_test_genes$lcl_vs_ipsc <- NA
t_test_genes$ipsc_vs_ipscCM <- NA
t_test_genes$lcl_vs_ipscCM <- NA


for (i in 1:(dim(t_test_genes)[1])){
  x <- gene_degrees[i,]
  lcl_vs_ipsc <- t.test(as.numeric(as.vector(x[c(3:121)])), as.numeric(as.vector(x[c(123:241)])), alternative = "two.sided")
  ipsc_vs_ipscCM <- t.test(as.numeric(as.vector(x[c(123:241)])), as.numeric(as.vector(x[c(243:361)])), alternative = "two.sided")
  lcl_vs_ipscCM <- t.test(as.numeric(as.vector(x[c(3:121)])), as.numeric(as.vector(x[c(243:361)])), alternative = "two.sided")
  t_test_genes$lcl_vs_ipsc[i] <- lcl_vs_ipsc$p.value
  t_test_genes$ipsc_vs_ipscCM[i] <- ipsc_vs_ipscCM$p.value
  t_test_genes$lcl_vs_ipscCM[i] <- lcl_vs_ipscCM$p.value
}
save(t_test_genes, file = "../outputs/t_test_gene_degree_diff_banovich_06302020.RData")

```


# Do a  t-test - tfs
```{r}
t_test_tfs <- tf_degrees[,c(1,362,363,364,365,366,367)]
t_test_tfs$lcl_vs_ipsc <- NA
t_test_tfs$ipsc_vs_ipscCM <- NA
t_test_tfs$lcl_vs_ipscCM <- NA


for (i in 1:(dim(t_test_tfs)[1])){
  x <- tf_degrees[i,]
  lcl_vs_ipsc <- t.test(as.numeric(as.vector(x[c(3:121)])), as.numeric(as.vector(x[c(123:241)])), alternative = "two.sided")
  ipsc_vs_ipscCM <- t.test(as.numeric(as.vector(x[c(123:241)])), as.numeric(as.vector(x[c(243:361)])), alternative = "two.sided")
  lcl_vs_ipscCM <- t.test(as.numeric(as.vector(x[c(3:121)])), as.numeric(as.vector(x[c(243:361)])), alternative = "two.sided")
  t_test_tfs$lcl_vs_ipsc[i] <- lcl_vs_ipsc$p.value
  t_test_tfs$ipsc_vs_ipscCM[i] <- ipsc_vs_ipscCM$p.value
  t_test_tfs$lcl_vs_ipscCM[i] <- lcl_vs_ipscCM$p.value
}
save(t_test_tfs, file = "../outputs/t_test_TFs_degree_diff_banovich_06302020.RData")
#load("../outputs/")

t_test_tfs$lcl_vs_ipsc_adjust <- p.adjust(t_test_tfs$lcl_vs_ipsc, method = "fdr")
t_test_tfs$lcl_vs_ipscCM_adjust <- p.adjust(t_test_tfs$lcl_vs_ipscCM, method = "fdr")
t_test_tfs$ipsc_vs_ipscCM_adjust <- p.adjust(t_test_tfs$ipsc_vs_ipscCM, method = "fdr")

# look at tfs which are GWAS hits for CAD - see what the t.tests say about difference in tf affects in different tissues. 
t_test_tfs[which(t_test_tfs$GWAS_CAD_tf == 1),]
```

```{r}
iPSC_CM_indiv <- colnames(tf_degrees)[243:361]
ttest_iPSC_CM_indiv <- as.data.frame(iPSC_CM_indiv)
colnames(ttest_iPSC_CM_indiv) <- "individual"
ttest_iPSC_CM_indiv$CAD <- NA

for (i in 1:119){
  tfs_gwas <- tf_degrees[which(tf_degrees$GWAS_CAD_tf == 1),(242+i)]
  tfs_not_gwas <- tf_degrees[which(tf_degrees$GWAS_CAD_tf == 0),(242+i)]
  ttest <- t.test(as.numeric(as.vector(tfs_gwas)), as.numeric(as.vector(tfs_not_gwas)), alternative = "greater")
  ttest_iPSC_CM_indiv[i,2] <- ttest$p.value
}


t_test_tfs$lcl_vs_ipsc_adjust <- p.adjust(t_test_tfs$lcl_vs_ipsc, method = "fdr")
t_test_tfs$lcl_vs_ipscCM_adjust <- p.adjust(t_test_tfs$lcl_vs_ipscCM, method = "fdr")
t_test_tfs$ipsc_vs_ipscCM_adjust <- p.adjust(t_test_tfs$ipsc_vs_ipscCM, method = "fdr")

# look at tfs which are GWAS hits for CAD - see what the t.tests say about difference in tf affects in different tissues. 
t_test_tfs[which(t_test_tfs$GWAS_CAD_tf == 1),]
```

# Enrichment tests on TFs
Let's try and see, for a given individual, are "high difference" TFs enriched in GWAS hits.

## BMI
```{r}
d = 10
#load("../outputs/tf_diff_degree_table.RData")
individuals <- c(c(3:121),c(123:241),c(243:361))
BMI_enrich <- sapply(individuals, FUN = function(i){
  high_diff_gwas <- length(which((tf_degrees[,i] >d) & (tf_degrees$GWAS_BMI_tf == 1)))
  low_diff_gwas <- length(which((tf_degrees[,i] <=d) & (tf_degrees$GWAS_BMI_tf == 1)))
  high_diff_notGwas <- length(which((tf_degrees[,i] >d) & (tf_degrees$GWAS_BMI_tf == 0)))
  low_diff_notGwas <- length(which((tf_degrees[,i] <=d) & (tf_degrees$GWAS_BMI_tf == 0)))
  contingency_table <- matrix(c(high_diff_gwas,low_diff_gwas,high_diff_notGwas,low_diff_notGwas), nrow = 2,dimnames = list(Diff = c("High", "Low"),GWAS = c("yes", "no")))
  test <- fisher.test(contingency_table, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(BMI_enrich) <- colnames(tf_degrees)[individuals]
bmi_enrich_individuals <- names(which(BMI_enrich < 0.05))
bmi_enrich_individuals


```

try a t test
```{r}
individuals <- c(c(3:121),c(123:241),c(243:361))
BMI_ttest <- sapply(individuals, FUN = function(i){
  gwas <- tf_degrees[which(tf_degrees$GWAS_BMI_tf == 1),i]
  notgwas <- tf_degrees[which(tf_degrees$GWAS_BMI_tf == 0),i]
  test <- t.test(gwas, notgwas, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(BMI_ttest) <- colnames(tf_degrees)[individuals]
BML_ttest_individuals <- names(which(BMI_ttest < 0.05))
BML_ttest_individuals
```

## CD
```{r}
d = 10
individuals <- c(c(3:121),c(123:241),c(243:361))
CD_enrich <- sapply(individuals, FUN = function(i){
  high_diff_gwas <- length(which((tf_degrees[,i] >d) & (tf_degrees$GWAS_CD_tf == 1)))
  low_diff_gwas <- length(which((tf_degrees[,i] <=d) & (tf_degrees$GWAS_CD_tf == 1)))
  high_diff_notGwas <- length(which((tf_degrees[,i] >d) & (tf_degrees$GWAS_CD_tf == 0)))
  low_diff_notGwas <- length(which((tf_degrees[,i] <=d) & (tf_degrees$GWAS_CD_tf == 0)))
  contingency_table <- matrix(c(high_diff_gwas,low_diff_gwas,high_diff_notGwas,low_diff_notGwas), nrow = 2,dimnames = list(Diff = c("High", "Low"),GWAS = c("yes", "no")))
  test <- fisher.test(contingency_table, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(CD_enrich) <- colnames(tf_degrees)[individuals]
cd_enriched_indiv <- names(which(CD_enrich < 0.05))
cd_enriched_indiv
```

```{r}
individuals <- c(c(3:121),c(123:241),c(243:361))
CD_ttest <- sapply(individuals, FUN = function(i){
  gwas <- tf_degrees[which(tf_degrees$GWAS_CD_tf == 1),i]
  notgwas <- tf_degrees[which(tf_degrees$GWAS_CD_tf == 0),i]
  test <- t.test(gwas, notgwas, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(CD_ttest) <- colnames(tf_degrees)[individuals]
CD_ttest_individuals <- names(which(CD_ttest < 0.05))
CD_ttest_individuals
```

## CAD
```{r}
d = 5
individuals <- c(c(3:121),c(123:241),c(243:361))
CAD_enrich <- sapply(individuals, FUN = function(i){
  high_diff_gwas <- length(which((tf_degrees[,i] >d) & (tf_degrees$GWAS_CAD_tf == 1)))
  low_diff_gwas <- length(which((tf_degrees[,i] <=d) & (tf_degrees$GWAS_CAD_tf == 1)))
  high_diff_notGwas <- length(which((tf_degrees[,i] >d) & (tf_degrees$GWAS_CAD_tf == 0)))
  low_diff_notGwas <- length(which((tf_degrees[,i] <=d) & (tf_degrees$GWAS_CAD_tf == 0)))
  contingency_table <- matrix(c(high_diff_gwas,low_diff_gwas,high_diff_notGwas,low_diff_notGwas), nrow = 2,dimnames = list(Diff = c("High", "Low"),GWAS = c("yes", "no")))
  test <- fisher.test(contingency_table, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(CAD_enrich) <- colnames(tf_degrees)[individuals]

cad_enriched_indiv <- names(which(CAD_enrich < 0.05))
cad_enriched_indiv
cad_df <- as.data.frame(CAD_enrich)
colnames(cad_df) <- "pvalue"
cad_df$Tissue <- NA
cad_df$Tissue[1:119] <- "LCL"
cad_df$Tissue[120:238] <- "iPSC"
cad_df$Tissue[239:357] <- "iPSC-CM"
cad_df$Individual <- NA
cad_df$Individual <- c(c(1:119),c(1:119),c(1:119))
ggplot(cad_df) + geom_point(shape = 19, aes(x = reorder(Individual, pvalue), y = -log10(pvalue), col = Tissue),) + theme_classic()+ scale_color_manual(values=c("red", "blue","orange","cyan", "darkgreen", "violet")) + geom_abline(slope = 0, intercept = -log10(0.05), linetype = "dashed", col = "black") + xlab("Individual") + theme( axis.text.x=element_blank(), axis.ticks.x=element_blank())
```



## MI
```{r}
d = 5
individuals <- c(c(3:121),c(123:241),c(243:361))
MI_enrich <- sapply(individuals, FUN = function(i){
  high_diff_gwas <- length(which((tf_degrees[,i] >d) & (tf_degrees$GWAS_MI_tf == 1)))
  low_diff_gwas <- length(which((tf_degrees[,i] <=d) & (tf_degrees$GWAS_MI_tf == 1)))
  high_diff_notGwas <- length(which((tf_degrees[,i] >d) & (tf_degrees$GWAS_MI_tf == 0)))
  low_diff_notGwas <- length(which((tf_degrees[,i] <=d) & (tf_degrees$GWAS_MI_tf == 0)))
  contingency_table <- matrix(c(high_diff_gwas,low_diff_gwas,high_diff_notGwas,low_diff_notGwas), nrow = 2,dimnames = list(Diff = c("High", "Low"),GWAS = c("yes", "no")))
  test <- fisher.test(contingency_table, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(MI_enrich) <- colnames(tf_degrees)[individuals]

cad_enriched_indiv <- names(which(MI_enrich < 0.05))
cad_enriched_indiv
```
```{r}
individuals <- c(c(3:121),c(123:241),c(243:361))
CAD_ttest <- sapply(individuals, FUN = function(i){
  gwas <- tf_degrees[which(tf_degrees$GWAS_CAD_tf == 1),i]
  notgwas <- tf_degrees[which(tf_degrees$GWAS_CAD_tf == 0),i]
  test <- t.test(gwas, notgwas, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(CAD_ttest) <- colnames(tf_degrees)[individuals]
CAD_ttest_individuals <- names(which(CAD_ttest < 0.05))
CAD_ttest_individuals
```

## RA
```{r}
d = 10
individuals <- c(c(3:121),c(123:241),c(243:361))
RA_enrich <- sapply(individuals, FUN = function(i){
  high_diff_gwas <- length(which((tf_degrees[,i] >d) & (tf_degrees$GWAS_RA_tf == 1)))
  low_diff_gwas <- length(which((tf_degrees[,i] <=d) & (tf_degrees$GWAS_RA_tf == 1)))
  high_diff_notGwas <- length(which((tf_degrees[,i] >d) & (tf_degrees$GWAS_RA_tf == 0)))
  low_diff_notGwas <- length(which((tf_degrees[,i] <=d) & (tf_degrees$GWAS_RA_tf == 0)))
  contingency_table <- matrix(c(high_diff_gwas,low_diff_gwas,high_diff_notGwas,low_diff_notGwas), nrow = 2,dimnames = list(Diff = c("High", "Low"),GWAS = c("yes", "no")))
  test <- fisher.test(contingency_table, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(RA_enrich) <- colnames(tf_degrees)[individuals]
ra_enriched_indiv <- names(which(RA_enrich < 0.05))
ra_enriched_indiv
```

```{r}
individuals <- c(c(3:121),c(123:241),c(243:361))
RA_ttest <- sapply(individuals, FUN = function(i){
  gwas <- tf_degrees[which(tf_degrees$GWAS_RA_tf == 1),i]
  notgwas <- tf_degrees[which(tf_degrees$GWAS_RA_tf == 0),i]
  test <- t.test(gwas, notgwas, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(RA_ttest) <- colnames(tf_degrees)[individuals]
RA_ttest_individuals <- names(which(RA_ttest < 0.05))
RA_ttest_individuals
```

```{r}
lcl_indiv <- colnames(tf_degrees)[c(3:121)]
ipsc_indiv <- colnames(tf_degrees)[c(123:241)]
ipsc_cm_indiv <- colnames(tf_degrees)[c(243:361)]

# CM individuals enriched in cardiometabolic traits
cm_trait <- length(which((ipsc_cm_indiv %in% bmi_enrich_individuals) | (ipsc_cm_indiv %in% cad_enriched_indiv)))
Ncm_trait <- length(which((lcl_indiv %in% bmi_enrich_individuals) | (lcl_indiv %in% cad_enriched_indiv))) + length(which((ipsc_indiv %in% bmi_enrich_individuals) | (ipsc_indiv %in% cad_enriched_indiv)))
cm_Ntrait <- length(which(!((ipsc_cm_indiv %in% bmi_enrich_individuals) | (ipsc_cm_indiv %in% cad_enriched_indiv))))
Ncm_Ntrait <- length(which(!((lcl_indiv %in% bmi_enrich_individuals) | (lcl_indiv %in% cad_enriched_indiv)))) + length(which(!((ipsc_indiv %in% bmi_enrich_individuals) | (ipsc_indiv %in% cad_enriched_indiv))))

contingency_table <- matrix(c(cm_trait,cm_Ntrait,Ncm_trait,Ncm_Ntrait), nrow = 2,dimnames = list(Trait = c("yes", "no"), Cell_type = c("CM", "Not CM")))


test <- fisher.test(contingency_table, alternative = "greater")
  test
  
y <- c((27/119),(22/238))
x <- c("iPSC-CM","LCL+iPSC")
df <- data.frame(Tissue=c("iPSC-CM","LCL+iPSC"),Fraction_TFs=c((27/119),(22/238)))
ggplot(data=df, aes(x=Tissue, y=Fraction_TFs, fill = Tissue)) + geom_bar(stat="identity") + theme_linedraw()+ scale_fill_manual(values=c("blue", "gold","orange","cyan", "darkgreen", "violet"))
  
```

# Enrichment tests on genes
Let's try and see, for a given individual, are "high difference" TFs enriched in GWAS hits.

## BMI
```{r}
d = 10
individuals <- c(c(3:121),c(123:241),c(243:361))
BMI_enrich_genes <- sapply(individuals, FUN = function(i){
  high_diff_gwas <- length(which((gene_degrees[,i] >d) & (gene_degrees$GWAS_BMI_tf == 1)))
  low_diff_gwas <- length(which((gene_degrees[,i] <=d) & (gene_degrees$GWAS_BMI_tf == 1)))
  high_diff_notGwas <- length(which((gene_degrees[,i] >d) & (gene_degrees$GWAS_BMI_tf == 0)))
  low_diff_notGwas <- length(which((gene_degrees[,i] <=d) & (gene_degrees$GWAS_BMI_tf == 0)))
  contingency_table <- matrix(c(high_diff_gwas,low_diff_gwas,high_diff_notGwas,low_diff_notGwas), nrow = 2,dimnames = list(Diff = c("High", "Low"),GWAS = c("yes", "no")))
  test <- fisher.test(contingency_table, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(BMI_enrich_genes) <- colnames(gene_degrees)[individuals]
which(BMI_enrich_genes < 0.05)
```



## CD
```{r}
d = 10
individuals <- c(c(3:121),c(123:241),c(243:361))
CD_enrich_genes <- sapply(individuals, FUN = function(i){
  high_diff_gwas <- length(which((gene_degrees[,i] >d) & (gene_degrees$GWAS_CD_gene == 1)))
  low_diff_gwas <- length(which((gene_degrees[,i] <=d) & (gene_degrees$GWAS_CD_gene == 1)))
  high_diff_notGwas <- length(which((gene_degrees[,i] >d) & (gene_degrees$GWAS_CD_gene == 0)))
  low_diff_notGwas <- length(which((gene_degrees[,i] <=d) & (gene_degrees$GWAS_CD_gene == 0)))
  contingency_table <- matrix(c(high_diff_gwas,low_diff_gwas,high_diff_notGwas,low_diff_notGwas), nrow = 2,dimnames = list(Diff = c("High", "Low"),GWAS = c("yes", "no")))
  test <- fisher.test(contingency_table, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(CD_enrich_genes) <- colnames(gene_degrees)[individuals]
which(CD_enrich_genes < 0.05)
```


## CAD
```{r}
d = 10
individuals <- c(c(3:121),c(123:241),c(243:361))
CAD_enrich_genes <- sapply(individuals, FUN = function(i){
  high_diff_gwas <- length(which((gene_degrees[,i] >d) & (gene_degrees$GWAS_CAD_gene == 1)))
  low_diff_gwas <- length(which((gene_degrees[,i] <=d) & (gene_degrees$GWAS_CAD_gene == 1)))
  high_diff_notGwas <- length(which((gene_degrees[,i] >d) & (gene_degrees$GWAS_CAD_gene == 0)))
  low_diff_notGwas <- length(which((gene_degrees[,i] <=d) & (gene_degrees$GWAS_CAD_gene == 0)))
  contingency_table <- matrix(c(high_diff_gwas,low_diff_gwas,high_diff_notGwas,low_diff_notGwas), nrow = 2,dimnames = list(Diff = c("High", "Low"),GWAS = c("yes", "no")))
  test <- fisher.test(contingency_table, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(CAD_enrich_genes) <- colnames(gene_degrees)[individuals]
which(CAD_enrich_genes < 0.05)
```



## RA
```{r}
d = 10
individuals <- c(c(3:121),c(123:241),c(243:361))
RA_enrich_genes <- sapply(individuals, FUN = function(i){
  high_diff_gwas <- length(which((gene_degrees[,i] >d) & (gene_degrees$GWAS_RA_gene == 1)))
  low_diff_gwas <- length(which((gene_degrees[,i] <=d) & (gene_degrees$GWAS_RA_gene == 1)))
  high_diff_notGwas <- length(which((gene_degrees[,i] >d) & (gene_degrees$GWAS_RA_gene == 0)))
  low_diff_notGwas <- length(which((gene_degrees[,i] <=d) & (gene_degrees$GWAS_RA_gene == 0)))
  contingency_table <- matrix(c(high_diff_gwas,low_diff_gwas,high_diff_notGwas,low_diff_notGwas), nrow = 2,dimnames = list(Diff = c("High", "Low"),GWAS = c("yes", "no")))
  test <- fisher.test(contingency_table, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(RA_enrich_genes) <- colnames(gene_degrees)[individuals]
which(RA_enrich_genes < 0.05)
```



# Further analysis of enrichment
So the interesting ones we are going to follow up on are these: CAD_enrich[which(CAD_enrich < 0.05)] which are the individuals whose high difference TFs are enriched in GWAS hits for CAD.

```{r}
cad_enriched_individuals <- CAD_enrich[which(CAD_enrich < 0.05)]
tfs_cad_enriched <- (tf_degrees[,c(1,362,363,364,365, which(colnames(tf_degrees) %in% names(cad_enriched_individuals)))])
tfs_cad_enriched[which(tfs_cad_enriched$GWAS_CAD_tf == 1),c(1,2,3,4,5,30)]
tfs_cad_enriched[which(tfs_cad_enriched$GWAS_CAD_tf == 1),c(1,2,3,4,5,30)]

tfs_cad_enriched[which(tfs_cad_enriched$GWAS_CAD_tf == 1),]
```

# Find CAD enriched individuals for which we have ATAC
```{r}
all_indiv <- as.vector(read.table("../data/individuals"))
atac_cm_indiv <- as.vector(read.table("../data/atac_indiv_nums"))
which(all_indiv %in% atac_cm_indiv$V1)
cad_enriched_individuals
```

# One example individual
```{r}
library(reshape2)
load("../outputs/great_egret_v3.0_banovich_PANDA_iPSC-CM_.RData")
net <- melt(regnet)
colnames(net) = c("tf","gene","score")
net$id <- paste0(net$tf,net$gene)
regnet_edge_table_iPSC_CM <- data.frame(net$id)
colnames(regnet_edge_table_iPSC_CM) <- c("id")
regnet_edge_table_iPSC_CM$tf <- as.vector(net$tf)
regnet_edge_table_iPSC_CM$gene <- as.vector(net$gene)
regnet_edge_table_iPSC_CM$panda <- as.vector(net$score)

load("../outputs/great_egret_v3.0_banovich_genotypes_iPSC-CM_112_egret28.RData")
net <- melt(regnet)
colnames(net) = c("tf","gene","score")
net$id <- paste0(net$tf,net$gene)
if(all.equal(as.vector(regnet_edge_table_iPSC_CM$id),as.vector(net$id))){
  regnet_edge_table_iPSC_CM$tempID <- as.vector(net$score)
  #print(colnames(regnet_edge_table))
  colnames(regnet_edge_table_iPSC_CM)[colnames(regnet_edge_table_iPSC_CM) == "tempID"] <-     paste0("iPSC-CM_indiv_112")
}
regnet_edge_table_iPSC_CM$diff <- abs(regnet_edge_table_iPSC_CM$`iPSC-CM_indiv_112`-regnet_edge_table_iPSC_CM$panda)
regnet_edge_table_iPSC_CM[which((regnet_edge_table_iPSC_CM$tf == "ATF3")&(regnet_edge_table_iPSC_CM$diff > 0.5)),]
```


```{r}
eqtl <- read.table("../inputs/iPSC-CM_eQTL_in_motif_promotor_adjacent_egene_banovich_v3.0_04202020.txt")
colnames(eqtl) <- c("tf",	"gene",	"eGene",	"snpPos",	"chr",	"pval",	"beta")
eqtl[which(eqtl$tf == "ATF3"),]
```


Overlap atac with promotor of genes - LCL
```{r}
library(GenomicRanges)
atac_lcl <- read.table("../atac/GSE108460_LCL-ATAC-counts.txt", header = TRUE)
atac_lcl <- separate(atac_lcl, window, c("chrom", "start", "stop"), "-")
columnNum <- which(colnames(atac_lcl) == "NA19128")
atac_lcl_112 <- atac_lcl[which(atac_lcl[,columnNum] > 0),c(1,2,3,columnNum)]
atac_lcl_112_gr <- GRanges(seqnames = atac_lcl_112$chrom, ranges = IRanges(start = as.numeric(as.vector(atac_lcl_112$start)), end = as.numeric(as.vector(atac_lcl_112$stop))), strand = NULL, mcols = atac_lcl_112[,4])

# read in the gene annotation file with gene ranges.
genes <- read.table("../annotation/UCSC_ensembl_hg19.txt", header = TRUE,sep = "\t")
genes$promotorLeft <- ifelse(genes$strand == "+", (genes$txStart + 1 - 750), (genes$txEnd-250))
genes$promotorRight <- ifelse(genes$strand == "+", (genes$txStart + 1 + 250), (genes$txEnd+750))
gr_genes_promotor <- GRanges(seqnames = genes$chrom, ranges = IRanges(start = genes$promotorLeft, end = genes$promotorRight), strand = NULL, mcols = genes[,13:15])

overlaps_lcl <- data.frame(findOverlaps(atac_lcl_112_gr, gr_genes_promotor))
overlaps_lcl$atac_lcl_count <- atac_lcl_112_gr$mcols[overlaps_lcl$queryHits]
overlaps_lcl$gene <- gr_genes_promotor$mcols.name2[overlaps_lcl$subjectHits]
```


Overlap atac with promotor of genes - ipsc
```{r}

atac_ipsc <- read.table("../atac/GSE108458_iPSC-ATAC-counts.txt", header = TRUE)
atac_ipsc <- separate(atac_ipsc, window, c("chrom", "start", "stop"), "-")
columnNum <- which(colnames(atac_ipsc) == "NA19128")
atac_ipsc_112 <- atac_ipsc[which(atac_ipsc[,columnNum] > 0),c(1,2,3,columnNum)]
atac_ipsc_112_gr <- GRanges(seqnames = atac_ipsc_112$chrom, ranges = IRanges(start = as.numeric(as.vector(atac_ipsc_112$start)), end = as.numeric(as.vector(atac_ipsc_112$stop))), strand = NULL, mcols = atac_ipsc_112[,4])

overlaps_ipsc <- data.frame(findOverlaps(atac_ipsc_112_gr, gr_genes_promotor))
overlaps_ipsc$atac_ipsc_count <- atac_ipsc_112_gr$mcols[overlaps_ipsc$queryHits]
overlaps_ipsc$gene <- gr_genes_promotor$mcols.name2[overlaps_ipsc$subjectHits]
```

Overlap atac with promotor of genes - ipsc-CM
```{r}
atac_ipsc_CM <- read.table("../atac/GSE108459_iPSC-CM-ATAC-counts.txt", header = TRUE)
atac_ipsc_CM <- separate(atac_ipsc_CM, window, c("chrom", "start", "stop"), "-")
columnNum <- which(colnames(atac_ipsc_CM) == "NA19128")
atac_ipsc_CM_112 <- atac_ipsc_CM[which(atac_ipsc_CM[,columnNum] > 0),c(1,2,3,columnNum)]
atac_ipsc_CM_112_gr <- GRanges(seqnames = atac_ipsc_CM_112$chrom, ranges = IRanges(start = as.numeric(as.vector(atac_ipsc_CM_112$start)), end = as.numeric(as.vector(atac_ipsc_CM_112$stop))), strand = NULL, mcols = atac_ipsc_CM_112[,4])

overlaps_ipsc_CM <- data.frame(findOverlaps(atac_ipsc_CM_112_gr, gr_genes_promotor))
overlaps_ipsc_CM$atac_ipsc_CM_count <- atac_ipsc_CM_112_gr$mcols[overlaps_ipsc_CM$queryHits]
overlaps_ipsc_CM$gene <- gr_genes_promotor$mcols.name2[overlaps_ipsc_CM$subjectHits]
```

```{r}
genesOfInterest <- c("ENSG00000013573","ENSG00000029363",	"ENSG00000176261")
g1 <- c("ENSG00000176261")
overlaps_ipsc_CM[which(overlaps_ipsc_CM$gene %in% g1),]
overlaps_ipsc[which(overlaps_ipsc$gene %in% g1),]
overlaps_lcl[which(overlaps_lcl$gene %in% g1),]

```

Make eQTL bed files
```{r}
lcl_eqtl <- read.table("../inputs/LCL_eQTL_in_motif_promotor_adjacent_egene_banovich_v3.0_03232020.txt")
colnames(lcl_eqtl) <- c("tf",	"gene",	"eGene",	"snpPos",	"chr",	"pval",	"beta")
lcl_eqtl$id <- paste0(lcl_eqtl$gene,"_",lcl_eqtl$tf)
write.table(lcl_eqtl[,c(5,4,4,8)], row.names = FALSE, col.names = FALSE, quote = FALSE,sep = "\t", file = "../viz/lcl_eqtl.bed")

ipsc_eqtl <- read.table("../inputs/iPSC_eQTL_in_motif_promotor_adjacent_egene_banovich_v3.0_03232020.txt")
colnames(ipsc_eqtl) <- c("tf",	"gene",	"eGene",	"snpPos",	"chr",	"pval",	"beta")
ipsc_eqtl$id <- paste0(ipsc_eqtl$gene,"_",ipsc_eqtl$tf)
write.table(ipsc_eqtl[,c(5,4,4,8)], row.names = FALSE, col.names = FALSE, quote = FALSE,sep = "\t", file = "../viz/ipsc_eqtl.bed")

ipsc_CM_eqtl <- read.table("../inputs/iPSC-CM_eQTL_in_motif_promotor_adjacent_egene_banovich_v3.0_04202020.txt")
colnames(ipsc_CM_eqtl) <- c("tf",	"gene",	"eGene",	"snpPos",	"chr",	"pval",	"beta")
ipsc_CM_eqtl$id <- paste0(ipsc_CM_eqtl$gene,"_",ipsc_CM_eqtl$tf)
write.table(ipsc_CM_eqtl[,c(5,4,4,8)], row.names = FALSE, col.names = FALSE, quote = FALSE,sep = "\t", file = "../viz/ipsc_CM_eqtl.bed")
```