---
title: "Final EGRET v1 Banovich application - Data Analysis"
author: "Deborah Weighill"
date: "July 5, 2020"
output:
  html_notebook:
    toc: true
    theme: cosmo
    df_print: paged
---

# Overview
Here we do some analysis on the banovich results for the "application". We start from the "master difference table" that we created.



# Load libraries
```{r}
library(dplyr)
library(ggplot2)
library(ggpubr)
library(reshape2)
library(jcolors)
library(gridExtra)
library(grid)
library(gridExtra)
```

# Load difference tables
```{r}
load("../outputs/tf_diff_degree_table_banovich_smart1_07042020.RData")
load("../outputs/gene_diff_degree_table_banovich_smart1_7042020.RData")

```

# set color pallet
```{r}
#pallet2 <- c("#42858C", "#E48F1B","#570D32")
#pallet2 <- c("#EB8F1E","#AD1F2F","#2592A7")
#pallet2 <- c("#0BB19F", "#FF4500","#570D32")    
#pallet2 <- c("#0BB19F", "#FF4500","#50D32")  
pallet2 <- c(    "#FFBF00",  "#870E75"  ,  "#0BB19F" )
```


We can now plot distributions of tf degree and gene degree in difference network for each tissue.
```{r}
colnames(tf_degrees)[1] <- "tf"
tf_degrees_scaled <- scale(tf_degrees[,c(2:361)], center = TRUE, scale = TRUE)
tf_degrees_scaled <- cbind(tf_degrees[,c(1)], tf_degrees_scaled, tf_degrees[,c(362:367)])
colnames(tf_degrees_scaled)[1] <-"tf"
tf_degree_lcl <- melt(tf_degrees[,c(1,c(3:121))], id.vars = c("tf"))
tf_degree_lcl$tissue <- "LCL"
tf_degree_iPSC <- melt(tf_degrees[,c(1,c(123:241))], id.vars = c("tf"))
tf_degree_iPSC$tissue <- "iPSC"
tf_degree_iPSC_CM <- melt(tf_degrees[,c(1,c(243:361))], id.vars = c("tf"))
tf_degree_iPSC_CM$tissue <- "iPSC-CM"
tf_degrees_melted <- rbind(tf_degree_lcl,tf_degree_iPSC,tf_degree_iPSC_CM)
colnames(tf_degrees_melted) <- c("tf","individual","diff_out_degree","Tissue")

pdf("banovich_tf_disruption_boxplot.pdf",width = 4, height = 3.5)
p <- ggplot(tf_degrees_melted, aes(x=Tissue, y=diff_out_degree, color=Tissue)) + geom_boxplot() + theme_bw() + scale_color_manual(values=pallet2) + labs(title="Distribution of TF Disruption Scores", x ="Cell type", y = "TF Disruption Score") + theme(legend.position = "none")
plot(p)
dev.off()


ggplot(tf_degrees_melted[which(tf_degrees_melted$Tissue == "iPSC"),], aes(x=Tissue, y=diff_out_degree, color=Tissue)) + geom_violin() + theme_linedraw() + scale_color_manual(values=c("red")) + labs(title="Distribution of TF Disruption Scores", x = NULL, y = "TF Disruption Score") + theme(legend.position = "right")

ggplot(tf_degrees_melted[which(tf_degrees_melted$Tissue == "iPSC-CM"),], aes(x=Tissue, y=diff_out_degree, color=Tissue)) + geom_violin() + theme_linedraw() + scale_color_manual(values=c("blue")) + labs(title="Distribution of TF Disruption Scores", x =NULL, y = "TF Disruption Score") + theme(legend.position = "right")

ggplot(tf_degrees_melted[which(tf_degrees_melted$Tissue == "LCL"),], aes(x=Tissue, y=diff_out_degree, color=Tissue)) + geom_violin() + theme_linedraw() + scale_color_manual(values=c("orange")) + labs(title="Distribution of TF Disruption Scores", x =NULL, y = "TF Disruption Score") + theme(legend.position = "right")

ggplot(tf_degrees_melted[which(tf_degrees_melted$individual == "iPSC-CM_indiv_21"),], aes(x=Tissue, y=diff_out_degree, color=Tissue)) + geom_violin() + theme_linedraw() + scale_color_manual(values=c("blue")) + labs(title="Distribution of TF Disruption Scores", x =NULL, y = "TF Disruption Score") + theme(legend.position = "right")

```

scaled plots
```{r}

tf_degree_lcl <- melt(tf_degrees_scaled[,c(1,c(3:121))], id.vars = c("tf"))
tf_degree_lcl$tissue <- "LCL"
tf_degree_iPSC <- melt(tf_degrees_scaled[,c(1,c(123:241))], id.vars = c("tf"))
tf_degree_iPSC$tissue <- "iPSC"
tf_degree_iPSC_CM <- melt(tf_degrees_scaled[,c(1,c(243:361))], id.vars = c("tf"))
tf_degree_iPSC_CM$tissue <- "iPSC-CM"
tf_degrees_melted <- rbind(tf_degree_lcl,tf_degree_iPSC,tf_degree_iPSC_CM)
colnames(tf_degrees_melted) <- c("tf","individual","diff_out_degree","Tissue")

pdf("banovich_scaled_tf_disruption_boxplot.pdf",width = 4, height = 3.5)
p <- ggplot(tf_degrees_melted, aes(x=Tissue, y=diff_out_degree, color=Tissue)) + geom_boxplot() + theme_bw() + scale_color_manual(values=pallet2) + labs(title="Distribution of scaled TF disruption scores", x ="Cell type", y = "Scaled TF disruption score") + theme(legend.position="none")
plot(p)
dev.off()

pdf("banovich_scaled_tf_disruption_violin.pdf",width = 4, height = 3.5)
ggplot(tf_degrees_melted, aes(x=Tissue, y=diff_out_degree, fill=Tissue, col = Tissue)) + geom_violin() + theme_bw() + scale_fill_manual(values=pallet2) + scale_color_manual(values=pallet2) + labs(title="Distribution of TF Disruption Scores", x ="Cell type", y = "TF Disruption Score") + theme(legend.position="none")
dev.off()

ggplot(tf_degrees_melted, aes(x=Tissue, y=diff_out_degree, color=Tissue,)) + geom_violin() + theme_bw() + scale_color_manual(values=pallet2) + labs(title="Distribution of scaled TF disruption scores", x ="Cell type", y = "Scaled TF disruption score") +  ylim(-2, 2) + theme(legend.position="none")

p1 <- ggplot(tf_degrees_melted, aes(x=diff_out_degree,  color=Tissue)) + geom_density() + theme_bw() + scale_color_manual(values=pallet2) +labs(title=NULL, x ="Scaled TF disruption score", y = "Frequency") 

p2 <- ggplot(tf_degrees_melted, aes(x=diff_out_degree,  color=Tissue)) + geom_density() + theme_bw() + scale_color_manual(values=pallet2) +labs(title=NULL, x ="Scaled TF disruption score", y = "Frequency") +  xlim(-1, 1)

p3 <- grid.arrange(p1 + theme(legend.position="hidden"), p2 + theme(legend.position="right"), nrow=1, widths = c(3,4))
p4 <- annotate_figure(p3,top = text_grob("Distribution of scaled TF disruption scores", color = "black", face = "bold", size = 14))
pdf("disruption_score_dist.pdf",width = 7, height = 3)
plot(p4)
dev.off()


```

```{r}
gene_degrees_scaled <- scale(gene_degrees[,c(2:361)], center = TRUE, scale = TRUE)
gene_degrees_scaled <- cbind(gene_degrees[,c(1)], gene_degrees_scaled, gene_degrees[,c(362:367)])
colnames(gene_degrees_scaled)[1] <- "gene"
gene_degree_lcl <- melt(gene_degrees_scaled[,c(1,c(3,121))], id.vars = c("gene"))
gene_degree_lcl$tissue <- "LCL"
gene_degree_iPSC <- melt(gene_degrees_scaled[,c(1,c(123,241))], id.vars = c("gene"))
gene_degree_iPSC$tissue <- "iPSC"
gene_degree_iPSC_CM <- melt(gene_degrees_scaled[,c(1,c(243,361))], id.vars = c("gene"))
gene_degree_iPSC_CM$tissue <- "iPSC-CM"
gene_degrees_melted <- rbind(gene_degree_lcl,gene_degree_iPSC,gene_degree_iPSC_CM)
colnames(gene_degrees_melted) <- c("gene","individual","diff_in_degree","tissue")
ggplot(gene_degrees_melted, aes(x=tissue, y=diff_in_degree, fill=tissue, col = tissue)) + geom_boxplot() + theme_linedraw() + scale_fill_manual(values=pallet2) + scale_color_manual(values=pallet2) + labs(title="Distribution of Gene Disruption Scores", x ="Cell type", y = "Gene Disruption Score")

```






# Enrichment tests on TFs
Let's try and see, for a given individual, are "high difference" TFs enriched in GWAS hits.

## CD
```{r}
d = 20
individuals <- c(c(3:121),c(123:241),c(243:361))
CD_enrich <- sapply(individuals, FUN = function(i){
  high_diff_gwas <- length(which((tf_degrees[,i] >d) & (tf_degrees$GWAS_CD_tf == 1)))
  low_diff_gwas <- length(which((tf_degrees[,i] <=d) & (tf_degrees$GWAS_CD_tf == 1)))
  high_diff_notGwas <- length(which((tf_degrees[,i] >d) & (tf_degrees$GWAS_CD_tf == 0)))
  low_diff_notGwas <- length(which((tf_degrees[,i] <=d) & (tf_degrees$GWAS_CD_tf == 0)))
  contingency_table <- matrix(c(high_diff_gwas,low_diff_gwas,high_diff_notGwas,low_diff_notGwas), nrow = 2,dimnames = list(Diff = c("High", "Low"),GWAS = c("yes", "no")))
  test <- fisher.test(contingency_table, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(CD_enrich) <- colnames(tf_degrees)[individuals]
cd_enriched_indiv <- names(which(CD_enrich < 0.05))
cd_enriched_indiv

cd_df <- as.data.frame(CD_enrich)
colnames(cd_df) <- "pvalue"
cd_df$Tissue <- NA
cd_df$Tissue[1:119] <- "LCL"
cd_df$Tissue[120:238] <- "iPSC"
cd_df$Tissue[239:357] <- "iPSC-CM"
cd_df$Individual <- NA
cd_df$Individual <- c(c(1:119),c(1:119),c(1:119))
#pallet2 <- c("#FFA733","#2CAFC9","#AD5C99")


#pallet2 <- c("#42858C", "#E48F1B","#570D32")
#pallet2 <- c("#EB8F1E","#AD1F2F","#2592A7")
#pallet2 <- c("#0BB19F", "#FF4500","#570D32")    

pdf("banovich_CD_enrichment.pdf",width = 5, height = 3.5)
ggplot(cd_df) + geom_point(shape = 19, aes(x = reorder(Individual, log10(pvalue)), y = -log10(pvalue), col = Tissue),) + theme_classic()+ scale_color_manual(values=pallet2) + geom_abline(slope = 0, intercept = -log10(0.05), linetype = "dashed", col = "black") + xlab("Individual") + theme( axis.text.x=element_blank(), axis.ticks.x=element_blank()) + labs(y = "-log10(p-value)", title = "Enrichment of disrupted TFs (d > 15) in CD genes",color = "Cell type") + annotate("text", x = 110, y = 1.4, label = "p = 0.05")
dev.off()
```
```{r}
tf_degrees[,c("tf",cd_enriched_indiv)]
tf_degrees[which(tf_degrees$GWAS_CD_tf == 1),c("tf","GWAS_CD_tf",cd_enriched_indiv)]
cd_tf_table <- diff_table[(which(diff_table$tf == c("ELF1","FOXP1,FOXP2"))),c("tf","gene",cd_enriched_indiv)]
cd_tf_table[apply(cd_tf_table[,cd_enriched_indiv], 1, function(x) any(x > 0.5)), ]
```
the first gene ENSG00000110768 is maybe differentially expressed in chrons disease
"Pathogenicity of In Vivo Generated Intestinal Th17 Lymphocytes is IFNÎ³ Dependent"

maybe look here for the second gene (ENSG00000167748) 
Genetically-modified stem cells in treatment of human diseases: Tissue kallikrein (KLK1)-based targeted therapy (Review)

## CD
```{r}
d = 2
individuals <- c(c(3:121),c(123:241),c(243:361))
CD_enrich <- sapply(individuals, FUN = function(i){
  high_diff_gwas <- length(which((tf_degrees_scaled[,i] >d) & (tf_degrees_scaled$GWAS_CD_tf == 1)))
  low_diff_gwas <- length(which((tf_degrees_scaled[,i] <=d) & (tf_degrees_scaled$GWAS_CD_tf == 1)))
  high_diff_notGwas <- length(which((tf_degrees_scaled[,i] >d) & (tf_degrees_scaled$GWAS_CD_tf == 0)))
  low_diff_notGwas <- length(which((tf_degrees_scaled[,i] <=d) & (tf_degrees_scaled$GWAS_CD_tf == 0)))
  contingency_table <- matrix(c(high_diff_gwas,low_diff_gwas,high_diff_notGwas,low_diff_notGwas), nrow = 2,dimnames = list(Diff = c("High", "Low"),GWAS = c("yes", "no")))
  test <- fisher.test(contingency_table, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(CD_enrich) <- colnames(tf_degrees_scaled)[individuals]
cd_enriched_indiv <- names(which(CD_enrich < 0.05))
cd_enriched_indiv

cd_df <- as.data.frame(CD_enrich)
colnames(cd_df) <- "pvalue"
cd_df$Tissue <- NA
cd_df$Tissue[1:119] <- "LCL"
cd_df$Tissue[120:238] <- "iPSC"
cd_df$Tissue[239:357] <- "iPSC-CM"
cd_df$Individual <- NA
cd_df$Individual <- c(c(1:119),c(1:119),c(1:119))

pdf("banovich_CD_enrichment_scaled.pdf",width = 5, height = 3.5)
ggplot(cd_df) + geom_point(shape = 19, aes(x = reorder(Individual, log10(pvalue)), y = -log10(pvalue), col = Tissue),) + theme_classic()+ scale_color_manual(values=pallet2) + geom_abline(slope = 0, intercept = -log10(0.05), linetype = "dashed", col = "black") + xlab("Individual") + theme( axis.text.x=element_blank(), axis.ticks.x=element_blank()) + labs(y = "-log10(p-value)", title = "Enrichment of disrupted TFs (d' > 2) in CD genes",color = "Cell type") + annotate("text", x = 110, y = 1.4, label = "p = 0.05")
dev.off()
```

```{r}
cd_enriched_indiv
tf_degrees_scaled[which(tf_degrees_scaled$GWAS_CD_tf == 1),c("tf","GWAS_CD_tf",cd_enriched_indiv)]

```


## CAD
```{r}
d = 15
individuals <- c(c(3:121),c(123:241),c(243:361))
CAD_enrich <- sapply(individuals, FUN = function(i){
  high_diff_gwas <- length(which((tf_degrees[,i] >d) & (tf_degrees$GWAS_CAD_tf == 1)))
  low_diff_gwas <- length(which((tf_degrees[,i] <=d) & (tf_degrees$GWAS_CAD_tf == 1)))
  high_diff_notGwas <- length(which((tf_degrees[,i] >d) & (tf_degrees$GWAS_CAD_tf == 0)))
  low_diff_notGwas <- length(which((tf_degrees[,i] <=d) & (tf_degrees$GWAS_CAD_tf == 0)))
  contingency_table <- matrix(c(high_diff_gwas,low_diff_gwas,high_diff_notGwas,low_diff_notGwas), nrow = 2,dimnames = list(Diff = c("High", "Low"),GWAS = c("yes", "no")))
  test <- fisher.test(contingency_table, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(CAD_enrich) <- colnames(tf_degrees)[individuals]

cad_enriched_indiv <- names(which(CAD_enrich < 0.05))
cad_enriched_indiv
cad_df <- as.data.frame(CAD_enrich)
colnames(cad_df) <- "pvalue"
cad_df$Tissue <- NA
cad_df$Tissue[1:119] <- "LCL"
cad_df$Tissue[120:238] <- "iPSC"
cad_df$Tissue[239:357] <- "iPSC-CM"
cad_df$Individual <- NA
cad_df$Individual <- c(c(1:119),c(1:119),c(1:119))

pdf("banovich_CAD_enrichment.pdf",width = 5, height = 3.5)
ggplot(cad_df) + geom_point(shape = 19, aes(x = reorder(Individual, log10(pvalue)), y = -log10(pvalue), col = Tissue),) + theme_classic()+ scale_color_manual(values=pallet2) + geom_abline(slope = 0, intercept = -log10(0.05), linetype = "dashed", col = "black") + xlab("Individual") + theme( axis.text.x=element_blank(), axis.ticks.x=element_blank())  + labs(y = "-log10(p-value)", title = "Enrichment of disrupted TFs (d > 15) in CAD genes",color = "Cell type") + annotate("text", x = 110, y = 1.4, label = "p = 0.05")
dev.off()
```

## CAD
```{r}
d = 2
individuals <- c(c(3:121),c(123:241),c(243:361))
CAD_enrich <- sapply(individuals, FUN = function(i){
  high_diff_gwas <- length(which((tf_degrees_scaled[,i] >d) & (tf_degrees_scaled$GWAS_CAD_tf == 1)))
  low_diff_gwas <- length(which((tf_degrees_scaled[,i] <=d) & (tf_degrees_scaled$GWAS_CAD_tf == 1)))
  high_diff_notGwas <- length(which((tf_degrees_scaled[,i] >d) & (tf_degrees_scaled$GWAS_CAD_tf == 0)))
  low_diff_notGwas <- length(which((tf_degrees_scaled[,i] <=d) & (tf_degrees_scaled$GWAS_CAD_tf == 0)))
  contingency_table <- matrix(c(high_diff_gwas,low_diff_gwas,high_diff_notGwas,low_diff_notGwas), nrow = 2,dimnames = list(Diff = c("High", "Low"),GWAS = c("yes", "no")))
  test <- fisher.test(contingency_table, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(CAD_enrich) <- colnames(tf_degrees_scaled)[individuals]

cad_enriched_indiv <- names(which(CAD_enrich < 0.05))
cad_enriched_indiv
cad_df <- as.data.frame(CAD_enrich)
colnames(cad_df) <- "pvalue"
cad_df$Tissue <- NA
cad_df$Tissue[1:119] <- "LCL"
cad_df$Tissue[120:238] <- "iPSC"
cad_df$Tissue[239:357] <- "iPSC-CM"
cad_df$Individual <- NA
cad_df$Individual <- c(c(1:119),c(1:119),c(1:119))

pdf("banovich_CAD_enrichment_scaled.pdf",width = 5, height = 3.5)
ggplot(cad_df) + geom_point(shape = 19, aes(x = reorder(Individual, log10(pvalue)), y = -log10(pvalue), col = Tissue),) + theme_classic()+ scale_color_manual(values=pallet2) + geom_abline(slope = 0, intercept = -log10(0.05), linetype = "dashed", col = "black") + xlab("Individual") + theme( axis.text.x=element_blank(), axis.ticks.x=element_blank()) + labs(y = "-log10(p-value)", title = "Enrichment of disrupted TFs (d' > 2) in CAD genes",color = "Cell type") + annotate("text", x = 110, y = 1.4, label = "p = 0.05")
dev.off()
```






# Some t tests and box plots

```{r}
library(ggpubr)
library(dplyr)

tf_degree_lcl <- melt(tf_degrees_scaled[,c(1,c(3:121))], id.vars = c("tf"))
tf_degree_lcl$tissue <- "LCL"
tf_degree_iPSC <- melt(tf_degrees_scaled[,c(1,c(123:241))], id.vars = c("tf"))
tf_degree_iPSC$tissue <- "iPSC"
tf_degree_iPSC_CM <- melt(tf_degrees_scaled[,c(1,c(243:361))], id.vars = c("tf"))
tf_degree_iPSC_CM$tissue <- "iPSC-CM"
tf_degrees_melted <- rbind(tf_degree_lcl,tf_degree_iPSC,tf_degree_iPSC_CM)
colnames(tf_degrees_melted) <- c("tf","individual","diff_out_degree","tissue")
ggplot(tf_degrees_melted, aes(x=tissue, y=diff_out_degree, color=tissue)) + geom_violin()

```

```{r}
colnames(tf_degree_iPSC_CM) <-  c("tf","individual","diff_out_degree","tissue")
tf_degree_iPSC_CM$CAD <- tf_degrees_scaled$GWAS_CAD_tf[match(tf_degree_iPSC_CM$tf, tf_degrees_scaled$tf)]

test <- t.test(tf_degree_iPSC_CM[which(tf_degree_iPSC_CM$CAD == 1),]$diff_out_degree,tf_degree_iPSC_CM[which(tf_degree_iPSC_CM$CAD == 0),]$diff_out_degree, alternative="greater")

mean0 <- mean(tf_degree_iPSC_CM[which(tf_degree_iPSC_CM$CAD == 0),]$diff_out_degree)
mean1 <- mean(tf_degree_iPSC_CM[which(tf_degree_iPSC_CM$CAD == 1),]$diff_out_degree)

p1 <- ggplot(tf_degree_iPSC_CM, aes(x = as.factor(CAD), y = diff_out_degree, group = as.factor(CAD), fill = as.factor(CAD))) + scale_fill_manual(values = pallet2) + geom_violin() + theme_bw() + geom_point(x = as.factor(0), y = mean0, shape=18,size=5, color="#E41A1C")+ geom_point(x = as.factor(1), y = mean1, shape=18,size=5, color="#E41A1C") + labs(title = "Distribution of disruption scores for CAD and non-CAD TFs in iPSC-CMs",y = "TF disruption score", x = "CAD TF?") + theme(legend.position = "none")

p2 <- ggplot(tf_degree_iPSC_CM, aes(x = as.factor(CAD), y = diff_out_degree, group = as.factor(CAD), fill = as.factor(CAD))) + scale_fill_manual(values = pallet2) + geom_violin() + theme_bw() + geom_point(x = as.factor(0), y = mean0, shape=18,size=5, color="#E41A1C")+ geom_point(x = as.factor(1), y = mean1, shape=18,size=5, color="#E41A1C")+ ylim(-1,1) + labs(title = "Distribution of disruption scores for CAD and non-CAD TFs in iPSC-CMs",y = "TF disruption score", x = "CAD TF?") + theme(legend.position = "none")

p3 <- grid.arrange(p1 + theme(legend.position="none"), p2 + theme(legend.position="none"), nrow=1, widths = c(3,3))
pdf("cad_1_0_ipsc_cm.pdf",width = 6, height = 3)
plot(p3)
dev.off()

ggplot(tf_degree_iPSC_CM, aes(x = as.factor(CAD), y = diff_out_degree,group = as.factor(CAD))) + geom_jitter(position=position_jitter(0.2), size = 0.1) + theme_bw() + ylim(-1,1) + geom_point(aes(x = as.factor(0), y = mean0), shape=18,size=5, color="#E41A1C")+ geom_point(x = as.factor(1), y = mean1, shape=18,size=5, color="#E41A1C")


colnames(tf_degree_iPSC_CM) <-  c("tf","individual","diff_out_degree","tissue")
tf_degree_iPSC_CM$CD <- tf_degrees_scaled$GWAS_CD_tf[match(tf_degree_iPSC_CM$tf, tf_degrees_scaled$tf)]

t.test(tf_degree_iPSC_CM[which(tf_degree_iPSC_CM$CD == 1),]$diff_out_degree,tf_degree_iPSC_CM[which(tf_degree_iPSC_CM$CD == 0),]$diff_out_degree, alternative="greater")




colnames(tf_degree_lcl) <-  c("tf","individual","diff_out_degree","tissue")
tf_degree_lcl$CD <- tf_degrees_scaled$GWAS_CD_tf[match(tf_degree_lcl$tf, tf_degrees_scaled$tf)]

t.test(tf_degree_lcl[which(tf_degree_lcl$CD == 1),]$diff_out_degree,tf_degree_lcl[which(tf_degree_lcl$CD == 0),]$diff_out_degree, alternative="greater")


colnames(tf_degree_lcl) <-  c("tf","individual","diff_out_degree","tissue")
tf_degree_lcl$CAD <- tf_degrees_scaled$GWAS_CAD_tf[match(tf_degree_lcl$tf, tf_degrees_scaled$tf)]
t.test(tf_degree_lcl[which(tf_degree_lcl$CAD == 1),]$diff_out_degree,tf_degree_lcl[which(tf_degree_lcl$CAD == 0),]$diff_out_degree, alternative="greater")


```
# Testing for pallet design

```{r}
d = 2
individuals <- c(c(3:121),c(123:241),c(243:361))
CD_enrich <- sapply(individuals, FUN = function(i){
  high_diff_gwas <- length(which((tf_degrees_scaled[,i] >d) & (tf_degrees_scaled$GWAS_CD_tf == 1)))
  low_diff_gwas <- length(which((tf_degrees_scaled[,i] <=d) & (tf_degrees_scaled$GWAS_CD_tf == 1)))
  high_diff_notGwas <- length(which((tf_degrees_scaled[,i] >d) & (tf_degrees_scaled$GWAS_CD_tf == 0)))
  low_diff_notGwas <- length(which((tf_degrees_scaled[,i] <=d) & (tf_degrees_scaled$GWAS_CD_tf == 0)))
  contingency_table <- matrix(c(high_diff_gwas,low_diff_gwas,high_diff_notGwas,low_diff_notGwas), nrow = 2,dimnames = list(Diff = c("High", "Low"),GWAS = c("yes", "no")))
  test <- fisher.test(contingency_table, alternative = "greater")
  return(as.numeric(test$p.value))
})
names(CD_enrich) <- colnames(tf_degrees_scaled)[individuals]
cd_enriched_indiv <- names(which(CD_enrich < 0.05))
cd_enriched_indiv
cd_df <- as.data.frame(CD_enrich)
colnames(cd_df) <- "pvalue"
cd_df$Tissue <- NA
cd_df$Tissue[1:119] <- "LCL"
cd_df$Tissue[120:238] <- "iPSC"
cd_df$Tissue[239:357] <- "iPSC-CM"
cd_df$Individual <- NA
cd_df$Individual <- c(c(1:119),c(1:119),c(1:119))

 # blue_yonder   arctic_lime    mardi_gras safety_orange  tiffany_blue 
 #   "#3E71A8"     "#CEFF1A"     "#870E75"     "#FE6900"     "#0BB19F" 
#   kelly_green rich_electric_blue        maximum_red     majorelle_blue 
 #        "#29BF12"          "#00A5CF"          "#DE1A1A"          "#574AE2" 
#fluorescent_orange 
     #    "#FFBF00" 
    
pallet2 <- c(    "#FFBF00",  "#870E75"  ,  "#0BB19F" )

ggplot(cd_df) + geom_point(shape = 19, aes(x = reorder(Individual, log10(pvalue)), y = -log10(pvalue), col = Tissue),) + theme_classic()+ scale_color_manual(values=pallet2) + geom_abline(slope = 0, intercept = -log10(0.05), linetype = "dashed", col = "black") + xlab("Individual") + theme( axis.text.x=element_blank(), axis.ticks.x=element_blank()) + labs(y = "-log10(p-value)", title = "Enrichment of disrupted TFs (d' > 2) in CD genes",color = "Cell type") + annotate("text", x = 110, y = 1.4, label = "p = 0.05")
```



