---
title: "Final EGRET v1 Banovich application - Data Analysis"
author: "Deborah Weighill"
date: "June 29, 2020"
output:
  html_notebook:
    toc: true
    theme: cosmo
    df_print: paged
---

# Overview
Here we do some analysis on the banovich results for the "application".

# Make Panda/EGRET edge table

Load all of the PANDA/EGRET regnets and merge into a single table.


## Load the LCL results
```{r echo=FALSE,results='hide'}
library(reshape2)
# panda"
load("../outputs/finalEgret_v1_banovich_LCL_06242020_1_panda.RData")
net <- melt(regnet)
colnames(net) = c("tf","gene","score")
net$id <- paste0(net$tf,net$gene)

regnet_edge_table_LCL <- data.frame(net$id)
colnames(regnet_edge_table_LCL) <- c("id")
regnet_edge_table_LCL$tf <- as.vector(net$tf)
regnet_edge_table_LCL$gene <- as.vector(net$gene)
regnet_edge_table_LCL$panda <- as.vector(net$score)

for (g in c(1:119)) {
  filename <- paste0("../outputs/finalEgret_v1_banovich_LCL_06242020_",g,"_egret.RData")
  load(filename)
  net <- melt(regnet)
  colnames(net) = c("tf","gene","score")
  net$id <- paste0(net$tf,net$gene)
  if(all.equal(as.vector(regnet_edge_table_LCL$id),as.vector(net$id))){
    print(g)
    regnet_edge_table_LCL$tempID <- as.vector(net$score)
    #print(colnames(regnet_edge_table))
    colnames(regnet_edge_table_LCL)[colnames(regnet_edge_table_LCL) == "tempID"] <- paste0("LCL_indiv_",g)
  }
}
save(regnet_edge_table_LCL, file = "../outputs/LCL_table_banovich_finalEgret_v1_06292020.RData")
```


## Load the iPSC results
```{r echo=FALSE,results='hide'}
load("../outputs/finalEgret_v1_banovich_iPSC_06242020_1_panda.RData")
net <- melt(regnet)
colnames(net) = c("tf","gene","score")
net$id <- paste0(net$tf,net$gene)

regnet_edge_table_iPSC <- data.frame(net$id)
colnames(regnet_edge_table_iPSC) <- c("id")
regnet_edge_table_iPSC$tf <- as.vector(net$tf)
regnet_edge_table_iPSC$gene <- as.vector(net$gene)
regnet_edge_table_iPSC$panda <- as.vector(net$score)
for (g in c(1:119)) {
  filename <- paste0("../outputs/finalEgret_v1_banovich_iPSC_06242020_",g,"_egret.RData")
  load(filename)
  net <- melt(regnet)
  colnames(net) = c("tf","gene","score")
  net$id <- paste0(net$tf,net$gene)
  if(all.equal(as.vector(regnet_edge_table_iPSC$id),as.vector(net$id))){
    print(g)
    regnet_edge_table_iPSC$tempID <- as.vector(net$score)
    #print(colnames(regnet_edge_table))
    colnames(regnet_edge_table_iPSC)[colnames(regnet_edge_table_iPSC) == "tempID"] <- paste0("iPSC_indiv_",g)
  }
}
save(regnet_edge_table_iPSC, file = "../outputs/iPSC_table_banovich_finalEgret_v1_06292020.RData")
```

## Load the iPSC-CM results
```{r echo=FALSE,results='hide'}
load("../outputs/finalEgret_v1_banovich_iPSC-CM_06242020_1_panda.RData")
net <- melt(regnet)
colnames(net) = c("tf","gene","score")
net$id <- paste0(net$tf,net$gene)

regnet_edge_table_iPSC_CM <- data.frame(net$id)
colnames(regnet_edge_table_iPSC_CM) <- c("id")
regnet_edge_table_iPSC_CM$tf <- as.vector(net$tf)
regnet_edge_table_iPSC_CM$gene <- as.vector(net$gene)
regnet_edge_table_iPSC_CM$panda <- as.vector(net$score)

for (g in c(1:119)) {
  filename <- paste0("../outputs/finalEgret_v1_banovich_iPSC-CM_06242020_",g,"_egret.RData")
  load(filename)
  net <- melt(regnet)
  colnames(net) = c("tf","gene","score")
  net$id <- paste0(net$tf,net$gene)
  if(all.equal(as.vector(regnet_edge_table_iPSC_CM$id),as.vector(net$id))){
    print(g)
    regnet_edge_table_iPSC_CM$tempID <- as.vector(net$score)
    #print(colnames(regnet_edge_table))
    colnames(regnet_edge_table_iPSC_CM)[colnames(regnet_edge_table_iPSC_CM) == "tempID"] <- paste0("iPSC-CM_indiv_",g)
  }
}
save(regnet_edge_table_iPSC_CM, file = "../outputs/iPSC-CM_table_banovich_finalEgret_v1_06292020.RData")
```


## Merge into a single table with an inner join
This ensures that we are considering the same set of edges for the two tissues (LCL and iPSC).

```{r}
load("../outputs/LCL_table_banovich_finalEgret_v1_06292020.RData")
load("../outputs/iPSC_table_banovich_finalEgret_v1_06292020.RData")
load("../outputs/iPSC-CM_table_banovich_finalEgret_v1_06292020.RData")

pre_merged_regnet_table <- merge(regnet_edge_table_LCL,regnet_edge_table_iPSC, by.x = c(1,2,3), by.y = c(1,2,3), all = FALSE)
merged_regnet_table <- merge(pre_merged_regnet_table,regnet_edge_table_iPSC_CM,by.x = c(1,2,3), by.y = c(1,2,3), all = FALSE)
colnames(merged_regnet_table)[4] <- "LCL_panda"
colnames(merged_regnet_table)[124] <- "iPSC_panda"
colnames(merged_regnet_table)[244] <- "iPSC-CM_panda"
save(merged_regnet_table, file = "../outputs/merged_regnet_table_06302020.RData")
```

# Analysis

## Edge differences from PANDA
This section will answer the question: *"Which Tf-gene edges are impacted by variants for each individual and each tissue?"* We first construct the difference table by substracting the PANDA edge weight from the EGRET edge weight and taking the absolute value of the difference. 

```{r}
diff_table <- merged_regnet_table
for (i in c(1:119)){
  diff_table[,4+i] <- abs(merged_regnet_table[,4+i]-merged_regnet_table[,4])
  diff_table[,124+i] <- abs(merged_regnet_table[,124+i]-merged_regnet_table[,124])
  diff_table[,244+i] <- abs(merged_regnet_table[,244+i]-merged_regnet_table[,244])
}

save(diff_table, file = "../outputs/diffTable_merged_LCL_iPSC_iPSC-CM_banovich_edge_table_06302020.RData")
```

We can then plot the distribution of these differences per individual, per tissue. Plotting all individuals takes too long.

```{r}
library(ggplot2)
library(reshape2)
# LCL
diff_melted_LCL <- melt(diff_table[,c(1,c(5:123))], id.vars = c("id"))
colnames(diff_melted_LCL) <- c("id","individual","diff_edgeweight")
diff_melted_LCL$tissue <- "LCL"

#iPSC
diff_melted_iPSC <- melt(diff_table[,c(1,c(125:243))], id.vars = c("id"))
colnames(diff_melted_iPSC) <- c("id","individual","diff_edgeweight")
diff_melted_iPSC$tissue <- "iPSC"

#iPSC-CM
diff_melted_iPSC_CM <- melt(diff_table[,c(1,c(245:363))], id.vars = c("id"))
colnames(diff_melted_iPSC_CM) <- c("id","individual","diff_edgeweight")
diff_melted_iPSC_CM$tissue <- "iPSC-CM"

# combine
diff_melted_combined <- rbind(diff_melted_LCL,diff_melted_iPSC, diff_melted_iPSC_CM)

#ggplot(diff_melted_combined, aes(x=tissue, y=diff_edgeweight, color=tissue)) + geom_violin()

#ggplot(diff_melted_combined[diff_melted_combined$diff_edgeweight>0.1,], aes(x=tissue, y=diff_edgeweight, color=tissue)) + geom_violin()

ggplot(diff_melted_combined[(diff_melted_combined$individual == "LCL_indiv_1") | (diff_melted_combined$individual == "iPSC_indiv_1"),], aes(x=tissue, y=diff_edgeweight, color=tissue)) + geom_violin()

ggplot(diff_melted_combined[(diff_melted_combined$diff_edgeweight > 0.5) & ((diff_melted_combined$individual == "LCL_indiv_1") | (diff_melted_combined$individual == "iPSC_indiv_1")| (diff_melted_combined$individual == "iPSC-CM_indiv_1")),], aes(x=tissue, y=diff_edgeweight, color=tissue)) + geom_violin()
```



## TF/gene degree differences from PANDA
This section will answer the question: *"Which TFs overall are impacted the most for each individual and each tissue?"* This will be determined by calculating the **out-degree** of each TF in each individual **difference network**. In other words, for each individual EGRET network, we take *abs(Eij-Pij)* for each edge, and then we sum the "*difference edges"* around each tf, per individual. Similarly, we will dermine the **in-degree** of each gene in the difference network.

```{r}
tf_degrees <- aggregate(diff_table[,c(4:363)], by = list(diff_table$tf), FUN = sum) 
gene_degrees <- aggregate(diff_table[,c(4:363)], by = list(diff_table$gene), FUN = sum) 
head(tf_degrees)
head(gene_degrees)
```

We can now plot distributions of tf degree and gene degree in difference network for each tissue.
```{r}
colnames(tf_degrees)[1] <- "tf"
tf_degree_lcl <- melt(tf_degrees[,c(1,c(3:121))], id.vars = c("tf"))
tf_degree_lcl$tissue <- "LCL"
tf_degree_iPSC <- melt(tf_degrees[,c(1,c(123:241))], id.vars = c("tf"))
tf_degree_iPSC$tissue <- "iPSC"
tf_degree_iPSC_CM <- melt(tf_degrees[,c(1,c(243:361))], id.vars = c("tf"))
tf_degree_iPSC_CM$tissue <- "iPSC-CM"
tf_degrees_melted <- rbind(tf_degree_lcl,tf_degree_iPSC,tf_degree_iPSC_CM)
colnames(tf_degrees_melted) <- c("tf","individual","diff_out_degree","tissue")
ggplot(tf_degrees_melted, aes(x=tissue, y=diff_out_degree, color=tissue)) + geom_violin()

```

```{r}
colnames(gene_degrees)[1] <- "gene"
gene_degree_lcl <- melt(gene_degrees[,c(1,c(3,121))], id.vars = c("gene"))
gene_degree_lcl$tissue <- "LCL"
gene_degree_iPSC <- melt(gene_degrees[,c(1,c(123,241))], id.vars = c("gene"))
gene_degree_iPSC$tissue <- "iPSC"
gene_degree_iPSC_CM <- melt(gene_degrees[,c(1,c(243,361))], id.vars = c("gene"))
gene_degree_iPSC_CM$tissue <- "iPSC-CM"
gene_degrees_melted <- rbind(gene_degree_lcl,gene_degree_iPSC,gene_degree_iPSC_CM)
colnames(gene_degrees_melted) <- c("gene","individual","diff_in_degree","tissue")
ggplot(gene_degrees_melted, aes(x=tissue, y=diff_in_degree, color=tissue)) + geom_violin()

```

## Which TFs are most impacted by variants per individual, per tissue
We determine this by selecting TFs with high TF degree in the difference network per indivdual, per tissue. 
```{r}
library(dplyr)
# which TFs are most impacted in each individual
high_impact_tfs<- tf_degrees_melted[which(tf_degrees_melted$diff_out_degree > 15),]
head(high_impact_tfs)
dim(high_impact_tfs)
```

We then count, per TF, in how many individuals is this TF highly impacted. 
```{r}
indicence_high_impact_tfs <- high_impact_tfs %>% count(tf,tissue)
head(indicence_high_impact_tfs)
indicence_high_impact_tfs_wide <- dcast(indicence_high_impact_tfs, tf ~ tissue)
indicence_high_impact_tfs_wide[is.na(indicence_high_impact_tfs_wide)] <- 0
head(indicence_high_impact_tfs_wide)
```


## Which TFs are highly impacted in one tissue but not another?
We can now look, per TF, which are differentially highly impacted across tissues. For example, a TF might be highly impacted in many individuals in one tissue, but hardly impacted in another. (Hint, bring in ATAC-seq here - maybe these differences are due to ATAC-seq/open chromatin.)

```{r}
diffCounts <- abs(indicence_high_impact_tfs_wide$iPSC - indicence_high_impact_tfs_wide$LCL)
indicence_high_impact_tfs_tissue_specific <- melt(indicence_high_impact_tfs_wide[which(diffCounts > 40),], id.vars = c("tf"))
dim(indicence_high_impact_tfs_tissue_specific)
dim(indicence_high_impact_tfs)
colnames(indicence_high_impact_tfs_tissue_specific) <- colnames(indicence_high_impact_tfs)
ggplot(indicence_high_impact_tfs_tissue_specific, aes(x=tf, y=n, fill = tissue)) + geom_bar(position="dodge", stat="identity")

```


## which tfs out-degree varies a lot across the population for each tissue
```{r}
high_variance_tf_degrees <- aggregate(tf_degrees_melted$diff_out_degree, by = list(tf_degrees_melted$tf,tf_degrees_melted$tissue), FUN = var)
colnames(high_variance_tf_degrees) <- c("tf","tissue","variance_tf_diff_degree")
ggplot(high_variance_tf_degrees, aes(x=tissue, y=variance_tf_diff_degree, color=tissue)) + geom_violin()
```


# Parse and load the GWAS results
Here we extract trait-gene associations using the perl script for parsing GWAS data. First get entire GWAS catalog.

wget https://www.ebi.ac.uk/gwas/api/search/downloads/full



```{r}
system("perl /home/ubuntu/EGRET/final_egret_v1_banovich/scripts/parse_GWAS_associations.pl /home/ubuntu/EGRET/final_egret_v1_banovich/gwas/gwas_catalog 'Coronary artery disease' /home/ubuntu/EGRET/final_egret_v1_banovich/gwas/CAD_gwas_parsed.txt")

system("perl /home/ubuntu/EGRET/final_egret_v1_banovich/scripts/parse_GWAS_associations.pl /home/ubuntu/EGRET/final_egret_v1_banovich/gwas/gwas_catalog 'Rheumatoid arthritis' /home/ubuntu/EGRET/final_egret_v1_banovich/gwas/RA_gwas_parsed.txt")

system("perl /home/ubuntu/EGRET/final_egret_v1_banovich/scripts/parse_GWAS_associations.pl /home/ubuntu/EGRET/final_egret_v1_banovich/gwas/gwas_catalog 'Body mass index' /home/ubuntu/EGRET/final_egret_v1_banovich/gwas/BMI_gwas_parsed.txt")

system("perl /home/ubuntu/EGRET/final_egret_v1_banovich/scripts/parse_GWAS_associations.pl /home/ubuntu/EGRET/final_egret_v1_banovich/gwas/gwas_catalog 'Myocardial infarction' /home/ubuntu/EGRET/final_egret_v1_banovich/gwas/MI_gwas_parsed.txt")

system("perl /home/ubuntu/EGRET/final_egret_v1_banovich/scripts/parse_GWAS_associations.pl /home/ubuntu/EGRET/final_egret_v1_banovich/gwas/gwas_catalog 'Multiple sclerosis' /home/ubuntu/EGRET/final_egret_v1_banovich/gwas/MS_gwas_parsed.txt")

term <- "\"Crohn's disease\""
command <- paste0("perl /home/ubuntu/EGRET/final_egret_v1_banovich/scripts/parse_GWAS_associations.pl /home/ubuntu/EGRET/final_egret_v1_banovich/gwas/gwas_catalog ",term," /home/ubuntu/EGRET/final_egret_v1_banovich/gwas/CD_gwas_parsed.txt")
system(command)
```

Now load the gwas results each into a data frame

```{r}
# load gene annotation to assign gene id to gene name
nameGeneMap <- read.table("../../final_egret_v1/annotation/geneID_name_map.txt", header = FALSE)
colnames(nameGeneMap) <- c("gene","name")

RA <- read.table("../gwas/RA_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(RA) <- c("RAtrait","name")
RA$gene <- nameGeneMap$gene[match(RA$name, nameGeneMap$name)]

CD <- read.table("../gwas/CD_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(CD) <- c("CDtrait","name")
CD$gene <- nameGeneMap$gene[match(CD$name, nameGeneMap$name)]

CAD <- read.table("../gwas/CAD_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(CAD) <- c("CADtrait","name")
CAD$gene <- nameGeneMap$gene[match(CAD$name, nameGeneMap$name)]

BMI <- read.table("../gwas/BMI_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(BMI) <- c("BMItrait","name")
BMI$gene <- nameGeneMap$gene[match(BMI$name, nameGeneMap$name)]

MS <- read.table("../gwas/MS_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(MS) <- c("MStrait","name")
MS$gene <- nameGeneMap$gene[match(MS$name, nameGeneMap$name)]

MI <- read.table("../gwas/MI_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(MI) <- c("MItrait","name")
MI$gene <- nameGeneMap$gene[match(MI$name, nameGeneMap$name)]
```

Now load the difference table we made before, and merge in the GWAS traits
```{r}
diff_table$GWAS_RA_tf <- ifelse(diff_table$tf %in% RA$name, 1, 0)
diff_table$GWAS_RA_gene <- ifelse(diff_table$gene %in% RA$gene, 1, 0)

diff_table$GWAS_CD_tf <- ifelse(diff_table$tf %in% CD$name, 1, 0)
diff_table$GWAS_CD_gene <- ifelse(diff_table$gene %in% CD$gene, 1, 0)

diff_table$GWAS_CAD_tf <- ifelse(diff_table$tf %in% CAD$name, 1, 0)
diff_table$GWAS_CAD_gene <- ifelse(diff_table$gene %in% CAD$gene, 1, 0)

diff_table$GWAS_BMI_tf <- ifelse(diff_table$tf %in% BMI$name, 1, 0)
diff_table$GWAS_BMI_gene <- ifelse(diff_table$gene %in% BMI$gene, 1, 0)

diff_table$GWAS_MS_tf <- ifelse(diff_table$tf %in% MS$name, 1, 0)
diff_table$GWAS_MS_gene <- ifelse(diff_table$gene %in% MS$gene, 1, 0)

diff_table$GWAS_MI_tf <- ifelse(diff_table$tf %in% MI$name, 1, 0)
diff_table$GWAS_MI_gene <- ifelse(diff_table$gene %in% MI$gene, 1, 0)

save(diff_table, file = "../outputs/diff_table_with_gwas_banovich_06302020.RData")
```

