---
title: "Analysis of EGRET networks"
subtitle: "Analysis of EGRET networks derived from 119 individuals and 3 cell types (Banovich et al dataset)"
author: "Deborah Weighill"
output:
  html_notebook:
    toc: true
    theme: cosmo
    df_print: paged
---

# Load libraries
```{r}
library(dplyr)
library(xtable)
library(preprocessCore)
library(ggplot2)
library(tidyr)
library(ggpubr)
library(reshape2)
library(jcolors)
library(ALPACA)
library(gridExtra)
library(grid)
library(gridExtra)
library(ggthemes)
library(forcats)

# This is our color pallet
#amber, magenta, teal
pallet <- c("#FFBF00",  "#870E75"  ,  "#0BB19F" )

```

# Data loading and parsing: Make Panda/EGRET edge table

Load all of the PANDA/EGRET regnets and merge into a single table.

## Load the LCL results
```{r echo=FALSE,results='hide'}
library(reshape2)
# panda"
load("../outputs/finalEgret_v1_banovich_LCL_allModels_smart1_07032020_1_panda.RData")
net <- melt(regnet)
colnames(net) = c("tf","gene","score")
net$id <- paste0(net$tf,net$gene)
regnet_edge_table_LCL <- data.frame(net$id)
colnames(regnet_edge_table_LCL) <- c("id")
regnet_edge_table_LCL$tf <- as.vector(net$tf)
regnet_edge_table_LCL$gene <- as.vector(net$gene)
regnet_edge_table_LCL$panda <- as.vector(net$score)
for (g in c(1:119)) {
  filename <- paste0("../outputs/finalEgret_v1_banovich_LCL_allModels_smart1_07032020_",g,"_egret.RData")
  load(filename)
  net <- melt(regnet)
  colnames(net) = c("tf","gene","score")
  net$id <- paste0(net$tf,net$gene)
  if(all.equal(as.vector(regnet_edge_table_LCL$id),as.vector(net$id))){
    print(g)
    regnet_edge_table_LCL$tempID <- as.vector(net$score)
    #print(colnames(regnet_edge_table))
    colnames(regnet_edge_table_LCL)[colnames(regnet_edge_table_LCL) == "tempID"] <- paste0("LCL_indiv_",g)
  }
}
save(regnet_edge_table_LCL, file = "../outputs/LCL_table_EGRET_banovich_allModels.RData")
```

## Load the iPSC results
```{r echo=FALSE,results='hide'}
load("../outputs/finalEgret_v1_banovich_iPSC_allModels_smart1_07032020_1_panda.RData")
net <- melt(regnet)
colnames(net) = c("tf","gene","score")
net$id <- paste0(net$tf,net$gene)
regnet_edge_table_iPSC <- data.frame(net$id)
colnames(regnet_edge_table_iPSC) <- c("id")
regnet_edge_table_iPSC$tf <- as.vector(net$tf)
regnet_edge_table_iPSC$gene <- as.vector(net$gene)
regnet_edge_table_iPSC$panda <- as.vector(net$score)
for (g in c(1:119)) {
  filename <- paste0("../outputs/finalEgret_v1_banovich_iPSC_allModels_smart1_07032020_",g,"_egret.RData")
  load(filename)
  net <- melt(regnet)
  colnames(net) = c("tf","gene","score")
  net$id <- paste0(net$tf,net$gene)
  if(all.equal(as.vector(regnet_edge_table_iPSC$id),as.vector(net$id))){
    print(g)
    regnet_edge_table_iPSC$tempID <- as.vector(net$score)
    #print(colnames(regnet_edge_table))
    colnames(regnet_edge_table_iPSC)[colnames(regnet_edge_table_iPSC) == "tempID"] <- paste0("iPSC_indiv_",g)
  }
}
save(regnet_edge_table_iPSC, file = "../outputs/iPSC_table_EGRET_banovich_allModels.RData")
```

## Load the iPSC-CM results
```{r echo=FALSE,results='hide'}
load("../outputs/finalEgret_v1_banovich_iPSC-CM_allModels_smart1_07032020_1_panda.RData")
net <- melt(regnet)
colnames(net) = c("tf","gene","score")
net$id <- paste0(net$tf,net$gene)
regnet_edge_table_iPSC_CM <- data.frame(net$id)
colnames(regnet_edge_table_iPSC_CM) <- c("id")
regnet_edge_table_iPSC_CM$tf <- as.vector(net$tf)
regnet_edge_table_iPSC_CM$gene <- as.vector(net$gene)
regnet_edge_table_iPSC_CM$panda <- as.vector(net$score)
for (g in c(1:119)) {
  filename <- paste0("../outputs/finalEgret_v1_banovich_iPSC-CM_allModels_smart1_07032020_",g,"_egret.RData")
  load(filename)
  net <- melt(regnet)
  colnames(net) = c("tf","gene","score")
  net$id <- paste0(net$tf,net$gene)
  if(all.equal(as.vector(regnet_edge_table_iPSC_CM$id),as.vector(net$id))){
    print(g)
    regnet_edge_table_iPSC_CM$tempID <- as.vector(net$score)
    #print(colnames(regnet_edge_table))
    colnames(regnet_edge_table_iPSC_CM)[colnames(regnet_edge_table_iPSC_CM) == "tempID"] <- paste0("iPSC-CM_indiv_",g)
  }
}
save(regnet_edge_table_iPSC_CM, file = "../outputs/CM_table_EGRET_banovich_allModels.RData")
```

## Merge into a single table with an inner join
This ensures that we are considering the same set of edges for the two tissues (LCL and iPSC).

```{r}
pre_merged_regnet_table <- merge(regnet_edge_table_LCL,regnet_edge_table_iPSC, by.x = c(1,2,3), by.y = c(1,2,3), all = FALSE)
merged_regnet_table <- merge(pre_merged_regnet_table,regnet_edge_table_iPSC_CM,by.x = c(1,2,3), by.y = c(1,2,3), all = FALSE)
colnames(merged_regnet_table)[4] <- "LCL_panda"
colnames(merged_regnet_table)[124] <- "iPSC_panda"
colnames(merged_regnet_table)[244] <- "iPSC-CM_panda"
save(merged_regnet_table, file = "../outputs/merged_regnet_table_EGRET_banovich.RData")
```

## Construct edge differences from PANDA
This section will answer the question: *"Which Tf-gene edges are impacted by variants for each individual and each tissue?"* We first construct the difference table by substracting the PANDA edge weight from the EGRET edge weight and taking the absolute value of the difference. 

```{r}
diff_table <- merged_regnet_table
for (i in c(1:119)){
  diff_table[,4+i] <- abs(merged_regnet_table[,4+i]-merged_regnet_table[,4])
  diff_table[,124+i] <- abs(merged_regnet_table[,124+i]-merged_regnet_table[,124])
  diff_table[,244+i] <- abs(merged_regnet_table[,244+i]-merged_regnet_table[,244])
}
save(diff_table, file = "../outputs/diffTable_merged_banovich_edge_table.RData")
```

## Parse and load the GWAS results
Here we extract trait-gene associations using the perl script for parsing GWAS data. First get entire GWAS catalog.

wget https://www.ebi.ac.uk/gwas/api/search/downloads/full

```{r}
system("perl /home/ubuntu/EGRET/final_egret_v1_banovich/scripts/parse_GWAS_associations.pl /home/ubuntu/EGRET/final_egret_v1_banovich/gwas/gwas_catalog 'Coronary artery disease' /home/ubuntu/EGRET/final_egret_v1_banovich/gwas/CAD_gwas_parsed.txt")

term <- "\"Crohn's disease\""
command <- paste0("perl /home/ubuntu/EGRET/final_egret_v1_banovich/scripts/parse_GWAS_associations.pl /home/ubuntu/EGRET/final_egret_v1_banovich/gwas/gwas_catalog ",term," /home/ubuntu/EGRET/final_egret_v1_banovich/gwas/CD_gwas_parsed.txt")
system(command)

# also make tables of studies involved
system("perl /home/ubuntu/EGRET/final_egret_v1_banovich/scripts/parse_GWAS_associations_for_table.pl /home/ubuntu/EGRET/final_egret_v1_banovich/gwas/gwas_catalog 'Coronary artery disease' /home/ubuntu/EGRET/final_egret_v1_banovich/tables/CAD_study_table.txt")

term <- "\"Crohn's disease\""
command <- paste0("perl /home/ubuntu/EGRET/final_egret_v1_banovich/scripts/parse_GWAS_associations_for_table.pl /home/ubuntu/EGRET/final_egret_v1_banovich/gwas/gwas_catalog ",term," /home/ubuntu/EGRET/final_egret_v1_banovich/tables/CD_study_table.txt")
system(command)
```

Now load the gwas results into a data frame

```{r}
# load gene annotation to assign gene id to gene name
nameGeneMap <- read.table("../../final_egret_v1/annotation/geneID_name_map.txt", header = FALSE)
colnames(nameGeneMap) <- c("gene","name")

CD <- read.table("../gwas/CD_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(CD) <- c("CDtrait","name")
CD$gene <- nameGeneMap$gene[match(CD$name, nameGeneMap$name)]
CAD <- read.table("../gwas/CAD_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(CAD) <- c("CADtrait","name")
CAD$gene <- nameGeneMap$gene[match(CAD$name, nameGeneMap$name)]

```

Mark genes/tfs in our EGRET networks which are GWAS hits
```{r}
diff_table$GWAS_CD_tf <- ifelse(diff_table$tf %in% CD$name, 1, 0)
diff_table$GWAS_CD_gene <- ifelse(diff_table$gene %in% CD$gene, 1, 0)
diff_table$GWAS_CAD_tf <- ifelse(diff_table$tf %in% CAD$name, 1, 0)
diff_table$GWAS_CAD_gene <- ifelse(diff_table$gene %in% CAD$gene, 1, 0)
save(diff_table, file = "../outputs/diffTable_withGWAS_merged_banovich_edge_table.RData")
```

# Investigation of TF disruption Scores
Load the differences table
```{r}
load("../outputs/diffTable_withGWAS_merged_banovich_edge_table.RData")
```

## Load gwas
```{r}
# load gene annotation to assign gene id to gene name
nameGeneMap <- read.table("../../final_egret_v1/annotation/geneID_name_map.txt", header = FALSE)
colnames(nameGeneMap) <- c("gene","name")

CD <- read.table("../gwas/CD_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(CD) <- c("CDtrait","name")
CD$gene <- nameGeneMap$gene[match(CD$name, nameGeneMap$name)]

CAD <- read.table("../gwas/CAD_gwas_parsed.txt", header = FALSE,  sep = "\t")
colnames(CAD) <- c("CADtrait","name")
CAD$gene <- nameGeneMap$gene[match(CAD$name, nameGeneMap$name)]
```

## Hard threshold edge differences, aggregate by tf and gene
```{r}
diff_table_hard_thresh <- diff_table[,c(c(5:123),c(125:243),c(245:363))]
diff_table_hard_thresh[diff_table_hard_thresh < 0.35] <- 0
diff_table_hard_thresh <- cbind(diff_table[,c(2,3)],diff_table_hard_thresh)
tf_degrees_hard_thresh <- aggregate(diff_table_hard_thresh[,c(3:359)], by = list(diff_table_hard_thresh$tf), FUN = sum) 
gene_degrees_hard_thresh <- aggregate(diff_table_hard_thresh[,c(3:359)], by = list(diff_table_hard_thresh$gene), FUN = sum) 
colnames(tf_degrees_hard_thresh)[1] <- "tf"
colnames(gene_degrees_hard_thresh)[1] <- "gene"
head(tf_degrees_hard_thresh)
head(gene_degrees_hard_thresh)
tf_degrees_hard_thresh$GWAS_CD_tf <- ifelse(tf_degrees_hard_thresh$tf %in% CD$name, 1, 0)
gene_degrees_hard_thresh$GWAS_CD_gene <- ifelse(gene_degrees_hard_thresh$gene %in% CD$gene, 1, 0)
tf_degrees_hard_thresh$GWAS_CAD_tf <- ifelse(tf_degrees_hard_thresh$tf %in% CAD$name, 1, 0)
gene_degrees_hard_thresh$GWAS_CAD_gene <- ifelse(gene_degrees_hard_thresh$gene %in% CAD$gene, 1, 0)
save(tf_degrees_hard_thresh,file = "../outputs/tf_degrees_hard_thresh_035.RData")
save(gene_degrees_hard_thresh,file = "../outputs/gene_degrees_hard_thresh_035.RData")
```


## Scale and plot distributions

```{r}
tf_degrees_hard_thresh_scaled <- scale(tf_degrees_hard_thresh[,c(2:358)], center = TRUE, scale = TRUE)
tf_degrees_hard_thresh_scaled <- cbind(tf_degrees_hard_thresh[,c(1)], tf_degrees_hard_thresh_scaled, tf_degrees_hard_thresh[,c(359:360)])
colnames(tf_degrees_hard_thresh_scaled)[1] <-"tf"

tf_degrees <- tf_degrees_hard_thresh
tf_degree_lcl <- melt(tf_degrees[,c(1,c(2:120))], id.vars = c("tf"))
tf_degree_lcl$tissue <- "LCL"
tf_degree_iPSC <- melt(tf_degrees[,c(1,c(121:239))], id.vars = c("tf"))
tf_degree_iPSC$tissue <- "iPSC"
tf_degree_iPSC_CM <- melt(tf_degrees[,c(1,c(240:358))], id.vars = c("tf"))
tf_degree_iPSC_CM$tissue <- "iPSC-CM"
tf_degrees_melted <- rbind(tf_degree_lcl,tf_degree_iPSC,tf_degree_iPSC_CM)
colnames(tf_degrees_melted) <- c("tf","individual","diff_out_degree","Tissue")

ggplot(tf_degrees_melted, aes(x=Tissue, y=diff_out_degree, fill=Tissue, col = Tissue)) + geom_violin() + theme_bw() + scale_fill_manual(values=pallet) + scale_color_manual(values=pallet) + labs(title="Distribution of TF Disruption Scores", x ="Cell type", y = "TF Disruption Score") + theme(legend.position="none")

tf_degrees <- tf_degrees_hard_thresh_scaled
tf_degree_lcl <- melt(tf_degrees[,c(1,c(2:120))], id.vars = c("tf"))
tf_degree_lcl$tissue <- "LCL"
tf_degree_iPSC <- melt(tf_degrees[,c(1,c(121:239))], id.vars = c("tf"))
tf_degree_iPSC$tissue <- "iPSC"
tf_degree_iPSC_CM <- melt(tf_degrees[,c(1,c(240:358))], id.vars = c("tf"))
tf_degree_iPSC_CM$tissue <- "iPSC-CM"
tf_degrees_melted <- rbind(tf_degree_lcl,tf_degree_iPSC,tf_degree_iPSC_CM)
colnames(tf_degrees_melted) <- c("tf","individual","diff_out_degree","Tissue")

ggplot(tf_degrees_melted, aes(x=Tissue, y=diff_out_degree, fill=Tissue, col = Tissue)) + geom_violin() + theme_bw() + scale_fill_manual(values=pallet) + scale_color_manual(values=pallet) + labs(title="Distribution of TF Disruption Scores", x ="Cell type", y = "TF Disruption Score") + theme(legend.position="none")

```



## T-tests for disease genes

```{r}
colnames(tf_degree_iPSC_CM) <-  c("tf","individual","diff_out_degree","tissue")
colnames(tf_degree_lcl) <-  c("tf","individual","diff_out_degree","tissue")
colnames(tf_degree_iPSC) <-  c("tf","individual","diff_out_degree","tissue")

tf_degree_iPSC_CM$CAD <- tf_degrees$GWAS_CAD_tf[match(tf_degree_iPSC_CM$tf, tf_degrees$tf)]
tf_degree_iPSC_CM$CD <- tf_degrees_hard_thresh$GWAS_CD_tf[match(tf_degree_iPSC_CM$tf, tf_degrees_hard_thresh$tf)]
tf_degree_lcl$CD <- tf_degrees_hard_thresh$GWAS_CD_tf[match(tf_degree_lcl$tf, tf_degrees_hard_thresh$tf)]
tf_degree_lcl$CAD <- tf_degrees_hard_thresh$GWAS_CAD_tf[match(tf_degree_lcl$tf, tf_degrees_hard_thresh$tf)]

test_cad_in_cm <- t.test(tf_degree_iPSC_CM[which(tf_degree_iPSC_CM$CAD == 1),]$diff_out_degree,tf_degree_iPSC_CM[which(tf_degree_iPSC_CM$CAD == 0),]$diff_out_degree, alternative="greater")

test_cd_in_cm <- t.test(tf_degree_iPSC_CM[which(tf_degree_iPSC_CM$CD == 1),]$diff_out_degree,tf_degree_iPSC_CM[which(tf_degree_iPSC_CM$CD == 0),]$diff_out_degree, alternative="greater")

test_cd_in_lcl <- t.test(tf_degree_lcl[which(tf_degree_lcl$CD == 1),]$diff_out_degree,tf_degree_lcl[which(tf_degree_lcl$CD == 0),]$diff_out_degree, alternative="greater")

test_cad_in_lcl <- t.test(tf_degree_lcl[which(tf_degree_lcl$CAD == 1),]$diff_out_degree,tf_degree_lcl[which(tf_degree_lcl$CAD == 0),]$diff_out_degree, alternative="greater")



test_cad_in_lcl$p.value
test_cad_in_cm$p.value
test_cd_in_lcl$p.value
test_cd_in_cm$p.value
ttest_df <- NULL
ttest_df$Disease <- c("CAD","CAD","CD","CD")
ttest_df$CellType <- c("LCL","CM","LCL","CM")
ttest_df$pval <- as.character(signif(c(test_cad_in_lcl$p.value,test_cad_in_cm$p.value,test_cd_in_lcl$p.value,test_cd_in_cm$p.value),5))
print(xtable(as.data.frame(ttest_df), type = "latex"), file = "../tables/ttest_diseases_tissues.tex", include.rownames=FALSE)
```

## Individual-level TF disruption scores for CAD and CD genes

TF disruption scores for CAD genes
```{r}
data <- tf_degrees_hard_thresh_scaled
cad_tfs_degree_hard_thresh <- data[which(data$GWAS_CAD_tf == 1),c(1:358)]
cad_tfs_degree_hard_thresh_melted <- melt(cad_tfs_degree_hard_thresh)
colnames(cad_tfs_degree_hard_thresh_melted) <- c("tf","sample","tf_diff_degree")
cad_tfs_degree_hard_thresh_melted <- separate(cad_tfs_degree_hard_thresh_melted, sample, c("cellType","tag", "indiv"), "_", remove = TRUE)

cad_plot <- ggplot(cad_tfs_degree_hard_thresh_melted) + geom_point(shape = 19, aes(x = reorder(indiv, -tf_diff_degree), y = tf_diff_degree, col = cellType),) + theme_classic()+ scale_color_manual(values=pallet)+ xlab("Individual") + theme( axis.text.x=element_blank(), axis.ticks.x=element_blank())  + labs(title = "CAD TFs",col = "Cell type",shape = "TF")  + facet_grid(. ~ cellType)+ theme(legend.position="none") + ylab(bquote(d^(TF)))

disease_test_tfs_degree_hard_thresh <- data[which(data$GWAS_CD == 1),c(1:358)]
disease_test_tfs_degree_hard_thresh_melted <- melt(disease_test_tfs_degree_hard_thresh)
colnames(disease_test_tfs_degree_hard_thresh_melted) <- c("tf","sample","tf_diff_degree")
disease_test_tfs_degree_hard_thresh_melted <- separate(disease_test_tfs_degree_hard_thresh_melted, sample, c("cellType","tag", "indiv"), "_", remove = TRUE)

cd_plot <- ggplot(disease_test_tfs_degree_hard_thresh_melted) + geom_point(shape = 19, aes(x = reorder(indiv, -tf_diff_degree), y = tf_diff_degree, col = cellType),) + theme_classic()+ scale_color_manual(values=pallet)+ xlab("Individual") + theme( axis.text.x=element_blank(), axis.ticks.x=element_blank())  + labs(title = "CD TFs",col = "Cell type",shape = "TF") + facet_grid(. ~ cellType)+ theme(legend.position="none") + ylab(bquote(d^(TF)))

cad_annotated <- annotate_figure(cad_plot,fig.lab = "A", fig.lab.face = "plain", fig.lab.pos = "top.left", fig.lab.size = 14)
cd_annotated <- annotate_figure(cd_plot,fig.lab = "B", fig.lab.face = "plain", fig.lab.pos = "top.left", fig.lab.size = 14)

combined_plot <- grid.arrange(cad_annotated, cd_annotated, nrow=2, heights = c(3,3))
combined_plot_annotated <- annotate_figure(combined_plot,top = text_grob("TF disruption scores for disease-related TFs", color = "black", face = "bold", size = 14))

pdf("../figures/disease_tfs_disruptionscore_plot.pdf",width = 6, height = 4)
plot(combined_plot_annotated)
dev.off()

```

```{r}
# which individuals have the ipsc cm disruptions for ERG
indivs_high_erg_disrup <- cad_tfs_degree_hard_thresh_melted[which((cad_tfs_degree_hard_thresh_melted$tf_diff_degree>3) & (cad_tfs_degree_hard_thresh_melted$cellType == "iPSC-CM")),c('indiv','tf_diff_degree')]
indivs_high_erg_disrup

# get tf disruption scores for ERG for all individuals in iPSC-CMs
tf_disrup_erg_cm <- cad_tfs_degree_hard_thresh_melted[which((cad_tfs_degree_hard_thresh_melted$tf == "ERG") & (cad_tfs_degree_hard_thresh_melted$cellType == "iPSC-CM")),c('indiv','tf_diff_degree')]
```
This is the eQTL for ERG
ERG	ENSG00000159176	201476815	chr1	0.461

# ALPACA analysis

## Run alpaca
Use the Rmd's to run alpaca

- ../scripts/run_alpaca_LCL.Rmd  
- ../scripts/run_alpaca_iPSC-CM.Rmd  
- ../scripts/run_alpaca_iPSC.Rmd

## Load alpaca results
LCLs
```{r}
net <- read.table("../alpaca/LCL_100_ALPACA_scores.txt", header = FALSE, sep = "\t")
colnames(net) = c("node","score")
net <- separate(net, node, c("node","set"), "_", remove = TRUE)

alpaca_score_table_LCL <- data.frame(net$node[which(net$set == "B")])
colnames(alpaca_score_table_LCL) = c("node")
for (n in c(1:119)) {
  file <- paste0("../alpaca/LCL_",n,"_ALPACA_scores.txt")
  net <- read.table(file, header = FALSE, sep = "\t")
  colnames(net) = c("node","score")
  net <- separate(net, node, c("node","set"), "_", remove = TRUE)
  print(n)
  alpaca_score_table_LCL$tempID <- net$score[match(alpaca_score_table_LCL$node, net$node)]
  colnames(alpaca_score_table_LCL)[colnames(alpaca_score_table_LCL) == "tempID"] <- paste0("LCL_indiv_",n)
  }
save(alpaca_score_table_LCL, file = "../alpaca/LCL_alpaca_score_table_genes.RData")
```

iPSCs
```{r}
net <- read.table("../alpaca/iPSC_100_ALPACA_scores.txt", header = FALSE, sep = "\t")
colnames(net) = c("node","score")
net <- separate(net, node, c("node","set"), "_", remove = TRUE)

alpaca_score_table_iPSC <- data.frame(net$node[which(net$set == "B")])
colnames(alpaca_score_table_iPSC) = c("node")
for (n in c(1:119)) {
  file <- paste0("../alpaca/iPSC_",n,"_ALPACA_scores.txt")
  net <- read.table(file, header = FALSE, sep = "\t")
  colnames(net) = c("node","score")
  net <- separate(net, node, c("node","set"), "_", remove = TRUE)
  print(n)
  alpaca_score_table_iPSC$tempID <- net$score[match(alpaca_score_table_iPSC$node, net$node)]
  colnames(alpaca_score_table_iPSC)[colnames(alpaca_score_table_iPSC) == "tempID"] <- paste0("iPSC_indiv_",n)
  }
save(alpaca_score_table_iPSC, file = "../alpaca/iPSC_alpaca_score_table_genes.RData")
```

iPSC-CMs
```{r}
net <- read.table("../alpaca/iPSC-CM_100_ALPACA_scores.txt", header = FALSE, sep = "\t")
colnames(net) = c("node","score")
net <- separate(net, node, c("node","set"), "_", remove = TRUE)

alpaca_score_table_iPSCCM <- data.frame(net$node[which(net$set == "B")])
colnames(alpaca_score_table_iPSCCM) = c("node")
for (n in c(1:119)) {
  file <- paste0("../alpaca/iPSC-CM_",n,"_ALPACA_scores.txt")
  net <- read.table(file, header = FALSE, sep = "\t")
  colnames(net) = c("node","score")
  net <- separate(net, node, c("node","set"), "_", remove = TRUE)
  print(n)
  alpaca_score_table_iPSCCM$tempID <- net$score[match(alpaca_score_table_iPSCCM$node, net$node)]
  colnames(alpaca_score_table_iPSCCM)[colnames(alpaca_score_table_iPSCCM) == "tempID"] <- paste0("iPSCCM_indiv_",n)
  }
save(alpaca_score_table_iPSCCM, file = "../alpaca/iPSC-CM_alpaca_score_table_genes.RData")
```


## Merge into a single table with an inner join

```{r}
load("../alpaca/LCL_alpaca_score_table_genes.RData")
load("../alpaca/iPSC_alpaca_score_table_genes.RData")
load("../alpaca/iPSC-CM_alpaca_score_table_genes.RData")

pre_merged_alpaca_table <- merge(alpaca_score_table_LCL,alpaca_score_table_iPSC, by.x = c(1), by.y = c(1), all = FALSE)
merged_alpaca_table <- merge(pre_merged_alpaca_table,alpaca_score_table_iPSCCM,by.x = c(1), by.y = c(1), all = FALSE)
save(merged_alpaca_table, file = "../alpaca/merged_alpaca_table_07302020_genes.RData")
load("../alpaca/merged_alpaca_table_07302020_genes.RData")
```

## Scale ALPACA scores

```{r}
merged_alpaca_table_qq <- cbind(merged_alpaca_table[,c(1)],as.data.frame(normalize.quantiles(as.matrix(merged_alpaca_table[,c(2:358)]))))
colnames(merged_alpaca_table_qq) <- colnames(merged_alpaca_table)

merged_alpaca_table_qq_sc <- cbind(merged_alpaca_table_qq[,c(1)],as.data.frame(t(scale(t(merged_alpaca_table_qq[,c(2:358)]), center = TRUE, scale = TRUE))))
colnames(merged_alpaca_table_qq_sc)[1] <- "node"


melted_alpaca <- melt(merged_alpaca_table_qq_sc)
colnames(melted_alpaca) <- c("node","sample","alpaca_score")
melted_alpaca <- separate(melted_alpaca, sample, c("cellType","tag", "indiv"), "_", remove = TRUE)
melted_alpaca$cellType[which(melted_alpaca$cellType == "iPSCCM")] <- "iPSC-CM"
alpaca_dist <- ggplot(melted_alpaca) + geom_violin(aes(x = cellType, y = (alpaca_score), fill = cellType, col = cellType)) + theme_classic()+ scale_fill_manual(values=pallet) + scale_color_manual(values=pallet)+ labs(y = "DM score",x = "Cell type", title = "Gene DM scores") + theme(legend.position = "none") 
alpaca_dist
pdf("../figures/alpaca_dist.pdf",width = 3, height = 3)
plot(alpaca_dist)
dev.off()

```

## Plots of ALPACA scores for top disease genes in relevant cell type

```{r}
data <- merged_alpaca_table_qq_sc
only_CAD_genes <- as.vector(CAD$gene)[which(!((as.vector(CAD$gene) %in% as.vector(CD$gene))))]
cads <- data[which(data$node %in% CAD$gene),c(1:358)]
cads_melted <- melt(cads)
colnames(cads_melted) <- c("node","sample","alpaca_score")
cads_melted <- separate(cads_melted, sample, c("cellType","tag", "indiv"), "_", remove = TRUE)
cads_melted$cellType[which(cads_melted$cellType == "iPSCCM")] <- "iPSC-CM"
high_score_nodes_cad <- unique(cads_melted[which(((cads_melted$alpaca_score) > 1.7) & (cads_melted$cellType == "iPSC-CM")),'node'])
cads <- data[which(data$node %in% high_score_nodes_cad),c(1:358)]
cads_melted <- melt(cads)
colnames(cads_melted) <- c("node","sample","alpaca_score")
cads_melted <- separate(cads_melted, sample, c("cellType","tag", "indiv"), "_", remove = TRUE)
cads_melted$cellType[which(cads_melted$cellType == "iPSCCM")] <- "iPSC-CM"
cad_alpaca_plot <- ggplot(cads_melted) + geom_violin(aes(x = cellType, y = (alpaca_score), fill = cellType, col = cellType)) + theme_classic()+ scale_fill_manual(values=pallet) + scale_color_manual(values=pallet)+ labs(y = "DM score",x = "Cell type", title = "Top CAD genes in CMs") + theme(legend.position = "none") 
cad_alpaca_plot <- annotate_figure(cad_alpaca_plot,fig.lab = "A", fig.lab.face = "plain", fig.lab.pos = "top.left", fig.lab.size = 14)


only_CD_genes <- as.vector(CD$gene)[which(!((as.vector(CD$gene) %in% as.vector(CAD$gene))))]
cds <- data[which(data$node %in% CD$gene),c(1:358)]
cds_melted <- melt(cds)
colnames(cds_melted) <- c("node","sample","alpaca_score")
cds_melted <- separate(cds_melted, sample, c("cellType","tag", "indiv"), "_", remove = TRUE)
high_score_nodes_cd <- unique(cds_melted[which((abs(cds_melted$alpaca_score) > 1.7) & (cds_melted$cellType == "LCL")),'node'])
cds <- data[which(data$node %in% high_score_nodes_cd),c(1:358)]
cds_melted <- melt(cds)
colnames(cds_melted) <- c("node","sample","alpaca_score")
cds_melted <- separate(cds_melted, sample, c("cellType","tag", "indiv"), "_", remove = TRUE)
cds_melted$cellType[which(cds_melted$cellType == "iPSCCM")] <- "iPSC-CM"
cd_alpaca_plot <- ggplot(cds_melted) + geom_violin(aes(x = cellType, y = (alpaca_score), fill = cellType, col = cellType)) + theme_classic()+ scale_fill_manual(values=pallet) + scale_color_manual(values=pallet)+ labs(y = "DM score",x = "Cell type", title  = "Top CD genes in LCLs") + theme(legend.position = "none") 
cd_alpaca_plot <- annotate_figure(cd_alpaca_plot,fig.lab = "B", fig.lab.face = "plain", fig.lab.pos = "top.left", fig.lab.size = 14)


combined_plot <- grid.arrange(cad_alpaca_plot, cd_alpaca_plot, nrow=1, widths = c(3,3))
combined_plot_annotated <- annotate_figure(combined_plot,top = text_grob("Distribution of differential modularity scores", color = "black", face = "bold", size = 14))


pdf("../figures/alpaca_scores.pdf",width = 6, height = 3)
plot(combined_plot_annotated)
dev.off()
```


## Example - individuals with mutation in ERG binding site affecting expression of CSRP1
```{r}
geno <- read.table("../data/genotypes/erg_variant_genotypes", header = FALSE)
data <- merged_alpaca_table_qq_sc
gene <- "ENSG00000159176"
genes <- data[which(data$node == gene),c(1:358)]
genes_melted <- melt(genes)
colnames(genes_melted) <- c("node","sample","alpaca_score")
genes_melted <- separate(genes_melted, sample, c("cellType","tag", "indiv"), "_", remove = TRUE)
genes_melted$tds <- tf_disrup_erg_cm$tf_diff_degree[match(genes_melted$indiv,tf_disrup_erg_cm$indiv)]
genes_melted$dosage <- as.numeric(as.vector(c(as.vector(geno[1,c(10:128)]),as.vector(geno[1,c(10:128)]),as.vector(geno[1,c(10:128)]))))

p1 <- ggplot(genes_melted[which(genes_melted$cellType == "iPSCCM"),]) + geom_point(shape = 21,aes(x = reorder(indiv, -alpaca_score), y = (alpaca_score), fill = dosage, size = tds)) + theme_classic()+ scale_fill_gradient(low = "white", high = "#870E75") + theme(axis.text.x=element_blank(), axis.ticks.x=element_blank())+ labs(y = "DM score",fill = "Alt. dosage",size = "ERG TF d'", x= "Individual") 

p2 <- annotate_figure(p1,top = text_grob("DM scores for CSRP1 (ENSG00000159176)", color = "black", face = "bold", size = 14))
pdf("../figures/example_gene_alpaca_dosage_and_tfds.pdf",width = 4, height = 3.8)
plot(p2)
dev.off()

plot(p2)
```


# Gene alpaca scores for GO enrichment - individual 18
```{r}
write.table(as.vector(alpaca_score_table_iPSCCM[order(-alpaca_score_table_iPSCCM$iPSCCM_indiv_18),1]),file = "../go/gene_alpaca_ipsc_cm_indiv_18.txt", sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)
```

```{r}
write.table(as.vector(alpaca_score_table_LCL[order(-alpaca_score_table_LCL$LCL_indiv_18),1]),file = "../go/gene_alpaca_lcl_indiv_18.txt", sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)
```

```{r}
write.table(as.vector(alpaca_score_table_iPSC[order(-alpaca_score_table_iPSC$iPSC_indiv_18),1]),file = "../go/gene_alpaca_ipsc_indiv_18.txt", sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)
```

GO enrichment was performed on the above ranked lists using GOrilla (http://cbl-gorilla.cs.technion.ac.il/). The code below creates the GO term plots. Because there are many terms enriched only in CMs making plotting diffifult for a main text figure, for terms enriched only in CMs, we plot only those terms we wish to discuss in the main text figure, and include a full plot of all terms as a suplemental figure. Both of these figures (reduced figure and full figure) are produced below:

plot go terms
cat GO_indiv18_lcl.txt | tr '^M' '\n' | grep "^GO:" >formatted_GO_indiv18_lcl.txt
cat GO_indiv18_ipsc.txt | tr '^M' '\n' | grep "^GO:" >formatted_GO_indiv18_ipsc.txt
cat GO_indiv18_cm.txt | tr '^M' '\n' | grep "^GO:" >formatted_GO_indiv18_cm.txt

```{r}
go_cm <- read.table("../go/formatted_GO_indiv18_cm.txt", header = FALSE, sep = "\t")
colnames(go_cm) <- c("GOTerm",	"Description",	"Pvalue",	"FDR",	"Enrichment",	"N"	,"B",	"n"	,"b",	"Genes")
go_cm$celltype <- "CM"

go_lcl <- read.table("../go/formatted_GO_indiv18_lcl.txt", header = FALSE, sep = "\t")
colnames(go_lcl) <- c("GOTerm",	"Description",	"Pvalue",	"FDR",	"Enrichment",	"N"	,"B",	"n"	,"b",	"Genes")
go_lcl$celltype <- "LCL"

go_ipsc <- read.table("../go/formatted_GO_indiv18_ipsc.txt", header = FALSE, sep = "\t")
colnames(go_ipsc) <- c("GOTerm",	"Description",	"Pvalue",	"FDR",	"Enrichment",	"N"	,"B",	"n"	,"b",	"Genes")
go_ipsc$celltype <- "iPSC"

go_enrich <- rbind(go_ipsc,go_lcl,go_cm)
terms_to_plot <- read.table("../go/plot_terms", header = FALSE, sep = "\t")

only_cm <- go_cm$Description[which((!(go_cm$Description %in% go_lcl$Description)) & (!(go_cm$Description %in% go_ipsc$Description)))]
only_lcl <- go_lcl$Description[which((!(go_lcl$Description %in% go_cm$Description)) & (!(go_lcl$Description %in% go_ipsc$Description)))]
only_ipsc <- go_ipsc$Description[which((!(go_ipsc$Description %in% go_lcl$Description)) & (!(go_ipsc$Description %in% go_cm$Description)))]

go_enrich_plot <- go_enrich[which(go_enrich$Description %in% terms_to_plot$V1),]
go_enrich_plot$color <- 'black'
go_enrich_plot$color[which(go_enrich_plot$Description %in% only_cm)] <- '#870E75'
go_enrich_plot$color[which(go_enrich_plot$Description %in% only_lcl)] <- '#0BB19F'
go_enrich_plot$color[which(go_enrich_plot$Description %in% only_ipsc)] <- '#FFBF00'
```


```{r}
color <- factor(unique(go_enrich_plot[,c("Description", "color")])$color, levels = c("black","#FFBF00","#0BB19F", "#870E75"))
go_enrich_plot$color2 <- color[match(go_enrich_plot$color, color)]
order <- with(unique(go_enrich_plot[,c("Description", "color2")]), order(color2))

Description <- factor(unique(go_enrich_plot[,c("Description", "color2")])$Description, levels = unique(go_enrich_plot[,c("Description", "color2")])[order,"Description"])
go_enrich_plot$Description2 <- Description[match(go_enrich_plot$Description, Description)]
 #
go_enrich_plot$celltype <- factor(go_enrich_plot$celltype, levels = c("CM","LCL","iPSC"))

p <- ggplot(go_enrich_plot, aes(x=-log10(Pvalue), y=Description2, colour=celltype, group = celltype, size=b)) + geom_point() + expand_limits(x=0) + labs(x="-log10(P-value)", y="GO term", colour="P value", size="Count") + theme_few() + scale_color_manual(values=pallet[c(2,3,1)]) +  facet_wrap(~celltype)
y_labs <- ggplot_build(p)$layout$panel_params[[1]]$y$get_labels()
col_vec <- as.vector(go_enrich_plot$color2)[match(y_labs,go_enrich_plot$Description2)]

p <- p + theme(axis.text.y = element_text(colour = col_vec)) + scale_x_continuous(name="-log10(p-value)",limits=c(1, 8))
pdf("goplot.pdf",width = 9, height = 8)
plot(p)
dev.off()
```


```{r}
go_enrich$color <- 'black'
go_enrich$color[which(go_enrich$Description %in% only_cm)] <- '#870E75'
go_enrich$color[which(go_enrich$Description %in% only_lcl)] <- '#0BB19F'
go_enrich$color[which(go_enrich$Description %in% only_ipsc)] <- '#FFBF00'

color <- factor(unique(go_enrich[,c("Description", "color")])$color, levels = c("black","#FFBF00","#0BB19F", "#870E75"))
go_enrich$color2 <- color[match(go_enrich$color, color)]
order <- with(unique(go_enrich[,c("Description", "color2")]), order(color2))

Description <- factor(unique(go_enrich[,c("Description", "color2")])$Description, levels = unique(go_enrich[,c("Description", "color2")])[order,"Description"])
go_enrich$Description2 <- Description[match(go_enrich$Description, Description)]
 #
go_enrich$celltype <- factor(go_enrich$celltype, levels = c("CM","LCL","iPSC"))

p <- ggplot(go_enrich, aes(x=-log10(Pvalue), y=Description2, colour=celltype, group = celltype, size=b)) + geom_point() + expand_limits(x=0) + labs(x="-log10(P-value)", y="GO term", colour="P value", size="Count") + theme_few() + scale_color_manual(values=pallet[c(2,3,1)]) +  facet_wrap(~celltype)
y_labs <- ggplot_build(p)$layout$panel_params[[1]]$y$get_labels()
col_vec <- as.vector(go_enrich$color2)[match(y_labs,go_enrich$Description2)]

p <- p + theme(axis.text.y = element_text(colour = col_vec)) + scale_x_continuous(name="-log10(p-value)")
pdf("goplot_all.pdf",width = 9, height = 16)
plot(p)
dev.off()

```
