---
title: "Comparison of genetically distict cell lines with EGRET"
author: "Deborah Weighill"
output:
  html_document:
    df_print: paged
    toc: yes
  html_notebook:
    df_print: paged
    theme: cosmo
    toc: yes
---


# Load libraries
Load libraries, set up color pallet and set the paths to the ChIP-seq data sets for the two cell lines.
```{r message=FALSE}
library(xtable)
library(jcolors)
library(precrec)
library(gridExtra)
library(grid)
library(tidyr)
library(plyr)
library(qvalue)
library(ggpubr)
library(dplyr)
library(pROC)
library(ALPACA)
library(viridis)
library(lemon)
library(GenomicRanges)
library(ggplot2)
library(reshape2)
library(rtracklayer)

# set color pallet
pallet <- c("#FFBF00",  "#870E75"  ,  "#0BB19F" )
chipseq_gm12878 <- "../chipseq/GM12878_remap2018_all_macs2_hg19_v1_2.bed"
chipseq_k562 <- "../chipseq/K562_remap2018_all_macs2_hg19_v1_2.bed"

# NOTE: a log file is used to print out statistics called out in the paper.
logfile <- paste0("../outputs/log_cell_line_comparison.txt")
write(date(), file = logfile, append = FALSE)

```

# Create validation table for GM12878

## Load GM12878 networks into table
Here, we load the EGRET network for GM12878 as well as the baseline PANDA network and melt/combine them it into a data frame with each row representing an edge.
```{r echo=FALSE,results='hide'}
load("../outputs/EGRET_allQBiCModels_gm12878_10142020_panda.RData")
net <- melt(regnet)
colnames(net) = c("tf","gene","score")
net$id <- paste0(net$tf,net$gene)

regnet_edge_table_gm12878 <- data.frame(net$id)
colnames(regnet_edge_table_gm12878) <- c("id")
regnet_edge_table_gm12878$tf <- as.vector(net$tf)
regnet_edge_table_gm12878$gene <- as.vector(net$gene)
regnet_edge_table_gm12878$panda <- as.vector(net$score)

#list of the file names of the rest of the networks
files <- c("EGRET_allQBiCModels_gm12878_10142020_egret.RData")

#nicknames of the networks
names <- c("egret_GM12878")

for (p in c(1:1)) {
  filename <- paste0("../outputs/",files[p])
  netname <- names[p]
  print(netname)
  load(filename)
  net <- melt(regnet)
  colnames(net) = c("tf","gene","score")
  net$id <- paste0(net$tf,net$gene)
  if(all.equal(as.vector(regnet_edge_table_gm12878$id),as.vector(net$id))){
    print(p)
    regnet_edge_table_gm12878$tempID <- as.vector(net$score)
    colnames(regnet_edge_table_gm12878)[colnames(regnet_edge_table_gm12878) == "tempID"] <- netname
  }
}
filename <- paste0("../outputs/edge_table_gm12878_10142020.RData")
save(regnet_edge_table_gm12878, file = filename)
```

## Load ChIP-seq data for GM12878
We load the chipseq data and make a ground truth regulatory network for validation.

```{r echo=FALSE,results='hide'}
# read in the gene annotation file with gene ranges.
genes <- read.table("../annotation/ensembl_genes_from_ucsc.txt", header = TRUE,sep = "\t")
genes$promotorLeft <- ifelse(genes$strand == "+", (genes$txStart + 1 - 750), (genes$txEnd-250))
genes$promotorRight <- ifelse(genes$strand == "+", (genes$txStart + 1 + 250), (genes$txEnd+750))
gr_genes_promotor <- GRanges(seqnames = genes$chrom, ranges = IRanges(start = genes$promotorLeft, end = genes$promotorRight), strand = NULL, mcols = genes[,13:15])

# load  chipseq data and overlap with gene promotor regions
gr_chipseq_gm12878 <- import(chipseq_gm12878)
overlaps_gm12878 <- data.frame(findOverlaps(gr_chipseq_gm12878, gr_genes_promotor))
overlaps_gm12878$tf_name <- gr_chipseq_gm12878$name[overlaps_gm12878$queryHits]
overlaps_gm12878$gene_name <- gr_genes_promotor$mcols.name2[overlaps_gm12878$subjectHits]
validation_regnet_gm12878 <- separate(overlaps_gm12878, tf_name, c("exp", "tf", "cell line"), "\\.")
chipseq_regnet_gm12878 <- distinct(validation_regnet_gm12878[,c("tf","gene_name")])
chipseq_regnet_gm12878$id <- paste0(chipseq_regnet_gm12878$tf,chipseq_regnet_gm12878$gene)

# get number of TFs with ChIP-seq data in GM12878
experiments <- as.data.frame(gr_chipseq_gm12878$name)
colnames(experiments)  <- c("experiment")
experiments <- separate(experiments, experiment, c("exp", "tf", "cell line"), "\\.")

# write to log file
message <- paste0("Number of TFs with ChIP-seq experiments in GM12878: ",length(unique(experiments$tf)))
write(c(message), file = logfile, append = TRUE)
write("\n", file = logfile, append = TRUE)
```

## Overlap egret/panda networks with ChIP-seq data and calcualte differences
We now filter down the panda/egret networks to only include those TFs which are in the ChIP-seq data.
```{r echo=FALSE,results='hide'}
filtered_net_table_gm12878 <- regnet_edge_table_gm12878[(regnet_edge_table_gm12878$tf %in% chipseq_regnet_gm12878$tf) ,]
filtered_net_table_gm12878$chipseq <- ifelse(filtered_net_table_gm12878$id %in% chipseq_regnet_gm12878$id,1,0)
# this calculates the edge dusruption scores in GM12878
filtered_net_table_gm12878$diff_egret <- abs(filtered_net_table_gm12878$egret_GM12878 - filtered_net_table_gm12878$panda)

validation_table_gm12878 <- filtered_net_table_gm12878
filename <- paste0("../outputs/validation_table_gm12878_10142020.RData")
save(validation_table_gm12878, file = filename)
load("../outputs/validation_table_gm12878_10142020.RData")
```

# Create validation table for K562 

## Load K562 networks into table

Here, we load the EGRET network for K562 as well as the baseline PANDA network and melt/combine them it into a data frame with each row representing an edge.

```{r echo=FALSE,results='hide'}
# panda - note - the panda network is the same file for K562 and GM12878 as there is no genotype information
load("../outputs/EGRET_allQBiCModels_gm12878_10142020_panda.RData")
net <- melt(regnet)
colnames(net) = c("tf","gene","score")
net$id <- paste0(net$tf,net$gene)

regnet_edge_table_k562 <- data.frame(net$id)
colnames(regnet_edge_table_k562) <- c("id")
regnet_edge_table_k562$tf <- as.vector(net$tf)
regnet_edge_table_k562$gene <- as.vector(net$gene)
regnet_edge_table_k562$panda <- as.vector(net$score)

#list of the file names of the rest of the networks
files <- c("EGRET_allQBiCModels_k562_10142020_egret.RData")

#nicknames of the networks
names <- c("egret_k562")

for (p in c(1:1)) {
  filename <- paste0("../outputs/",files[p])
  netname <- names[p]
  print(netname)
  load(filename)
  net <- melt(regnet)
  colnames(net) = c("tf","gene","score")
  net$id <- paste0(net$tf,net$gene)
  if(all.equal(as.vector(regnet_edge_table_k562$id),as.vector(net$id))){
    print(p)
    regnet_edge_table_k562$tempID <- as.vector(net$score)
    colnames(regnet_edge_table_k562)[colnames(regnet_edge_table_k562) == "tempID"] <- netname
  }
}
filename <- paste0("../outputs/edge_table_k562_10142020.RData")
save(regnet_edge_table_k562, file = filename)
```

## Load ChIP-seq data for K562
We load the chipseq data and make a ground truth regulatory network for validation.

```{r echo=FALSE,results='hide'}
gr_chipseq_k562 <- import(chipseq_k562)
overlaps_k562 <- data.frame(findOverlaps(gr_chipseq_k562, gr_genes_promotor))
overlaps_k562$tf_name <- gr_chipseq_k562$name[overlaps_k562$queryHits]
overlaps_k562$gene_name <- gr_genes_promotor$mcols.name2[overlaps_k562$subjectHits]
validation_regnet_k562 <- separate(overlaps_k562, tf_name, c("exp", "tf", "cell line"), "\\.")
chipseq_regnet_k562 <- distinct(validation_regnet_k562[,c("tf","gene_name")])
chipseq_regnet_k562$id <- paste0(chipseq_regnet_k562$tf,chipseq_regnet_k562$gene)

# print number of TFs with ChIP-seq data in K562
experiments <- as.data.frame(gr_chipseq_k562$name)
colnames(experiments)  <- c("experiment")
experiments <- separate(experiments, experiment, c("exp", "tf", "cell line"), "\\.")
length(unique(experiments$tf))

# write to log file
message <- paste0("Number of TFs with ChIP-seq experiments in K562: ",length(unique(experiments$tf)))
write(c(message), file = logfile, append = TRUE)
write("\n", file = logfile, append = TRUE)
```

## Overlap egret/panda networks with ChIP-seq data and calcualte differences
We now filter down the panda/egret networks to only include those TFs which are in the ChIP-seq data (our validation set).
```{r echo=FALSE,results='hide'}
filtered_net_table_k562 <- regnet_edge_table_k562[(regnet_edge_table_k562$tf %in% chipseq_regnet_k562$tf) ,]
filtered_net_table_k562$chipseq <- ifelse(filtered_net_table_k562$id %in% chipseq_regnet_k562$id,1,0)
# this calculates the edge dusruption scores in K562
filtered_net_table_k562$diff_egret <- abs(filtered_net_table_k562$egret_k562 - filtered_net_table_k562$panda)

validation_table_k562 <- filtered_net_table_k562
filename <- paste0("../outputs/validation_table_k562_10142020.RData")
save(validation_table_k562, file = filename)
load("../outputs/validation_table_k562_10142020.RData")
```


# Plot distributions
We plot the distribution of the edge disruption scores for K562 and GM12878.
```{r}
# violin plot of complete distributions
colnames(validation_table_gm12878) <- c("id","tf","gene","panda","egret_gm12878","chipseq_gm12878","diff_gm12878")
colnames(validation_table_k562) <- c("id","tf","gene","panda","egret_k562","chipseq_k562","diff_k562")
combined_table <- join(validation_table_gm12878,validation_table_k562,by = c("id","tf","gene"),type="inner")
save(combined_table, file = "../outputs/combined_validation_table_10142020.RData")
load("../outputs/combined_validation_table_10142020.RData")
colnames(combined_table)[which(colnames(combined_table) == 'diff_gm12878')] <- "GM12878"
colnames(combined_table)[which(colnames(combined_table) == 'diff_k562')] <- "K562"
dis_melted <- melt(combined_table[,c('id','GM12878','K562')], id.vars = 'id')
colnames(dis_melted) <- c("edge","cell_line","d")

p1 <- ggplot(dis_melted, aes(x=cell_line, y = d, group = cell_line, col = cell_line)) + geom_violin() + theme_bw() + scale_color_manual(values=pallet[c(3,2)]) + scale_fill_manual(values=pallet[c(3,2)]) +labs(x ="Genotype", y = "Edge disruption score") + theme(legend.position = "none")
p1 <- annotate_figure(p1,fig.lab = "A", fig.lab.face = "plain", fig.lab.pos = "top.left", fig.lab.size = 14)
           
# Log boxplot to see quantiles
p2 <- ggplot(dis_melted, aes(x=cell_line, y = log(d), group = cell_line, col = cell_line)) + geom_boxplot() + theme_bw() + scale_color_manual(values=pallet[c(3,2)]) + scale_fill_manual(values=pallet[c(3,2)]) +labs(x ="Genotype", y = "log10(Edge disruption score)") + theme(legend.position = "none") 
p2 <- annotate_figure(p2,fig.lab = "B", fig.lab.face = "plain", fig.lab.pos = "top.left", fig.lab.size = 14) 

p3 <- grid.arrange(p1 + theme(legend.position="none"), p2 + theme(legend.position="none"), nrow=1, widths = c(4,4))
p4 <- annotate_figure(p3,top = text_grob("Distribution of edge disruption scores", color = "black", face = "bold", size = 14))
pdf("../figures/edge_disrup_score_cell_lines.pdf",width = 6, height = 3)
plot(p4)
dev.off()
```

# Select examples for visualizing in genome browser
Here we select some examples for visualization in IGV, to show examples where the GM12878 and K562 EGRET networks have differential ChIP-seq binding and K562 has a high edge disruption score.
```{r}
load("../outputs/combined_validation_table_10142020.RData")
examples <- combined_table[which((combined_table$chipseq_k562 != combined_table$chipseq_gm12878) & (combined_table$diff_k562 > 0.5)),c(1:11)]
examples

# write to log file
message <- paste0("Example 1: TF disruptions in K562 and not GM12878: ")
write(c(message), file = logfile, append = TRUE)
write.table(examples[which(examples$id == "RELAENSG00000165449"),], file = logfile, append = TRUE, quote = FALSE, col.names = TRUE, row.names = FALSE, sep = "\t")
write("\n", file = logfile, append = TRUE)

# write to log file
message <- paste0("Example 2: TF disruptions in K562 and not GM12878: ")
write(c(message), file = logfile, append = TRUE)  
write.table(examples[which(examples$id == "ARID3AENSG00000187953"),], file = logfile, append = TRUE, quote = FALSE, col.names = TRUE, row.names = FALSE, sep = "\t")
write("\n", file = logfile, append = TRUE)
```

# Cell line regulatory difference validation
## Approach 1: ChIP-seq (differential binding edge comparison)

Are differences in cell lines based on EGRET edge weights enriched for difference in ChIP-seq binding between cell lines

```{r}
# load the combined data of GM12878 and K562, which overlaps with chipseq data for both.
load("../outputs/combined_validation_table_10142020.RData")

# This calulates a regulatory difference score between the two genotypes as the difference between the edge disruption scores. 
combined_table$diff_celline_egret_g_k <- combined_table$diff_gm12878-combined_table$diff_k562
# calculate difference in ChIP-seq binding networks
combined_table$diff_chip_k_g <- combined_table$chipseq_k562-combined_table$chipseq_gm12878

```

Test if top difference edges between genotypes are enriched in diff chipseq, using a t test and a fisher test
```{r}
# construct the contingency table - test if top 10% of scores are enriched for differential TF binding
colnames(combined_table)[8] <- "panda2"
top10 <- top_frac(combined_table,0.1,diff_celline_egret_g_k)
high_d_chip <- length(which((combined_table$id %in% top10$id) & (combined_table$diff_chip_k_g == 1)))
low_d_chip <- length(which((!(combined_table$id %in% top10$id)) & (combined_table$diff_chip_k_g == 1)))
high_d_not_chip <- length(which((combined_table$id %in% top10$id) & (combined_table$diff_chip_k_g != 1)))
low_d_not_chip <- length(which((!(combined_table$id %in% top10$id)) & (combined_table$diff_chip_k_g != 1)))
contingency_table <- matrix(c(high_d_chip,low_d_chip,high_d_not_chip,low_d_not_chip), nrow = 2,dimnames = list(Diff = c("High", "Low"),DiffChip = c("yes", "no")))
  
#Perform a right-tailed fisher exact test
fishertest <- fisher.test(contingency_table, alternative = "greater")
fishertest$p.value

# write to log file
message <- paste0("Fisher exact test p-value: enrichment of differential TF binding in high regulatory difference edges:")
write(c(message), file = logfile, append = TRUE)  
write(fishertest$p.value, file = logfile, append = TRUE)
write("\n", file = logfile, append = TRUE)
```
```{r}
# This calulates a regulatory difference score between the two genotypes as the difference between the edge disruption scores. 
combined_table$diff_celline_egret_k_g <- combined_table$diff_k562-combined_table$diff_gm12878
# calculate difference in ChIP-seq binding networks
combined_table$diff_chip_g_k <- combined_table$chipseq_gm12878-combined_table$chipseq_k562

```

Test if top difference edges between genotypes are enriched in diff chipseq, using a t test and a fisher test
```{r}
# construct the contingency table - test if top 10% of scores are enriched for differential TF binding
colnames(combined_table)[8] <- "panda2"
top10 <- top_frac(combined_table,0.1,diff_celline_egret_k_g)
high_d_chip <- length(which((combined_table$id %in% top10$id) & (combined_table$diff_chip_g_k == 1)))
low_d_chip <- length(which((!(combined_table$id %in% top10$id)) & (combined_table$diff_chip_g_k == 1)))
high_d_not_chip <- length(which((combined_table$id %in% top10$id) & (combined_table$diff_chip_g_k != 1)))
low_d_not_chip <- length(which((!(combined_table$id %in% top10$id)) & (combined_table$diff_chip_g_k != 1)))
contingency_table <- matrix(c(high_d_chip,low_d_chip,high_d_not_chip,low_d_not_chip), nrow = 2,dimnames = list(Diff = c("High", "Low"),DiffChip = c("yes", "no")))
  
#Perform a right-tailed fisher exact test
fishertest <- fisher.test(contingency_table, alternative = "greater")
fishertest$p.value


# write to log file
message <- paste0("Fisher exact test p-value: enrichment of differential TF binding in high regulatory difference edges:")
write(c(message), file = logfile, append = TRUE)  
write(fishertest$p.value, file = logfile, append = TRUE)
write("\n", file = logfile, append = TRUE)
```

## Approach 2: Allele-specific expression (ASE)  - differential gene regulation comparison

Are genes with high disruption scores enriched for ASE?
```{r}
load("../outputs/edge_table_gm12878_10142020.RData")
load("../outputs/edge_table_k562_10142020.RData")

# combine the GM12878 and K562 network tables
colnames(regnet_edge_table_gm12878) <- c("id","tf","gene","panda","egret_gm12878")
colnames(regnet_edge_table_k562) <- c("id","tf","gene","panda","egret_k562")
regnet_edge_table_gm12878$diff_g <- abs(regnet_edge_table_gm12878$panda-regnet_edge_table_gm12878$egret_gm12878)
regnet_edge_table_k562$diff_k <- abs(regnet_edge_table_k562$panda-regnet_edge_table_k562$egret_k562)
combined_table_all <- join(regnet_edge_table_gm12878,regnet_edge_table_k562,by = c("id","tf","gene"),type="inner")
save(combined_table_all, file = "../outputs/combined_edge_table.RData")

load("../outputs/combined_edge_table.RData")

# This calulates a regulatory difference score between the two genotypes as the difference between the edge disruption scores
combined_table_all$diff_celline_egret <- abs(combined_table_all$diff_k-combined_table_all$diff_g)

# load data for ASE variants within motifs in promoter regions
ase_promotor <- read.table("../ase/ase_mapped_motifs_09012020.txt", header = FALSE)
colnames(ase_promotor) <- c("tf","gene","pval","fdr","chr","pos")
ase_promotor$snpID <- paste0(ase_promotor$chr,"_",ase_promotor$pos)
ase_promotor$id <- paste0(ase_promotor$tf,ase_promotor$gene)
# mark significant ASE variants (FDR < 0.1)
ase_promotor_signif <- ase_promotor[which(ase_promotor$fdr <= 0.1),]

# select the subset our network which maps to ASE variants which were tested by the ASE experiment, so as to have the appropriate background.
combined_table_all_ase_tested <- combined_table_all[which(combined_table_all$id %in% ase_promotor$id),]
combined_table_all_ase_tested$ase_signif <- ifelse(combined_table_all_ase_tested$id %in% ase_promotor_signif$id, 1, 0)

# Calculate the regulatory difference scores for genes
gene_ag <- aggregate(combined_table_all_ase_tested$diff_celline_egret, by = list(combined_table_all_ase_tested$gene), FUN = sum)
colnames(gene_ag) <- c("gene","sum_diff_k_g")
gene_ag$ase_signif <- ifelse(gene_ag$gene %in% ase_promotor_signif$gene, 1, 0)


# construct contingency table - test if top 10% of scores are enriched for ASE
top10 <- top_frac(gene_ag,0.1,sum_diff_k_g)
high_d_ase <- length(which((gene_ag$gene %in% top10$gene) & (gene_ag$ase_signif == 1)))
low_d_ase <- length(which((!(gene_ag$gene %in% top10$gene)) & (gene_ag$ase_signif == 1)))
high_d_not_ase <- length(which((gene_ag$gene %in% top10$gene) & (gene_ag$ase_signif == 0)))
low_d_not_ase <- length(which((!(gene_ag$gene %in% top10$gene)) & (gene_ag$ase_signif == 0)))

contingency_table <- matrix(c(high_d_ase,low_d_ase,high_d_not_ase,low_d_not_ase), nrow = 2,dimnames = list(Diff = c("High", "Low"),ASE = c("yes", "no")))
fishertest <- fisher.test(contingency_table, alternative = "greater")
fishertest$p.value

# write to log file
message <- paste0("Fisher exact test p-value: enrichment of ASE in high regulatory difference genes:")
write(c(message), file = logfile, append = TRUE)  
write(fishertest$p.value, file = logfile, append = TRUE)
write("\n", file = logfile, append = TRUE)

```


# Approach 3: caQTLs  - differential gene regulation comparison 
Are genes with high disruption scores enriched for caQTLs?

```{r}
# load caQTLs in motifs in promotors
caqtl <- read.table("../caqtl/caqtl_mapped_motifs_09092020.txt", header = FALSE)
colnames(caqtl) <- c("query","subject","tf","gene","chrom","pos","pval")
caqtl$snpID <- paste0(caqtl$chrom,"_",caqtl$pos)
caqtl$edge_id <- paste0(caqtl$tf,caqtl$gene)
caqtl$fdr <- qvalue(caqtl$pval)$qvalues
caqtl_signif_fdr <- caqtl[which(caqtl$fdr < 0.1),]

# select the subset our network which maps to caQTL variants which were tested by the caQTL experiment, so as to have the appropriate background.
combined_table_all_caqtl_tested <- combined_table_all[which(combined_table_all$id %in% caqtl$edge_id),]
# mark edges as significant caQTL or not
combined_table_all_caqtl_tested$caqtl_signif <- ifelse(combined_table_all_caqtl_tested$id %in% caqtl_signif_fdr$edge_id, 1, 0)

# Calculate the regulatory difference scores for genes
gene_ag <- aggregate(combined_table_all_caqtl_tested$diff_celline_egret, by = list(combined_table_all_caqtl_tested$gene), FUN = sum)
colnames(gene_ag) <- c("gene","sum_diff_k_g")
gene_ag$cqtl <- ifelse(gene_ag$gene %in% caqtl_signif_fdr$gene, 1, 0)

# construct contingency table - test if top 10% of scores are enriched for ASE
top10 <- top_frac(gene_ag,0.1,sum_diff_k_g)
high_d_caqtl <- length(which((gene_ag$gene %in% top10$gene) & (gene_ag$cqtl == 1)))
low_d_caqtl <- length(which((!(gene_ag$gene %in% top10$gene)) & (gene_ag$cqtl == 1)))
high_d_not_caqtl <- length(which((gene_ag$gene %in% top10$gene) & (gene_ag$cqtl == 0)))
low_d_not_caqtl <- length(which((!(gene_ag$gene %in% top10$gene)) & (gene_ag$cqtl == 0)))

contingency_table <- matrix(c(high_d_caqtl,low_d_caqtl,high_d_not_caqtl,low_d_not_caqtl), nrow = 2,dimnames = list(Diff = c("High", "Low"),caQTL = c("yes", "no")))
fishertest <- fisher.test(contingency_table, alternative = "greater")
fishertest$p.value

# write to log file
message <- paste0("Fisher exact test p-value: enrichment of caQTLs in high regulatory difference genes:")
write(c(message), file = logfile, append = TRUE)  
write(fishertest$p.value, file = logfile, append = TRUE)
write("\n", file = logfile, append = TRUE)

```

# Performance (AUC-ROCs) of ChIP-seq prediction
Calculate AUC-ROCs of prediction of ChIP-seq network from EGRET edge scores, for different cutoffs of edge disruption score. Tables indicating the AUC-ROC improvement and its statistical significance are printed.

## K562 performance by edge disruption threshold
```{r,results='hide',fig.show=TRUE}
names <- c("egret_k562")
diff_cols <- c(7)
egret_net_nums <- c(5)
panda_net_nums <- c(4)
chipseq_col <- 6
get_roc_for_diff_thresh <- function(egretNum, thresh) {
  diffcolnum <- diff_cols[egretNum]
  egretcol <- egret_net_nums[egretNum]
  panda_col <- panda_net_nums[egretNum]
  
  top_diff_table <- validation_table_k562[(validation_table_k562[,c(diffcolnum)]>thresh),c(panda_col,egretcol,chipseq_col)]
  scores <- join_scores(top_diff_table[,1],top_diff_table[,2])
  labels <- join_labels(top_diff_table[,3],top_diff_table[,3])
  specs <- paste0(names[egretNum],", thresh: ",thresh)
  
  rocP <- roc(top_diff_table$chipseq, top_diff_table[,1], auc = TRUE, direction = "<")
  rocE <- roc(top_diff_table$chipseq, top_diff_table[,2],auc = TRUE, direction = "<")
  testD <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
  pandaAUC <- testD$estimate["AUC of roc1"]
  egretAUC <- testD$estimate["AUC of roc2"]
  pvalD <- testD$p.value
  print(paste(egretNum,thresh,pvalD))
  
  curve_data <- mmdata(scores, labels,modnames = c("panda", specs), dsids = c(1, 2))
  curves <- evalmod(curve_data)
  nn <- attr(curves$rocs,"data_info")$nn[1]
  np <- attr(curves$rocs,"data_info")$np[1]
  result <- c(egretNum, names[egretNum], thresh, nn, np, pandaAUC, egretAUC, pvalD)
  return(result)
  autoplot(curves)
}

egret_priors <- c(1)
thresholds <- c(0.1, 0.15, 0.2, 0.25, 0.3, 0.35, 0.4, 0.45, 0.5, 0.55, 0.6, 0.65, 0.7, 0.75, 0.8, 0.85, 0.9,1,2,3,4)

egret_auc_pvals<- data.frame()

for (p in egret_priors){
  for (t in thresholds){
    result <- get_roc_for_diff_thresh(p,t)
    df <- data.frame(t(result), stringsAsFactors = FALSE)
    egret_auc_pvals <- rbind(egret_auc_pvals,df)
  }
}

colnames(egret_auc_pvals) <- c("egretNum", "name", "thresh", "nn", "np", "pandaAUC", "egretAUC", "pvalD")
egret_auc_pvals
egret_auc_pvals_table <- egret_auc_pvals[,c(3,4,5)]
egret_auc_pvals_table$diff <- signif((as.numeric(egret_auc_pvals$egretAUC) - as.numeric(egret_auc_pvals$pandaAUC)), digits = 4)
egret_auc_pvals_table$pvalue <- signif(as.numeric(egret_auc_pvals$pvalD), digits = 4)
print(xtable(egret_auc_pvals_table, type = "latex"), file = "../tables/pval_table_extended_thresh_k562_10142020.tex", include.rownames=FALSE)
```

## GM12878 performance by edge disruption threshold
```{r,results='hide',fig.show=TRUE}
names <- c("egret_GM12878")
diff_cols <- c(7)
egret_net_nums <- c(5)
panda_net_nums <- c(4)
chipseq_col <- 6
get_roc_for_diff_thresh <- function(egretNum, thresh) {
  diffcolnum <- diff_cols[egretNum]
  egretcol <- egret_net_nums[egretNum]
  panda_col <- panda_net_nums[egretNum]
  
  top_diff_table <- validation_table_gm12878[(validation_table_gm12878[,c(diffcolnum)]>thresh),c(panda_col,egretcol,chipseq_col)]
  scores <- join_scores(top_diff_table[,1],top_diff_table[,2])
  labels <- join_labels(top_diff_table[,3],top_diff_table[,3])
  specs <- paste0(names[egretNum],", thresh: ",thresh)
  
  rocP <- roc(top_diff_table$chipseq, top_diff_table[,1], auc = TRUE, direction = "<")
  rocE <- roc(top_diff_table$chipseq, top_diff_table[,2],auc = TRUE, direction = "<")
  test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
  pandaAUC <- test$estimate["AUC of roc1"]
  egretAUC <- test$estimate["AUC of roc2"]
  pval <- test$p.value
  print(paste(egretNum,thresh,pval))
  
  curve_data <- mmdata(scores, labels,modnames = c("panda", specs), dsids = c(1, 2))
  curves <- evalmod(curve_data)
  nn <- attr(curves$rocs,"data_info")$nn[1]
  np <- attr(curves$rocs,"data_info")$np[1]
  result <- c(egretNum, names[egretNum], thresh, nn, np, pandaAUC, egretAUC, pval)
  return(result)
  autoplot(curves)
}

egret_priors <- c(1)
thresholds <- c(0.1, 0.15, 0.2, 0.25, 0.3, 0.35, 0.4, 0.45, 0.5, 0.55, 0.6, 0.65, 0.7, 0.75, 0.8, 0.85, 0.9,1,2,3,4)

egret_auc_pvals<- data.frame()

for (p in egret_priors){
  for (t in thresholds){
    result <- get_roc_for_diff_thresh(p,t)
    df <- data.frame(t(result), stringsAsFactors = FALSE)
    egret_auc_pvals <- rbind(egret_auc_pvals,df)
  }
}

colnames(egret_auc_pvals) <- c("egretNum", "name", "thresh", "nn", "np", "pandaAUC", "egretAUC", "pvalD")
egret_auc_pvals
egret_auc_pvals_table <- egret_auc_pvals[,c(3,4,5)]
egret_auc_pvals_table$diff <- signif((as.numeric(egret_auc_pvals$egretAUC) - as.numeric(egret_auc_pvals$pandaAUC)), digits = 4)
egret_auc_pvals_table$pvalue <- signif(as.numeric(egret_auc_pvals$pvalD), digits = 4)
print(xtable(egret_auc_pvals_table, type = "latex"), file = "../tables/pval_table_extended_thresh_gm12878_10142020.tex", include.rownames=FALSE)
```


# Analysis of edited prior edges

First print the total number of prior edges in the naive prior:

```{r}
edge_count <- system("wc -l /home/ubuntu/EGRET/final_egret_v1/outputs/priors_EGRET_allQBiCModels_gm12878_10142020.txt",intern = TRUE)

# write to log file
message <- paste0("Total number of edges in naive motif prior:")
write(c(message), file = logfile, append = TRUE)  
write(edge_count, file = logfile, append = TRUE)
write("\n", file = logfile, append = TRUE)
```

Plot the distributions of prior modifications - the modifications made to motif prior edges which were penalized because of variant disruptions.

```{r}
# load gm12878 edited prior edges
edited_gm12878 <- read.table("../outputs/edge_modifications_EGRET_allQBiCModels_gm12878_10142020.txt", header = FALSE)
# dimension shows number of edited prior edges - each row is an edited edge
dim(edited_gm12878)

# write to log file
message <- paste0("Total number of edited prior edges in GM12878:")
write(c(message), file = logfile, append = TRUE)  
write(dim(edited_gm12878)[1], file = logfile, append = TRUE)
write("\n", file = logfile, append = TRUE)

colnames(edited_gm12878) <- c("tf","gene","id","mod")

p1 <- ggplot(edited_gm12878, aes(x=mod)) + geom_density(fill=pallet[3], col = pallet[3]) + theme_bw() + scale_color_manual(values=pallet) + scale_fill_manual(values=pallet) +labs(title="GM12878", x ="Prior modification qAB", y = "Density")
p1 <- annotate_figure(p1,fig.lab = "A", fig.lab.face = "plain", fig.lab.pos = "top.left", fig.lab.size = 14)

edited_gm12878_chip <- edited_gm12878[which(edited_gm12878$tf %in% chipseq_regnet_gm12878$tf),]
edited_gm12878_chip$chipseq <- ifelse(edited_gm12878_chip$id %in% chipseq_regnet_gm12878$id, 1, 0)

# count number of 1's
dim(edited_gm12878_chip[edited_gm12878_chip$chipseq == 1,])
dim(edited_gm12878_chip[edited_gm12878_chip$chipseq == 0,])
```


```{r}
# load k562 edited prior edges
edited_k562 <- read.table("../outputs/edge_modifications_EGRET_allQBiCModels_k562_10142020.txt", header = FALSE)
# dimension shows number of edited prior edges - each row is an edited edge
dim(edited_k562)

# write to log file
message <- paste0("Total number of edited prior edges in K562:")
write(c(message), file = logfile, append = TRUE)  
write(dim(edited_k562)[1], file = logfile, append = TRUE)
write("\n", file = logfile, append = TRUE)

colnames(edited_k562) <- c("tf","gene","id","mod")

p2 <- ggplot(edited_k562, aes(x=mod)) + geom_density(fill=pallet[2], col = pallet[2]) + theme_bw() + scale_color_manual(values=pallet) + scale_fill_manual(values=pallet) +labs(title="K562", x ="Prior modification qAB", y = "Density")
p2 <- annotate_figure(p2,fig.lab = "A", fig.lab.face = "plain", fig.lab.pos = "top.left", fig.lab.size = 14)
p3 <- grid.arrange(p1 + theme(legend.position="none"), p2 + theme(legend.position="none"), nrow=1, widths = c(4,4))
#p3

p4 <- annotate_figure(p3,top = text_grob("Prior modification distributions", color = "black", face = "bold", size = 14))
pdf("../figures/modification_distribution_allQBiCModels.pdf",width = 6, height = 4)
plot(p4)
dev.off()

edited_k562_chip <- edited_k562[which(edited_k562$tf %in% chipseq_regnet_k562$tf),]
edited_k562_chip$chipseq <- ifelse(edited_k562_chip$id %in% chipseq_regnet_k562$id, 1, 0)

# count number of 1's
dim(edited_k562_chip[edited_k562_chip$chipseq == 1,])
dim(edited_k562_chip[edited_k562_chip$chipseq == 0,])
```

