---
title: "Contribution of different datatypes"
author: "Deborah Weighill"
output:
  html_document:
    df_print: paged
    toc: yes
  html_notebook:
    df_print: paged
    theme: cosmo
    toc: yes
---
AUC for ChIP-seq prediction when leaving out data types from EGRET modifiers.

# Load libraries
```{r}
library(reshape2)
library(rtracklayer)
library(tidyr)
library(ggplot2)
library(pROC)
library(reshape)
library(PRROC)
library(dplyr)
library(GenomicRanges)
library(precrec)

writeLines(capture.output(sessionInfo()), "egret_datatype_contribution.sessionInfo")
```


# Make Panda/EGRET edge table

Load all of the PANDA/EGRET regnets

```{r echo=FALSE,results='hide'}

# panda"
load("../outputs/EGRET_allQBiCModels_gm12878_10142020_panda.RData")
net <- melt(regnet)
colnames(net) = c("tf","gene","score")
net$id <- paste0(net$tf,net$gene)

regnet_edge_table <- data.frame(net$id)
colnames(regnet_edge_table) <- c("id")
regnet_edge_table$tf <- as.vector(net$tf)
regnet_edge_table$gene <- as.vector(net$gene)
regnet_edge_table$panda <- as.vector(net$score)

#list of the file names of the rest of the networks
files <- c("finalEGRET_v1_noExpr_noPPI_gm12878_egret.RData",
           "finalEGRET_v1_noExpr_noPPI_gm12878_panda.RData",
"finalEGRET_v1_onlyQTL_gm12878_egret.RData",
"finalEgret_v1_onlyQbic_gm12878_egret.RData",
"finalEGRET_v1_noQbic_gm12878_egret.RData",
"EGRET_allQBiCModels_gm12878_10142020_egret.RData")

#nicknames of the networks
names <- c("noExp_noPPI","panda_noExp_noPPI","onlyQTL","onlyQbic","noQBiC","allDataTypes")

for (p in c(1:6)) {
  filename <- paste0("../outputs/",files[p])
  netname <- names[p]
  print(netname)
  load(filename)
  net <- melt(regnet)
  colnames(net) = c("tf","gene","score")
  net$id <- paste0(net$tf,net$gene)
  if(all.equal(as.vector(regnet_edge_table$id),as.vector(net$id))){
    print(p)
    regnet_edge_table$tempID <- as.vector(net$score)
    #print(colnames(regnet_edge_table))
    colnames(regnet_edge_table)[colnames(regnet_edge_table) == "tempID"] <- netname
  }
}
save(regnet_edge_table, file = "../outputs/egret_table_different_data_types.RData")
load("../outputs/egret_table_different_data_types.RData")
```

# Load ChIP-seq data
We load the chipseq data and make a ground truth regulatory network for validation.

```{r echo=FALSE,results='hide'}
# read in the gene annotation file with gene ranges.
genes <- read.table("../annotation/ensembl_genes_from_ucsc.txt", header = TRUE,sep = "\t")
genes$promotorLeft <- ifelse(genes$strand == "+", (genes$txStart + 1 - 750), (genes$txEnd-250))
genes$promotorRight <- ifelse(genes$strand == "+", (genes$txStart + 1 + 250), (genes$txEnd+750))
gr_genes_promotor <- GRanges(seqnames = genes$chrom, ranges = IRanges(start = genes$promotorLeft, end = genes$promotorRight), strand = NULL, mcols = genes[,13:15])

# GM12878 chipseq data
gr_chipseq_gm12878 <- import("../chipseq/GM12878_remap2018_all_macs2_hg19_v1_2.bed")
overlaps_gm12878 <- data.frame(findOverlaps(gr_chipseq_gm12878, gr_genes_promotor))
overlaps_gm12878$tf_name <- gr_chipseq_gm12878$name[overlaps_gm12878$queryHits]
overlaps_gm12878$gene_name <- gr_genes_promotor$mcols.name2[overlaps_gm12878$subjectHits]
validation_regnet_gm12878 <- separate(overlaps_gm12878, tf_name, c("exp", "tf", "cell line"), "\\.")
chipseq_regnet_gm12878 <- distinct(validation_regnet_gm12878[,c("tf","gene_name")])
chipseq_regnet_gm12878$id <- paste0(chipseq_regnet_gm12878$tf,chipseq_regnet_gm12878$gene)
```

# Overlap networks with ChIP-seq data
```{r echo=FALSE,results='hide'}
filtered_net_table_gm12878 <- regnet_edge_table[(regnet_edge_table$tf %in% chipseq_regnet_gm12878$tf) ,]
filtered_net_table_gm12878$chipseq <- ifelse(filtered_net_table_gm12878$id %in% chipseq_regnet_gm12878$id,1,0)
```

# Validate highest difference edges with chipseq
## GM12878
```{r echo=FALSE,results='hide'}
filtered_net_table_gm12878$diff_noExpPPI <- abs(filtered_net_table_gm12878$noExp_noPPI - filtered_net_table_gm12878$panda_noExp_noPPI)
filtered_net_table_gm12878$diff_onlyQbic <- abs(filtered_net_table_gm12878$onlyQbic - filtered_net_table_gm12878$panda)
filtered_net_table_gm12878$diff_egret_onlyQTL <- abs(filtered_net_table_gm12878$onlyQTL - filtered_net_table_gm12878$panda)
filtered_net_table_gm12878$diff_noQbic <- abs(filtered_net_table_gm12878$noQBiC - filtered_net_table_gm12878$panda)
filtered_net_table_gm12878$diff_complete <- abs(filtered_net_table_gm12878$allDataTypes - filtered_net_table_gm12878$panda)


validation_table_gm12878 <- filtered_net_table_gm12878
save(validation_table_gm12878, file = "../outputs/validation_table_egret_datatypes_gm12878.RData")
load("../outputs/validation_table_egret_datatypes_gm12878.RData")
```


# Validate networks against ChIP-seq

Plot ROC/PR curves for each PANDA and EGRET nework, as well as table of AUCs and PRs
```{r}
# GM12878 validation
for (i in c(5,10)){
  print(i)
  specs <- colnames(filtered_net_table_gm12878)[i]
  print(specs)
  curve <- evalmod(scores = filtered_net_table_gm12878[,i], labels = as.factor(filtered_net_table_gm12878$chipseq),modnames = c(specs))
  autoplot(curve)
  aucs <- precrec::auc(curve)
  print(aucs)
}
```

Calculate the percentage improvement in the ROC of EGRET with PPI and expression data compared to that of EGRET without PPI and expression data.
```{r}
# percentage improvement= (abs(egret_all_datatypes_ROC - egret_noPPI_noExp_ROC)/egret_noPPI_noExp_ROC))*100 
perc_improvement <- (abs(0.5749349-0.6031124)/0.5749349)*100
perc_improvement
```

# Validate likely variant-impacted edges

```{r,results='hide',fig.show=TRUE}

names <- c("onlyQTL","onlyQbic","noQBiC","allDataTypes")
diff_cols <- c(13,14,15,16)
egret_net_nums <- c(7,8,9,10)
panda_net_nums <- c(4,4,4,4)
chipseq_col <- 11
get_roc_for_diff_thresh <- function(egretNum) {
  diffcolnum <- diff_cols[egretNum]
  egretcol <- egret_net_nums[egretNum]
  panda_col <- panda_net_nums[egretNum]
  
  top_diff_table <- validation_table_gm12878[(validation_table_gm12878[,c(diffcolnum)]>0.35),c(panda_col,egretcol,chipseq_col)]
  scores <- join_scores(top_diff_table[,1],top_diff_table[,2])
  labels <- join_labels(top_diff_table[,3],top_diff_table[,3])
  specs <- paste0(names[egretNum],", thresh: ",0.35)
  
  rocP <- roc(top_diff_table$chipseq, top_diff_table[,1], auc = TRUE, direction = "<")
  rocE <- roc(top_diff_table$chipseq, top_diff_table[,2],auc = TRUE, direction = "<")
  test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
  pandaAUC <- test$estimate["AUC of roc1"]
  egretAUC <- test$estimate["AUC of roc2"]
  pval <- test$p.value
  print(paste(egretNum,0.35,pval))
  
  curve_data <- mmdata(scores, labels,modnames = c("panda", specs), dsids = c(1, 2))
  curves <- evalmod(curve_data)
  nn <- attr(curves$rocs,"data_info")$nn[1]
  np <- attr(curves$rocs,"data_info")$np[1]
  result <- c(egretNum, names[egretNum], 0.35, nn, np, pandaAUC, egretAUC, pval)
  return(result)
  autoplot(curves)
}
egret_priors <- c(1:4)
egret_auc_pvals_gm12878 <- data.frame()

for (p in egret_priors){
    result <- get_roc_for_diff_thresh(p)
    df <- data.frame(t(result), stringsAsFactors = FALSE)
    egret_auc_pvals_gm12878 <- rbind(egret_auc_pvals_gm12878,df)
  }

colnames(egret_auc_pvals_gm12878) <- c("egretNum", "name", "thresh", "nn", "np", "pandaAUC", "egretAUC", "pval")
egret_auc_pvals_gm12878

egret_auc_pvals_gm12878$improvement <-((as.numeric(egret_auc_pvals_gm12878$egretAUC)-as.numeric(egret_auc_pvals_gm12878$pandaAUC))/as.numeric(egret_auc_pvals_gm12878$pandaAUC))*100
egret_auc_pvals_gm12878
```


```{r}
pallet <- c("#FFBF00",  "#870E75"  ,  "#0BB19F" )

p <- ggplot(data = egret_auc_pvals_gm12878, aes(x = reorder(name,-as.numeric(improvement)), y = improvement)) + geom_bar(stat="identity", fill = pallet[3]) + theme_bw()  +  theme(text = element_text(size=12))  +  geom_text(aes(label=signif(improvement,3),vjust = -1) )+  scale_x_discrete("EGRET prior modification",labels=c("E - |qAB|","E - |AB|","E - |B|","E - |qA|")) +  labs(y = "% improvement in AUC-ROC", title = "Contribution of EGRET data types") + ylim(-5,42)

pdf("../figures/egret_data_type_contribution.pdf",width = 4, height = 3)
plot(p)
dev.off()
```



