---
title: "Plots for GM12878 and K562"
author: "Deborah Weighill"
date: "August 5, 2020"
output:
  html_notebook:
    toc: true
    theme: cosmo
    df_print: paged
---

# Load libraries, preliminaries
```{r}
library(xtable)
library(jcolors)
library(precrec)
library(gridExtra)
library(grid)
library(tidyr)
library(plyr)
library(ggpubr)
library(dplyr)
library(pROC)
library(viridis)
library(lemon)
library(GenomicRanges)
library(ggplot2)
library(reshape2)
library(rtracklayer)

# set color pallet
pallet <- c("#FFBF00",  "#870E75"  ,  "#0BB19F" )
chipseq_gm12878 <- "../chipseq/GM12878_remap2018_all_macs2_hg19_v1_2.bed"
chipseq_k562 <- "../chipseq/K562_remap2018_all_macs2_hg19_v1_2.bed"
```

# Create validation table for GM12878 - "human only version"
QBiC only using human models.


## Load GM12878 networks into table
```{r echo=FALSE,results='hide'}
load("../outputs/finalEgret_v1_na12878_eQTLs_06182020_panda.RData")
net <- melt(regnet)
colnames(net) = c("tf","gene","score")
net$id <- paste0(net$tf,net$gene)

regnet_edge_table_gm12878 <- data.frame(net$id)
colnames(regnet_edge_table_gm12878) <- c("id")
regnet_edge_table_gm12878$tf <- as.vector(net$tf)
regnet_edge_table_gm12878$gene <- as.vector(net$gene)
regnet_edge_table_gm12878$panda <- as.vector(net$score)

#list of the file names of the rest of the networks
files <- c("finalEgret_v1_na12878_eQTLs_06182020_egret.RData")

#nicknames of the networks
names <- c("egret_GM12878")

for (p in c(1:1)) {
  filename <- paste0("../outputs/",files[p])
  netname <- names[p]
  print(netname)
  load(filename)
  net <- melt(regnet)
  colnames(net) = c("tf","gene","score")
  net$id <- paste0(net$tf,net$gene)
  if(all.equal(as.vector(regnet_edge_table_gm12878$id),as.vector(net$id))){
    print(p)
    regnet_edge_table_gm12878$tempID <- as.vector(net$score)
    colnames(regnet_edge_table_gm12878)[colnames(regnet_edge_table_gm12878) == "tempID"] <- netname
  }
}
filename <- paste0("../outputs/edge_table_gm12878_08052020.RData")
save(regnet_edge_table_gm12878, file = filename)
```

## Load ChIP-seq data for GM12878
We load the chipseq data and make a ground truth regulatory network for validation.

```{r echo=FALSE,results='hide'}
# read in the gene annotation file with gene ranges.
genes <- read.table("../annotation/ensembl_genes_from_ucsc.txt", header = TRUE,sep = "\t")
genes$promotorLeft <- ifelse(genes$strand == "+", (genes$txStart + 1 - 750), (genes$txEnd-250))
genes$promotorRight <- ifelse(genes$strand == "+", (genes$txStart + 1 + 250), (genes$txEnd+750))
gr_genes_promotor <- GRanges(seqnames = genes$chrom, ranges = IRanges(start = genes$promotorLeft, end = genes$promotorRight), strand = NULL, mcols = genes[,13:15])

# chipseq chipseq data
gr_chipseq_gm12878 <- import(chipseq_gm12878)
overlaps_gm12878 <- data.frame(findOverlaps(gr_chipseq_gm12878, gr_genes_promotor))
overlaps_gm12878$tf_name <- gr_chipseq_gm12878$name[overlaps_gm12878$queryHits]
overlaps_gm12878$gene_name <- gr_genes_promotor$mcols.name2[overlaps_gm12878$subjectHits]
validation_regnet_gm12878 <- separate(overlaps_gm12878, tf_name, c("exp", "tf", "cell line"), "\\.")
chipseq_regnet_gm12878 <- distinct(validation_regnet_gm12878[,c("tf","gene_name")])
chipseq_regnet_gm12878$id <- paste0(chipseq_regnet_gm12878$tf,chipseq_regnet_gm12878$gene)
```

## Overlap egret/panda networks with ChIP-seq data and calcualte differences
We now filter down the panda/egret networks to only include those TFs which are in the ChIP-seq data (our validation set).
```{r echo=FALSE,results='hide'}
filtered_net_table_gm12878 <- regnet_edge_table_gm12878[(regnet_edge_table_gm12878$tf %in% chipseq_regnet_gm12878$tf) ,]
filtered_net_table_gm12878$chipseq <- ifelse(filtered_net_table_gm12878$id %in% chipseq_regnet_gm12878$id,1,0)

filtered_net_table_gm12878$diff_egret <- abs(filtered_net_table_gm12878$egret_GM12878 - filtered_net_table_gm12878$panda)

validation_table_gm12878 <- filtered_net_table_gm12878
filename <- paste0("../outputs/validation_table_gm12878_08052020.RData")
save(validation_table_gm12878, file = filename)
```

# Create validation table for K562 - "human only version"

## Load K562 networks into table

Load all of the PANDA/EGRET regnets

```{r echo=FALSE,results='hide'}
# panda - note - the panda network is the same file for K562 and GM12878 as there is no genotype information
load("../outputs/finalEgret_v1_na12878_eQTLs_06182020_panda.RData")
net <- melt(regnet)
colnames(net) = c("tf","gene","score")
net$id <- paste0(net$tf,net$gene)

regnet_edge_table_k562 <- data.frame(net$id)
colnames(regnet_edge_table_k562) <- c("id")
regnet_edge_table_k562$tf <- as.vector(net$tf)
regnet_edge_table_k562$gene <- as.vector(net$gene)
regnet_edge_table_k562$panda <- as.vector(net$score)

#list of the file names of the rest of the networks
files <- c("finalEgret_v1_k562_eQTL_06182020_egret.RData")

#nicknames of the networks
names <- c("egret_k562")

for (p in c(1:1)) {
  filename <- paste0("../outputs/",files[p])
  netname <- names[p]
  print(netname)
  load(filename)
  net <- melt(regnet)
  colnames(net) = c("tf","gene","score")
  net$id <- paste0(net$tf,net$gene)
  if(all.equal(as.vector(regnet_edge_table_k562$id),as.vector(net$id))){
    print(p)
    regnet_edge_table_k562$tempID <- as.vector(net$score)
    colnames(regnet_edge_table_k562)[colnames(regnet_edge_table_k562) == "tempID"] <- netname
  }
}
filename <- paste0("../outputs/edge_table_k562_08052020.RData")
save(regnet_edge_table_k562, file = filename)
```

## Load ChIP-seq data for K562
We load the chipseq data and make a ground truth regulatory network for validation.

```{r echo=FALSE,results='hide'}
gr_chipseq_k562 <- import(chipseq_k562)
overlaps_k562 <- data.frame(findOverlaps(gr_chipseq_k562, gr_genes_promotor))
overlaps_k562$tf_name <- gr_chipseq_k562$name[overlaps_k562$queryHits]
overlaps_k562$gene_name <- gr_genes_promotor$mcols.name2[overlaps_k562$subjectHits]
validation_regnet_k562 <- separate(overlaps_k562, tf_name, c("exp", "tf", "cell line"), "\\.")
chipseq_regnet_k562 <- distinct(validation_regnet_k562[,c("tf","gene_name")])
chipseq_regnet_k562$id <- paste0(chipseq_regnet_k562$tf,chipseq_regnet_k562$gene)
```

## Overlap egret/panda networks with ChIP-seq data and calcualte differences
We now filter down the panda/egret networks to only include those TFs which are in the ChIP-seq data (our validation set).
```{r echo=FALSE,results='hide'}
filtered_net_table_k562 <- regnet_edge_table_k562[(regnet_edge_table_k562$tf %in% chipseq_regnet_k562$tf) ,]
filtered_net_table_k562$chipseq <- ifelse(filtered_net_table_k562$id %in% chipseq_regnet_k562$id,1,0)
filtered_net_table_k562$diff_egret <- abs(filtered_net_table_k562$egret_k562 - filtered_net_table_k562$panda)

validation_table_k562 <- filtered_net_table_k562
filename <- paste0("../outputs/validation_table_k562_08052020.RData")
save(validation_table_k562, file = filename)
```


# Calculate performance curves and AUCs, pvals and print to latex table

## K562 performance by edge disruption threshold
```{r,results='hide',fig.show=TRUE}
names <- c("egret_k562")
diff_cols <- c(7)
egret_net_nums <- c(5)
panda_net_nums <- c(4)
chipseq_col <- 6
get_roc_for_diff_thresh <- function(egretNum, thresh) {
  diffcolnum <- diff_cols[egretNum]
  egretcol <- egret_net_nums[egretNum]
  panda_col <- panda_net_nums[egretNum]
  
  top_diff_table <- validation_table_k562[(validation_table_k562[,c(diffcolnum)]>thresh),c(panda_col,egretcol,chipseq_col)]
  scores <- join_scores(top_diff_table[,1],top_diff_table[,2])
  labels <- join_labels(top_diff_table[,3],top_diff_table[,3])
  specs <- paste0(names[egretNum],", thresh: ",thresh)
  
  rocP <- roc(top_diff_table$chipseq, top_diff_table[,1], auc = TRUE, direction = "<")
  rocE <- roc(top_diff_table$chipseq, top_diff_table[,2],auc = TRUE, direction = "<")
  testD <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
  pandaAUC <- testD$estimate["AUC of roc1"]
  egretAUC <- testD$estimate["AUC of roc2"]
  pvalD <- testD$p.value
  print(paste(egretNum,thresh,pvalD))
  
  curve_data <- mmdata(scores, labels,modnames = c("panda", specs), dsids = c(1, 2))
  curves <- evalmod(curve_data)
  nn <- attr(curves$rocs,"data_info")$nn[1]
  np <- attr(curves$rocs,"data_info")$np[1]
  result <- c(egretNum, names[egretNum], thresh, nn, np, pandaAUC, egretAUC, pvalD)
  return(result)
  autoplot(curves)
}

egret_priors <- c(1)
thresholds <- c(0.1, 0.15, 0.2, 0.25, 0.3, 0.35, 0.4, 0.45, 0.5, 0.55, 0.6, 0.65, 0.7, 0.75, 0.8, 0.85, 0.9,1,2,3,4)

egret_auc_pvals<- data.frame()

for (p in egret_priors){
  for (t in thresholds){
    result <- get_roc_for_diff_thresh(p,t)
    df <- data.frame(t(result), stringsAsFactors = FALSE)
    egret_auc_pvals <- rbind(egret_auc_pvals,df)
  }
}

colnames(egret_auc_pvals) <- c("egretNum", "name", "thresh", "nn", "np", "pandaAUC", "egretAUC", "pvalD")
egret_auc_pvals
egret_auc_pvals_table <- egret_auc_pvals[,c(3,4,5)]
egret_auc_pvals_table$pandaAUC <- signif(as.numeric(egret_auc_pvals$pandaAUC, digits = 4))
egret_auc_pvals_table$egretAUC <- signif(as.numeric(egret_auc_pvals$egretAUC, digits = 4))
egret_auc_pvals_table$pvalue <- signif(as.numeric(egret_auc_pvals$pvalD, digits = 4))
print(xtable(egret_auc_pvals_table, type = "latex"), file = "../tables/pval_table_allHumanModels_thresh_k562.tex", include.rownames=FALSE)
```

## K562 performance by "top n" disrupted edges
```{r,results='hide',fig.show=TRUE}
names <- c("egret_k562")
diff_cols <- c(7)
egret_net_nums <- c(5)
panda_net_nums <- c(4)
chipseq_col <- 6
get_roc_for_top_x <- function(egretNum, x) {
  egretcol <- egret_net_nums[egretNum]
  panda_col <- panda_net_nums[egretNum]
  
  top_x_table <- top_n(validation_table_k562,x,diff_k562)[,c(panda_col,egretcol,chipseq_col)] 
  scores <- join_scores(top_x_table[,1],top_x_table[,2])
  labels <- join_labels(top_x_table[,3],top_x_table[,3])
  specs <- paste0(names[egretNum],", top ",x)
  
  rocP <- roc(top_x_table$chipseq, top_x_table[,1], auc = TRUE, direction = "<")
  rocE <- roc(top_x_table$chipseq, top_x_table[,2],auc = TRUE, direction = "<")
  test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
  pandaAUC <- test$estimate["AUC of roc1"]
  egretAUC <- test$estimate["AUC of roc2"]
  pval <- test$p.value
  print(paste(egretNum,x,pval))
  
  curve_data <- mmdata(scores, labels,modnames = c("panda", specs), dsids = c(1, 2))
  curves <- evalmod(curve_data)
  nn <- attr(curves$rocs,"data_info")$nn[1]
  np <- attr(curves$rocs,"data_info")$np[1]
  result <- c(egretNum, names[egretNum], x, nn, np, pandaAUC, egretAUC, pval)
  return(result)
  autoplot(curves)
}

egret_priors <- c(1)
numEdges <- c(10,15,20,25,30,35,40,45,50)

egret_auc_pvals_topn<- data.frame()

for (p in egret_priors){
  for (x in numEdges){
    result <- get_roc_for_top_x(p,x)
    df <- data.frame(t(result), stringsAsFactors = FALSE)
    egret_auc_pvals_topn <- rbind(egret_auc_pvals_topn,df)
  }
}
colnames(egret_auc_pvals_topn) <- c("egretNum", "name", "thresh", "nn", "np", "pandaAUC", "egretAUC", "pvalD")
egret_auc_pvals_topn
egret_auc_pvals_table <- egret_auc_pvals_topn[,c(3,4,5)]
egret_auc_pvals_table$pandaAUC <- signif(as.numeric(egret_auc_pvals_topn$pandaAUC), digits = 5)
egret_auc_pvals_table$egretAUC <- signif(as.numeric(egret_auc_pvals_topn$egretAUC), digits = 5)
egret_auc_pvals_table$pvalue <- signif(as.numeric(egret_auc_pvals_topn$pval), digits = 5)
print(xtable(egret_auc_pvals_table, type = "latex", digits = 5), file= "../tables/pval_table_allHumanModels_topn_k562.tex",include.rownames=FALSE)
```


## GM12878 performance by edge disruption threshold
```{r,results='hide',fig.show=TRUE}
names <- c("egret_GM12878")
diff_cols <- c(7)
egret_net_nums <- c(5)
panda_net_nums <- c(4)
chipseq_col <- 6
get_roc_for_diff_thresh <- function(egretNum, thresh) {
  diffcolnum <- diff_cols[egretNum]
  egretcol <- egret_net_nums[egretNum]
  panda_col <- panda_net_nums[egretNum]
  
  top_diff_table <- validation_table_gm12878[(validation_table_gm12878[,c(diffcolnum)]>thresh),c(panda_col,egretcol,chipseq_col)]
  scores <- join_scores(top_diff_table[,1],top_diff_table[,2])
  labels <- join_labels(top_diff_table[,3],top_diff_table[,3])
  specs <- paste0(names[egretNum],", thresh: ",thresh)
  
  rocP <- roc(top_diff_table$chipseq, top_diff_table[,1], auc = TRUE, direction = "<")
  rocE <- roc(top_diff_table$chipseq, top_diff_table[,2],auc = TRUE, direction = "<")
  test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
  pandaAUC <- test$estimate["AUC of roc1"]
  egretAUC <- test$estimate["AUC of roc2"]
  pval <- test$p.value
  print(paste(egretNum,thresh,pval))
  
  curve_data <- mmdata(scores, labels,modnames = c("panda", specs), dsids = c(1, 2))
  curves <- evalmod(curve_data)
  nn <- attr(curves$rocs,"data_info")$nn[1]
  np <- attr(curves$rocs,"data_info")$np[1]
  result <- c(egretNum, names[egretNum], thresh, nn, np, pandaAUC, egretAUC, pval)
  return(result)
  autoplot(curves)
}

egret_priors <- c(1)
thresholds <- c(0.1, 0.15, 0.2, 0.25, 0.3, 0.35, 0.4, 0.45, 0.5, 0.55, 0.6, 0.65, 0.7, 0.75, 0.8, 0.85, 0.9, 1)

egret_auc_pvals<- data.frame()

for (p in egret_priors){
  for (t in thresholds){
    result <- get_roc_for_diff_thresh(p,t)
    df <- data.frame(t(result), stringsAsFactors = FALSE)
    egret_auc_pvals <- rbind(egret_auc_pvals,df)
  }
}

colnames(egret_auc_pvals) <- c("egretNum", "name", "thresh", "nn", "np", "pandaAUC", "egretAUC", "pval")
egret_auc_pvals
print(xtable(egret_auc_pvals[,c(3:8)], type = "latex"), file = "../tables/pval_table_allHumanModels_thresh_gm12878.tex",include.rownames=FALSE)
```

## GM12878 performance by "top n" disrupted edges

```{r,results='hide',fig.show=TRUE}

names <- c("egret_GM12878")
diff_cols <- c(7)
egret_net_nums <- c(5)
panda_net_nums <- c(4)
chipseq_col <- 6
get_roc_for_top_x <- function(egretNum, x) {
  egretcol <- egret_net_nums[egretNum]
  panda_col <- panda_net_nums[egretNum]
  
  top_x_table <- top_n(validation_table_gm12878,x,diff_gm12878)[,c(panda_col,egretcol,chipseq_col)] 
  scores <- join_scores(top_x_table[,1],top_x_table[,2])
  labels <- join_labels(top_x_table[,3],top_x_table[,3])
  specs <- paste0(names[egretNum],", top ",x)
  
  rocP <- roc(top_x_table$chipseq, top_x_table[,1], auc = TRUE, direction = "<")
  rocE <- roc(top_x_table$chipseq, top_x_table[,2],auc = TRUE, direction = "<")
  test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
  pandaAUC <- test$estimate["AUC of roc1"]
  egretAUC <- test$estimate["AUC of roc2"]
  pval <- test$p.value
  print(paste(egretNum,x,pval))
  
  curve_data <- mmdata(scores, labels,modnames = c("panda", specs), dsids = c(1, 2))
  curves <- evalmod(curve_data)
  nn <- attr(curves$rocs,"data_info")$nn[1]
  np <- attr(curves$rocs,"data_info")$np[1]
  result <- c(egretNum, names[egretNum], x, nn, np, pandaAUC, egretAUC, pval)
  return(result)
  autoplot(curves)
}

egret_priors <- c(1)
numEdges <- c(10,15,20,25,30,35,40,45,50)

egret_auc_pvals_topn<- data.frame()

for (p in egret_priors){
  for (x in numEdges){
    result <- get_roc_for_top_x(p,x)
    df <- data.frame(t(result), stringsAsFactors = FALSE)
    egret_auc_pvals_topn <- rbind(egret_auc_pvals_topn,df)
  }
}

colnames(egret_auc_pvals_topn) <- c("egretNum", "name", "topN", "nn", "np", "pandaAUC", "egretAUC", "pval")
egret_auc_pvals_topn
egret_auc_pvals_table <- egret_auc_pvals_topn[,c(3,4,5)]
egret_auc_pvals_table$pandaAUC <- signif(as.numeric(egret_auc_pvals_topn$pandaAUC), digits = 5)
egret_auc_pvals_table$egretAUC <- signif(as.numeric(egret_auc_pvals_topn$egretAUC), digits = 5)
egret_auc_pvals_table$pvalue <- signif(as.numeric(egret_auc_pvals_topn$pval), digits = 5)
print(xtable(egret_auc_pvals_table, type = "latex", digits = 5), file = "../tables/pval_table_allHumanModels_topn_gm12878.tex",include.rownames=FALSE,)
```

# Select examples for visualizing in genome browser

```{r}
colnames(validation_table_gm12878) <- c("id","tf","gene","panda","egret_gm12878","chipseq_gm12878","diff_gm12878")
colnames(validation_table_k562) <- c("id","tf","gene","panda","egret_k562","chipseq_k562","diff_k562")
combined_table <- join(validation_table_gm12878,validation_table_k562,by = c("id","tf","gene"),type="inner")
combined_table[which((combined_table$chipseq_k562 != combined_table$chipseq_gm12878) & (combined_table$diff_k562 > 0.5)),c(1:11)]
```



# Box plots of edge differences for top difference edges
Make a box plot showing the distribution in egret edge weights for top diff edges. Also show distribution of differences, each time grouped by chipseq +/-. Remember, top difference here is not chosen as top difference in chipseq or anything, its chosen as top difference between EGRET and PANDA. We're trying to isolate the effect of the variants.


```{r}
topn_gm12878 <- top_n(validation_table_gm12878, 35, diff_gm12878)
test <- t.test(topn_gm12878[which(topn_gm12878$chipseq_gm12878 == 0),]$diff_gm12878,topn_gm12878[which(topn_gm12878$chipseq_gm12878 == 1),]$diff_gm12878, alternative="greater")


p1 <- ggplot(topn_gm12878, aes(x = as.factor(chipseq_gm12878), y = diff_gm12878, color = as.factor(chipseq_gm12878), group = as.factor(chipseq_gm12878)))+ geom_boxplot() + geom_jitter(position=position_jitter(0.1))+  stat_summary(fun.y=mean, geom="point", shape=18,size=5, color="#E41A1C") + theme_bw() + scale_color_manual(values=pallet)  + theme(legend.position="right") + labs(title="GM12878", x ="ChIP-seq binding", y = "Edge disruption score", color  = "ChIP-seq binding") + annotate("text", x = 1.6, y = 7.5, label = paste0("t-test p= ",signif(test$p.value,4)))

topn_k562 <- top_n(validation_table_k562, 35, diff_k562)

test <- t.test(topn_k562[which(topn_k562$chipseq_k562 == 0),]$diff_k562,topn_k562[which(topn_k562$chipseq_k562 == 1),]$diff_k562, alternative="greater")


p2 <- ggplot(topn_k562, aes(x = as.factor(chipseq_k562), y = diff_k562, color = as.factor(chipseq_k562), group = as.factor(chipseq_k562)))+ geom_boxplot() + geom_jitter(position=position_jitter(0.1))+  stat_summary(fun.y=mean, geom="point", shape=18,size=5, color="#E41A1C") + theme_bw() + scale_color_manual(values=pallet)  + theme(legend.position="right") + labs(title="K562", x ="ChIP-seq binding", y = "Edge disruption score", color  = "ChIP-seq binding") + annotate("text", x = 1.7, y = 10, label = paste0("t-test p= ",signif(test$p.value,4)))

p3 <- grid.arrange(p1 + theme(legend.position="none"), p2 + theme(legend.position="right"), nrow=1, widths = c(3.5,5.9))
p4 <- annotate_figure(p3,top = text_grob("Top 35 edge disruption scores", color = "black", face = "bold", size = 14))

pdf("../figures/top35_diffvalues_boxplots_k562_gm12878.pdf",width = 6, height = 3.5)
plot(p4)
dev.off()
plot(p4)
```



# Performance curves AUCs/AU-PRs for top35 edges
```{r}
pallet <- c("#870E75"  ,  "#0BB19F" )
top_x_table_gm12878 <- top_n(validation_table_gm12878, 35, diff_gm12878)[,c(4,5,6)]
scores <- join_scores(top_x_table_gm12878[,1],top_x_table_gm12878[,2])
labels <- join_labels(top_x_table_gm12878[,3],top_x_table_gm12878[,3])
curve_data <- mmdata(scores, labels,modnames = c("panda", "egret"))
curves <- evalmod(curve_data)
#autoplot(curves)
#curves
#auc(curves)

rocP <- roc(top_x_table_gm12878$chipseq_gm12878, top_x_table_gm12878$panda, auc = TRUE, direction = "<")
rocE <- roc(top_x_table_gm12878$chipseq_gm12878, top_x_table_gm12878$egret_gm12878,auc = TRUE, direction = "<")
test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
pandaAUC <- test$estimate["AUC of roc1"]
egretAUC <- test$estimate["AUC of roc2"]
pval <- round(test$p.value,digits = 4)

pandaROC <- as.data.frame(curves$rocs[[1]]$x)
colnames(pandaROC) <- c("x")
pandaROC$y <- curves$rocs[[1]]$y
pandaROC$Method <- "PANDA"

egretROC <- as.data.frame(curves$rocs[[2]]$x)
colnames(egretROC) <- c("x")
egretROC$y <- curves$rocs[[2]]$y
egretROC$Method <- "EGRET"

roc_data <- rbind(pandaROC, egretROC)
roccurves_gm12878 <- ggplot(roc_data, aes(x=x, y=y, color=Method, group = Method)) + geom_line() +  theme_bw() + scale_color_manual(values=pallet) + xlim(0,1) + ylim(0,1) + geom_abline(intercept = 0, slope = 1, linetype = 2) + labs(title="ROC (GM12878)", x ="1 - Specificity", y = "Sensitivity") + annotate("text", x = 0.3, y = 0.8, label = paste0("Delong p: ",pval)) 


pandaPR <- as.data.frame(curves$prcs[[1]]$x)
colnames(pandaPR) <- c("x")
pandaPR$y <- curves$prcs[[1]]$y
pandaPR$Method <- "PANDA"

egretPR <- as.data.frame(curves$prcs[[2]]$x)
colnames(egretPR) <- c("x")
egretPR$y <- curves$prcs[[2]]$y
egretPR$Method <- "EGRET"

pr_data <- rbind(pandaPR, egretPR)
prcurves_gm12878 <- ggplot(pr_data, aes(x=x, y=y, color=Method, group = Method)) + geom_line() +  theme_bw() + scale_color_manual(values=pallet) + xlim(0,1) + ylim(0,1) + labs(title="PR (GM12878)", x ="Recall", y = "Precision") 
```


```{r}
top_x_table_k562 <- top_n(validation_table_k562, 35, diff_k562)[,c(4,5,6)]
scores <- join_scores(top_x_table_k562[,1],top_x_table_k562[,2])
labels <- join_labels(top_x_table_k562[,3],top_x_table_k562[,3])
curve_data <- mmdata(scores, labels,modnames = c("panda", "egret"))
curves <- evalmod(curve_data)
#autoplot(curves)
#curves
#auc(curves)

rocP <- roc(top_x_table_k562$chipseq_k562, top_x_table_k562$panda, auc = TRUE, direction = "<")
rocE <- roc(top_x_table_k562$chipseq_k562, top_x_table_k562$egret_k562,auc = TRUE, direction = "<")
test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
pandaAUC <- test$estimate["AUC of roc1"]
egretAUC <- test$estimate["AUC of roc2"]
pval <- round(test$p.value,digits = 4)

pandaROC <- as.data.frame(curves$rocs[[1]]$x)
colnames(pandaROC) <- c("x")
pandaROC$y <- curves$rocs[[1]]$y
pandaROC$Method <- "PANDA"

egretROC <- as.data.frame(curves$rocs[[2]]$x)
colnames(egretROC) <- c("x")
egretROC$y <- curves$rocs[[2]]$y
egretROC$Method <- "EGRET"

roc_data <- rbind(pandaROC, egretROC)
roccurves_k562 <- ggplot(roc_data, aes(x=x, y=y, color=Method, group = Method)) + geom_line() +  theme_bw() + scale_color_manual(values=pallet) + xlim(0,1) + ylim(0,1) + geom_abline(intercept = 0, slope = 1, linetype = 2) + labs(title="ROC (K562)", x ="1 - Specificity", y = "Sensitivity") +annotate("text", x = 0.25, y = 0.8, label = paste0("Delong p: ",pval))


pandaPR <- as.data.frame(curves$prcs[[1]]$x)
colnames(pandaPR) <- c("x")
pandaPR$y <- curves$prcs[[1]]$y
pandaPR$Method <- "PANDA"

egretPR <- as.data.frame(curves$prcs[[2]]$x)
colnames(egretPR) <- c("x")
egretPR$y <- curves$prcs[[2]]$y
egretPR$Method <- "EGRET"

pr_data <- rbind(pandaPR, egretPR)
prcurves_k562 <- ggplot(pr_data, aes(x=x, y=y, color=Method, group = Method)) + geom_line() +  theme_bw() + scale_color_manual(values=pallet) + xlim(0,1) + ylim(0,1) + labs(title="PR (K562)", x ="Recall", y = "Precision") 
```

```{r}
p3 <- grid.arrange(roccurves_gm12878 + theme(legend.position="hidden"), prcurves_gm12878 + theme(legend.position="none"),roccurves_k562 + theme(legend.position="hidden"), prcurves_k562 + theme(legend.position="none"), nrow=2, widths = c(4,4),heights = c(4,4))

p4 <- annotate_figure(p3,top = text_grob("Performance curves for top 35 disrupted edges", color = "black", face = "bold", size = 14))
pdf("../figures/performance_curves_top35_humanModels.pdf",width = 6, height = 4)
plot(p4)
dev.off()
```



# Now show that difference scores are predictive of disruption with AUCs
```{r}
pallet <- c("#870E75"  ,  "#0BB19F" )
color <- "#870E75"
#GM12878
top_x_table_gm12878 <- top_n(validation_table_gm12878, 35, diff_gm12878)
top_x_table_gm12878$disrupt <- 1 - top_x_table_gm12878$chipseq
curve <- evalmod(scores = top_x_table_gm12878$diff_gm12878, labels = top_x_table_gm12878$disrupt, posclass = 1)
autoplot(curve)

roc <- as.data.frame(curve$rocs[[1]]$x)
colnames(roc) <- c("x")
roc$y <- curve$rocs[[1]]$y

pr <- as.data.frame(curve$prcs[[1]]$x)
colnames(pr) <- c("x")
pr$y <- curve$prcs[[1]]$y

roccurves_gm12878 <- ggplot(roc, aes(x=x, y=y, color = color)) + geom_line() +  theme_bw() + scale_color_manual(values=c(color)) + xlim(0,1) + ylim(0,1) + geom_abline(intercept = 0, slope = 1, linetype = 2) + labs(title="ROC-disruption (GM12878)", x ="1 - Specificity", y = "Sensitivity")

prcurves_gm12878 <- ggplot(pr, aes(x=x, y=y, color=color)) + geom_line() +  theme_bw() + scale_color_manual(values=c(color)) + xlim(0,1) + ylim(0,1) + labs(title="PR-disruption (GM12878)", x ="Recall", y = "Precision") 

# K562
top_x_table_k562 <- top_n(validation_table_k562, 35, diff_k562)
top_x_table_k562$disrupt <- 1 - top_x_table_k562$chipseq
curve <- evalmod(scores = top_x_table_k562$diff_k562, labels = top_x_table_k562$disrupt, posclass = 1)
autoplot(curve)

roc <- as.data.frame(curve$rocs[[1]]$x)
colnames(roc) <- c("x")
roc$y <- curve$rocs[[1]]$y

pr <- as.data.frame(curve$prcs[[1]]$x)
colnames(pr) <- c("x")
pr$y <- curve$prcs[[1]]$y

roccurves_k562 <- ggplot(roc, aes(x=x, y=y, color = color)) + geom_line() +  theme_bw() + scale_color_manual(values=color) + xlim(0,1) + ylim(0,1) + geom_abline(intercept = 0, slope = 1, linetype = 2) + labs(title="ROC-disruption (K562)", x ="1 - Specificity", y = "Sensitivity")

prcurves_k562 <- ggplot(pr, aes(x=x, y=y, color=color)) + geom_line() +  theme_bw() + scale_color_manual(values=color) + xlim(0,1) + ylim(0,1) + labs(title="PR-disruption (K562)", x ="Recall", y = "Precision") 

p3 <- grid.arrange(roccurves_gm12878 + theme(legend.position="hidden"), prcurves_gm12878 + theme(legend.position="none"),roccurves_k562 + theme(legend.position="hidden"), prcurves_k562 + theme(legend.position="none"), nrow=2, widths = c(4,4),heights = c(4,4))

p4 <- annotate_figure(p3,top = text_grob("Performance curves predicting disruption", color = "black", face = "bold", size = 14))
pdf("../figures/disruption_performance_curves_top35_humanModels.pdf",width = 6, height = 4)
plot(p4)
dev.off()

```

