---
title: "Plots for GM12878 and K562"
author: "Deborah Weighill"
date: "August 5, 2020"
output:
  html_notebook:
    toc: true
    theme: cosmo
    df_print: paged
---

# Load libraries, preliminaries
```{r}
library(xtable)
library(jcolors)
library(precrec)
library(gridExtra)
library(grid)
library(tidyr)
library(plyr)
library(ggpubr)
library(dplyr)
library(pROC)
library(viridis)
library(lemon)
library(GenomicRanges)
library(ggplot2)
library(reshape2)
library(rtracklayer)

# set color pallet
pallet <- c("#FFBF00",  "#870E75"  ,  "#0BB19F" )
chipseq_gm12878 <- "../chipseq/GM12878_remap2018_all_macs2_hg19_v1_2.bed"
chipseq_k562 <- "../chipseq/K562_remap2018_all_macs2_hg19_v1_2.bed"
```

# Create validation table for GM12878 - "human only version"
QBiC only using human models.


## Load GM12878 networks into table
```{r echo=FALSE,results='hide'}
load("../outputs/finalEgret_v1_na12878_eQTLs_06182020_panda.RData")
net <- melt(regnet)
colnames(net) = c("tf","gene","score")
net$id <- paste0(net$tf,net$gene)

regnet_edge_table_gm12878 <- data.frame(net$id)
colnames(regnet_edge_table_gm12878) <- c("id")
regnet_edge_table_gm12878$tf <- as.vector(net$tf)
regnet_edge_table_gm12878$gene <- as.vector(net$gene)
regnet_edge_table_gm12878$panda <- as.vector(net$score)

#list of the file names of the rest of the networks
files <- c("finalEgret_v1_na12878_eQTLs_06182020_egret.RData")

#nicknames of the networks
names <- c("egret_GM12878")

for (p in c(1:1)) {
  filename <- paste0("../outputs/",files[p])
  netname <- names[p]
  print(netname)
  load(filename)
  net <- melt(regnet)
  colnames(net) = c("tf","gene","score")
  net$id <- paste0(net$tf,net$gene)
  if(all.equal(as.vector(regnet_edge_table_gm12878$id),as.vector(net$id))){
    print(p)
    regnet_edge_table_gm12878$tempID <- as.vector(net$score)
    colnames(regnet_edge_table_gm12878)[colnames(regnet_edge_table_gm12878) == "tempID"] <- netname
  }
}
filename <- paste0("../outputs/edge_table_gm12878_08052020.RData")
save(regnet_edge_table_gm12878, file = filename)

```

## Load ChIP-seq data for GM12878
We load the chipseq data and make a ground truth regulatory network for validation.

```{r echo=FALSE,results='hide'}
# read in the gene annotation file with gene ranges.
genes <- read.table("../annotation/ensembl_genes_from_ucsc.txt", header = TRUE,sep = "\t")
genes$promotorLeft <- ifelse(genes$strand == "+", (genes$txStart + 1 - 750), (genes$txEnd-250))
genes$promotorRight <- ifelse(genes$strand == "+", (genes$txStart + 1 + 250), (genes$txEnd+750))
gr_genes_promotor <- GRanges(seqnames = genes$chrom, ranges = IRanges(start = genes$promotorLeft, end = genes$promotorRight), strand = NULL, mcols = genes[,13:15])

# chipseq chipseq data
gr_chipseq_gm12878 <- import(chipseq_gm12878)
overlaps_gm12878 <- data.frame(findOverlaps(gr_chipseq_gm12878, gr_genes_promotor))
overlaps_gm12878$tf_name <- gr_chipseq_gm12878$name[overlaps_gm12878$queryHits]
overlaps_gm12878$gene_name <- gr_genes_promotor$mcols.name2[overlaps_gm12878$subjectHits]
validation_regnet_gm12878 <- separate(overlaps_gm12878, tf_name, c("exp", "tf", "cell line"), "\\.")
chipseq_regnet_gm12878 <- distinct(validation_regnet_gm12878[,c("tf","gene_name")])
chipseq_regnet_gm12878$id <- paste0(chipseq_regnet_gm12878$tf,chipseq_regnet_gm12878$gene)
```

## Overlap egret/panda networks with ChIP-seq data and calcualte differences
We now filter down the panda/egret networks to only include those TFs which are in the ChIP-seq data (our validation set).
```{r echo=FALSE,results='hide'}
filtered_net_table_gm12878 <- regnet_edge_table_gm12878[(regnet_edge_table_gm12878$tf %in% chipseq_regnet_gm12878$tf) ,]
filtered_net_table_gm12878$chipseq <- ifelse(filtered_net_table_gm12878$id %in% chipseq_regnet_gm12878$id,1,0)
filtered_net_table_gm12878$diff_egret <- abs(filtered_net_table_gm12878$egret_GM12878 - filtered_net_table_gm12878$panda)
validation_table_gm12878 <- filtered_net_table_gm12878
filename <- paste0("../outputs/validation_table_gm12878_08052020.RData")
save(validation_table_gm12878, file = filename)
load("../outputs/validation_table_gm12878_08052020.RData")
```


# Create validation table for K562 - "human only version"

## Load K562 networks into table

Load all of the PANDA/EGRET regnets

```{r echo=FALSE,results='hide'}
# panda - note - the panda network is the same file for K562 and GM12878 as there is no genotype information
load("../outputs/finalEgret_v1_na12878_eQTLs_06182020_panda.RData")
net <- melt(regnet)
colnames(net) = c("tf","gene","score")
net$id <- paste0(net$tf,net$gene)

regnet_edge_table_k562 <- data.frame(net$id)
colnames(regnet_edge_table_k562) <- c("id")
regnet_edge_table_k562$tf <- as.vector(net$tf)
regnet_edge_table_k562$gene <- as.vector(net$gene)
regnet_edge_table_k562$panda <- as.vector(net$score)

#list of the file names of the rest of the networks
files <- c("finalEgret_v1_k562_eQTL_06182020_egret.RData")

#nicknames of the networks
names <- c("egret_k562")

for (p in c(1:1)) {
  filename <- paste0("../outputs/",files[p])
  netname <- names[p]
  print(netname)
  load(filename)
  net <- melt(regnet)
  colnames(net) = c("tf","gene","score")
  net$id <- paste0(net$tf,net$gene)
  if(all.equal(as.vector(regnet_edge_table_k562$id),as.vector(net$id))){
    print(p)
    regnet_edge_table_k562$tempID <- as.vector(net$score)
    colnames(regnet_edge_table_k562)[colnames(regnet_edge_table_k562) == "tempID"] <- netname
  }
}
filename <- paste0("../outputs/edge_table_k562_08052020.RData")
save(regnet_edge_table_k562, file = filename)
```

## Load ChIP-seq data for K562
We load the chipseq data and make a ground truth regulatory network for validation.

```{r echo=FALSE,results='hide'}
gr_chipseq_k562 <- import(chipseq_k562)
overlaps_k562 <- data.frame(findOverlaps(gr_chipseq_k562, gr_genes_promotor))
overlaps_k562$tf_name <- gr_chipseq_k562$name[overlaps_k562$queryHits]
overlaps_k562$gene_name <- gr_genes_promotor$mcols.name2[overlaps_k562$subjectHits]
validation_regnet_k562 <- separate(overlaps_k562, tf_name, c("exp", "tf", "cell line"), "\\.")
chipseq_regnet_k562 <- distinct(validation_regnet_k562[,c("tf","gene_name")])
chipseq_regnet_k562$id <- paste0(chipseq_regnet_k562$tf,chipseq_regnet_k562$gene)
```

## Overlap egret/panda networks with ChIP-seq data and calcualte differences
We now filter down the panda/egret networks to only include those TFs which are in the ChIP-seq data (our validation set).
```{r echo=FALSE,results='hide'}
filtered_net_table_k562 <- regnet_edge_table_k562[(regnet_edge_table_k562$tf %in% chipseq_regnet_k562$tf) ,]
filtered_net_table_k562$chipseq <- ifelse(filtered_net_table_k562$id %in% chipseq_regnet_k562$id,1,0)
filtered_net_table_k562$diff_egret <- abs(filtered_net_table_k562$egret_k562 - filtered_net_table_k562$panda)
validation_table_k562 <- filtered_net_table_k562
filename <- paste0("../outputs/validation_table_k562_08052020.RData")
save(validation_table_k562, file = filename)
load("../outputs/validation_table_k562_08052020.RData")
```


# Calculate performance curves and AUCs, pvals and print to latex table

## K562 performance by edge disruption threshold
```{r,results='hide',fig.show=TRUE}
names <- c("egret_k562")
diff_cols <- c(7)
egret_net_nums <- c(5)
panda_net_nums <- c(4)
chipseq_col <- 6
get_roc_for_diff_thresh <- function(egretNum, thresh) {
  diffcolnum <- diff_cols[egretNum]
  egretcol <- egret_net_nums[egretNum]
  panda_col <- panda_net_nums[egretNum]
  
  top_diff_table <- validation_table_k562[(validation_table_k562[,c(diffcolnum)]>thresh),c(panda_col,egretcol,chipseq_col)]
  scores <- join_scores(top_diff_table[,1],top_diff_table[,2])
  labels <- join_labels(top_diff_table[,3],top_diff_table[,3])
  specs <- paste0(names[egretNum],", thresh: ",thresh)
  
  rocP <- roc(top_diff_table$chipseq, top_diff_table[,1], auc = TRUE, direction = "<")
  rocE <- roc(top_diff_table$chipseq, top_diff_table[,2],auc = TRUE, direction = "<")
  testD <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
  pandaAUC <- testD$estimate["AUC of roc1"]
  egretAUC <- testD$estimate["AUC of roc2"]
  pvalD <- testD$p.value
  print(paste(egretNum,thresh,pvalD))
  
  curve_data <- mmdata(scores, labels,modnames = c("panda", specs), dsids = c(1, 2))
  curves <- evalmod(curve_data)
  nn <- attr(curves$rocs,"data_info")$nn[1]
  np <- attr(curves$rocs,"data_info")$np[1]
  result <- c(egretNum, names[egretNum], thresh, nn, np, pandaAUC, egretAUC, pvalD)
  return(result)
  autoplot(curves)
}

egret_priors <- c(1)
thresholds <- c(0.1, 0.15, 0.2, 0.25, 0.3, 0.35, 0.4, 0.45, 0.5, 0.55, 0.6, 0.65, 0.7, 0.75, 0.8, 0.85, 0.9,1,2,3,4)

egret_auc_pvals<- data.frame()

for (p in egret_priors){
  for (t in thresholds){
    result <- get_roc_for_diff_thresh(p,t)
    df <- data.frame(t(result), stringsAsFactors = FALSE)
    egret_auc_pvals <- rbind(egret_auc_pvals,df)
  }
}

colnames(egret_auc_pvals) <- c("egretNum", "name", "thresh", "nn", "np", "pandaAUC", "egretAUC", "pvalD")
egret_auc_pvals
egret_auc_pvals_table <- egret_auc_pvals[,c(3,4,5)]
egret_auc_pvals_table$pandaAUC <- signif(as.numeric(egret_auc_pvals$pandaAUC, digits = 4))
egret_auc_pvals_table$egretAUC <- signif(as.numeric(egret_auc_pvals$egretAUC, digits = 4))
egret_auc_pvals_table$pvalue <- signif(as.numeric(egret_auc_pvals$pvalD, digits = 4))
print(xtable(egret_auc_pvals_table, type = "latex"), file = "../tables/pval_table_allHumanModels_thresh_k562.tex", include.rownames=FALSE)
```

## K562 performance by "top n" disrupted edges
```{r,results='hide',fig.show=TRUE}
names <- c("egret_k562")
diff_cols <- c(7)
egret_net_nums <- c(5)
panda_net_nums <- c(4)
chipseq_col <- 6
get_roc_for_top_x <- function(egretNum, x) {
  egretcol <- egret_net_nums[egretNum]
  panda_col <- panda_net_nums[egretNum]
  
  top_x_table <- top_n(validation_table_k562,x,diff_k562)[,c(panda_col,egretcol,chipseq_col)] 
  scores <- join_scores(top_x_table[,1],top_x_table[,2])
  labels <- join_labels(top_x_table[,3],top_x_table[,3])
  specs <- paste0(names[egretNum],", top ",x)
  
  rocP <- roc(top_x_table$chipseq, top_x_table[,1], auc = TRUE, direction = "<")
  rocE <- roc(top_x_table$chipseq, top_x_table[,2],auc = TRUE, direction = "<")
  test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
  pandaAUC <- test$estimate["AUC of roc1"]
  egretAUC <- test$estimate["AUC of roc2"]
  pval <- test$p.value
  print(paste(egretNum,x,pval))
  
  curve_data <- mmdata(scores, labels,modnames = c("panda", specs), dsids = c(1, 2))
  curves <- evalmod(curve_data)
  nn <- attr(curves$rocs,"data_info")$nn[1]
  np <- attr(curves$rocs,"data_info")$np[1]
  result <- c(egretNum, names[egretNum], x, nn, np, pandaAUC, egretAUC, pval)
  return(result)
  autoplot(curves)
}

egret_priors <- c(1)
numEdges <- c(10,15,20,25,30,35,40,45,50)

egret_auc_pvals_topn<- data.frame()

for (p in egret_priors){
  for (x in numEdges){
    result <- get_roc_for_top_x(p,x)
    df <- data.frame(t(result), stringsAsFactors = FALSE)
    egret_auc_pvals_topn <- rbind(egret_auc_pvals_topn,df)
  }
}
colnames(egret_auc_pvals_topn) <- c("egretNum", "name", "thresh", "nn", "np", "pandaAUC", "egretAUC", "pvalD")
egret_auc_pvals_topn
egret_auc_pvals_table <- egret_auc_pvals_topn[,c(3,4,5)]
egret_auc_pvals_table$pandaAUC <- signif(as.numeric(egret_auc_pvals_topn$pandaAUC), digits = 5)
egret_auc_pvals_table$egretAUC <- signif(as.numeric(egret_auc_pvals_topn$egretAUC), digits = 5)
egret_auc_pvals_table$pvalue <- signif(as.numeric(egret_auc_pvals_topn$pval), digits = 5)
print(xtable(egret_auc_pvals_table, type = "latex", digits = 5), file= "../tables/pval_table_allHumanModels_topn_k562.tex",include.rownames=FALSE)
```

Make a bar plot

```{r}
auc_improvement_table <- data.frame()

edited_edges_k562 <- read.table("../outputs/edge_modifications_finalEgret_v1_k562_eQTL_06182020.txt")
colnames(edited_edges_k562) <- c("tf","gene","id","mod")
edited_subset <- validation_table_k562[which(validation_table_k562$id %in% edited_edges_k562$id),]

rocP <- roc(edited_subset$chipseq, edited_subset$panda, auc = TRUE, direction = "<")
rocE <- roc(edited_subset$chipseq, edited_subset$egret_k562,auc = TRUE, direction = "<")
test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
pandaAUC <- test$estimate["AUC of roc1"]
egretAUC <- test$estimate["AUC of roc2"]
pval <- test$p.value
print(paste(pandaAUC,egretAUC,pval))
 
auc_improvement_table <- as.data.frame(t(c("K562",((egretAUC-pandaAUC)/pandaAUC)*100,pval)))

```


## GM12878 performance by edge disruption threshold
```{r,results='hide',fig.show=TRUE}
names <- c("egret_GM12878")
diff_cols <- c(7)
egret_net_nums <- c(5)
panda_net_nums <- c(4)
chipseq_col <- 6
get_roc_for_diff_thresh <- function(egretNum, thresh) {
  diffcolnum <- diff_cols[egretNum]
  egretcol <- egret_net_nums[egretNum]
  panda_col <- panda_net_nums[egretNum]
  
  top_diff_table <- validation_table_gm12878[(validation_table_gm12878[,c(diffcolnum)]>thresh),c(panda_col,egretcol,chipseq_col)]
  scores <- join_scores(top_diff_table[,1],top_diff_table[,2])
  labels <- join_labels(top_diff_table[,3],top_diff_table[,3])
  specs <- paste0(names[egretNum],", thresh: ",thresh)
  
  rocP <- roc(top_diff_table$chipseq, top_diff_table[,1], auc = TRUE, direction = "<")
  rocE <- roc(top_diff_table$chipseq, top_diff_table[,2],auc = TRUE, direction = "<")
  test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
  pandaAUC <- test$estimate["AUC of roc1"]
  egretAUC <- test$estimate["AUC of roc2"]
  pval <- test$p.value
  print(paste(egretNum,thresh,pval))
  
  curve_data <- mmdata(scores, labels,modnames = c("panda", specs), dsids = c(1, 2))
  curves <- evalmod(curve_data)
  nn <- attr(curves$rocs,"data_info")$nn[1]
  np <- attr(curves$rocs,"data_info")$np[1]
  result <- c(egretNum, names[egretNum], thresh, nn, np, pandaAUC, egretAUC, pval)
  return(result)
  autoplot(curves)
}

egret_priors <- c(1)
thresholds <- c(0.1, 0.15, 0.2, 0.25, 0.3, 0.35, 0.4, 0.45, 0.5, 0.55, 0.6, 0.65, 0.7, 0.75, 0.8, 0.85, 0.9, 1)

egret_auc_pvals<- data.frame()

for (p in egret_priors){
  for (t in thresholds){
    result <- get_roc_for_diff_thresh(p,t)
    df <- data.frame(t(result), stringsAsFactors = FALSE)
    egret_auc_pvals <- rbind(egret_auc_pvals,df)
  }
}

colnames(egret_auc_pvals) <- c("egretNum", "name", "thresh", "nn", "np", "pandaAUC", "egretAUC", "pval")
egret_auc_pvals
print(xtable(egret_auc_pvals[,c(3:8)], type = "latex"), file = "../tables/pval_table_allHumanModels_thresh_gm12878.tex",include.rownames=FALSE)
```

## GM12878 performance by "top n" disrupted edges

```{r,results='hide',fig.show=TRUE}

names <- c("egret_GM12878")
diff_cols <- c(7)
egret_net_nums <- c(5)
panda_net_nums <- c(4)
chipseq_col <- 6
get_roc_for_top_x <- function(egretNum, x) {
  egretcol <- egret_net_nums[egretNum]
  panda_col <- panda_net_nums[egretNum]
  
  top_x_table <- top_n(validation_table_gm12878,x,diff_gm12878)[,c(panda_col,egretcol,chipseq_col)] 
  scores <- join_scores(top_x_table[,1],top_x_table[,2])
  labels <- join_labels(top_x_table[,3],top_x_table[,3])
  specs <- paste0(names[egretNum],", top ",x)
  
  rocP <- roc(top_x_table$chipseq, top_x_table[,1], auc = TRUE, direction = "<")
  rocE <- roc(top_x_table$chipseq, top_x_table[,2],auc = TRUE, direction = "<")
  test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
  pandaAUC <- test$estimate["AUC of roc1"]
  egretAUC <- test$estimate["AUC of roc2"]
  pval <- test$p.value
  print(paste(egretNum,x,pval))
  
  curve_data <- mmdata(scores, labels,modnames = c("panda", specs), dsids = c(1, 2))
  curves <- evalmod(curve_data)
  nn <- attr(curves$rocs,"data_info")$nn[1]
  np <- attr(curves$rocs,"data_info")$np[1]
  result <- c(egretNum, names[egretNum], x, nn, np, pandaAUC, egretAUC, pval)
  return(result)
  autoplot(curves)
}

egret_priors <- c(1)
numEdges <- c(10,15,20,25,30,35,40,45,50)

egret_auc_pvals_topn<- data.frame()

for (p in egret_priors){
  for (x in numEdges){
    result <- get_roc_for_top_x(p,x)
    df <- data.frame(t(result), stringsAsFactors = FALSE)
    egret_auc_pvals_topn <- rbind(egret_auc_pvals_topn,df)
  }
}

colnames(egret_auc_pvals_topn) <- c("egretNum", "name", "topN", "nn", "np", "pandaAUC", "egretAUC", "pval")
egret_auc_pvals_topn
egret_auc_pvals_table <- egret_auc_pvals_topn[,c(3,4,5)]
egret_auc_pvals_table$pandaAUC <- signif(as.numeric(egret_auc_pvals_topn$pandaAUC), digits = 5)
egret_auc_pvals_table$egretAUC <- signif(as.numeric(egret_auc_pvals_topn$egretAUC), digits = 5)
egret_auc_pvals_table$pvalue <- signif(as.numeric(egret_auc_pvals_topn$pval), digits = 5)
print(xtable(egret_auc_pvals_table, type = "latex", digits = 5), file = "../tables/pval_table_allHumanModels_topn_gm12878.tex",include.rownames=FALSE,)
```

```{r}
edited_edges_gm12878 <- read.table("../outputs/edge_modifications_EGRET_na12878_08142020.txt")
colnames(edited_edges_gm12878) <- c("tf","gene","id","mod")
edited_subset_g <- validation_table_gm12878[which(validation_table_gm12878$id %in% edited_edges_gm12878$id),]

rocP <- roc(edited_subset_g$chipseq, edited_subset_g$panda, auc = TRUE, direction = "<")
rocE <- roc(edited_subset_g$chipseq, edited_subset_g$egret_GM12878,auc = TRUE, direction = "<")
test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
pandaAUC <- test$estimate["AUC of roc1"]
egretAUC <- test$estimate["AUC of roc2"]
pval <- test$p.value
print(paste(pandaAUC,egretAUC,pval))
  
auc_improvement_table <- rbind(auc_improvement_table,(t(c("GM12878",((egretAUC-pandaAUC)/pandaAUC)*100,pval))))
colnames(auc_improvement_table) <- c("Genotype","AUC_Improvement","pvalue")
auc_improvement_table$p_round <- paste0("Delong p = ",signif(as.numeric(as.vector(auc_improvement_table$pvalue)), digits = 5))

p <- ggplot(auc_improvement_table, aes(x=Genotype, y=as.numeric(as.vector((AUC_Improvement))), fill = Genotype, group = Genotype)) +geom_bar(stat="identity")  +theme_bw() +  scale_fill_manual(values=pallet) + ylim(0,95)+ labs(x ="Genotype", y = "AU-ROC percentage imprvement")  + theme(legend.position="none")  + geom_text(aes(label=c(" ","*")), position=position_dodge(width=0.9), vjust=-0.6)

pdf("../figures/auc_improvement_human.pdf",width = 3.5, height = 3.5)
plot(p)
dev.off()
```

# Select examples for visualizing in genome browser

```{r}
colnames(validation_table_gm12878) <- c("id","tf","gene","panda","egret_gm12878","chipseq_gm12878","diff_gm12878")
colnames(validation_table_k562) <- c("id","tf","gene","panda","egret_k562","chipseq_k562","diff_k562")
combined_table <- join(validation_table_gm12878,validation_table_k562,by = c("id","tf","gene"),type="inner")
combined_table[which((combined_table$chipseq_k562 != combined_table$chipseq_gm12878) & (combined_table$diff_k562 > 0.5)),c(1:11)]
```



# Box plots of edge differences for top difference edges
Make a box plot showing the distribution in egret edge weights for top diff edges. Also show distribution of differences, each time grouped by chipseq +/-. Remember, top difference here is not chosen as top difference in chipseq or anything, its chosen as top difference between EGRET and PANDA. We're trying to isolate the effect of the variants.


```{r}
topn_gm12878 <- top_n(validation_table_gm12878, 48, diff_gm12878)
test <- t.test(topn_gm12878[which(topn_gm12878$chipseq_gm12878 == 0),]$diff_gm12878,topn_gm12878[which(topn_gm12878$chipseq_gm12878 == 1),]$diff_gm12878, alternative="greater")


p1 <- ggplot(topn_gm12878, aes(x = as.factor(chipseq_gm12878), y = diff_gm12878, color = as.factor(chipseq_gm12878), group = as.factor(chipseq_gm12878)))+ geom_boxplot() + geom_jitter(position=position_jitter(0.1))+  stat_summary(fun.y=mean, geom="point", shape=18,size=5, color="#E41A1C") + theme_bw() + scale_color_manual(values=pallet)  + theme(legend.position="right") + labs(title="GM12878", x ="ChIP-seq binding", y = "Edge disruption score", color  = "ChIP-seq binding") + annotate("text", x = 1.6, y = 7.5, label = paste0("t-test p= ",signif(test$p.value,4)))

topn_k562 <- top_n(validation_table_k562, 48, diff_k562)

test <- t.test(topn_k562[which(topn_k562$chipseq_k562 == 0),]$diff_k562,topn_k562[which(topn_k562$chipseq_k562 == 1),]$diff_k562, alternative="greater")


p2 <- ggplot(topn_k562, aes(x = as.factor(chipseq_k562), y = diff_k562, color = as.factor(chipseq_k562), group = as.factor(chipseq_k562)))+ geom_boxplot() + geom_jitter(position=position_jitter(0.1))+  stat_summary(fun.y=mean, geom="point", shape=18,size=5, color="#E41A1C") + theme_bw() + scale_color_manual(values=pallet)  + theme(legend.position="right") + labs(title="K562", x ="ChIP-seq binding", y = "Edge disruption score", color  = "ChIP-seq binding") + annotate("text", x = 1.7, y = 10, label = paste0("t-test p= ",signif(test$p.value,4)))

p3 <- grid.arrange(p1 + theme(legend.position="none"), p2 + theme(legend.position="right"), nrow=1, widths = c(3.5,5.9))
p4 <- annotate_figure(p3,top = text_grob("Top 35 edge disruption scores", color = "black", face = "bold", size = 14))

pdf("../figures/top35_diffvalues_boxplots_k562_gm12878.pdf",width = 6, height = 3.5)
plot(p4)
dev.off()
plot(p4)
```

```{r}
topn_gm12878 <- top_n(validation_table_gm12878, 48, diff_gm12878)
test <- t.test(topn_gm12878[which(topn_gm12878$chipseq_gm12878 == 0),]$egret_gm12878,topn_gm12878[which(topn_gm12878$chipseq_gm12878 == 1),]$egret_gm12878  , alternative="less")


p1 <- ggplot(topn_gm12878, aes(x = as.factor(chipseq_gm12878), y = egret_gm12878, color = as.factor(chipseq_gm12878), group = as.factor(chipseq_gm12878)))+ geom_boxplot() + geom_jitter(position=position_jitter(0.1))+  stat_summary(fun.y=mean, geom="point", shape=18,size=5, color="#E41A1C") + theme_bw() + scale_color_manual(values=pallet)  + theme(legend.position="right") + labs(title="GM12878", x ="ChIP-seq binding", y = "Edge disruption score", color  = "ChIP-seq binding") + annotate("text", x = 1.6, y = 7.5, label = paste0("t-test p= ",signif(test$p.value,4)))

topn_k562 <- top_n(validation_table_k562, 48, diff_k562)

test <- t.test(topn_k562[which(topn_k562$chipseq_k562 == 0),]$egret_k562,topn_k562[which(topn_k562$chipseq_k562 == 1),]$egret_k562, alternative="less")


p2 <- ggplot(topn_k562, aes(x = as.factor(chipseq_k562), y = egret_k562, color = as.factor(chipseq_k562), group = as.factor(chipseq_k562)))+ geom_boxplot() + geom_jitter(position=position_jitter(0.1))+  stat_summary(fun.y=mean, geom="point", shape=18,size=5, color="#E41A1C") + theme_bw() + scale_color_manual(values=pallet)  + theme(legend.position="right") + labs(title="K562", x ="ChIP-seq binding", y = "Edge disruption score", color  = "ChIP-seq binding") + annotate("text", x = 1.7, y = 10, label = paste0("t-test p= ",signif(test$p.value,4)))

p3 <- grid.arrange(p1 + theme(legend.position="none"), p2 + theme(legend.position="right"), nrow=1, widths = c(3.5,5.9))
p4 <- annotate_figure(p3,top = text_grob("Top 35 edge disruption scores", color = "black", face = "bold", size = 14))

#pdf("../figures/top35_diffvalues_boxplots_k562_gm12878.pdf",width = 6, height = 3.5)
plot(p4)
#dev.off()
#plot(p4)
```



# Performance curves AUCs/AU-PRs for top35 edges
```{r}
pallet <- c("#870E75"  ,  "#0BB19F" )
top_x_table_gm12878 <- top_n(validation_table_gm12878, 35, diff_gm12878)[,c(4,5,6)]
scores <- join_scores(top_x_table_gm12878[,1],top_x_table_gm12878[,2])
labels <- join_labels(top_x_table_gm12878[,3],top_x_table_gm12878[,3])
curve_data <- mmdata(scores, labels,modnames = c("panda", "egret"))
curves <- evalmod(curve_data)
#autoplot(curves)
#curves
#auc(curves)

rocP <- roc(top_x_table_gm12878$chipseq_gm12878, top_x_table_gm12878$panda, auc = TRUE, direction = "<")
rocE <- roc(top_x_table_gm12878$chipseq_gm12878, top_x_table_gm12878$egret_gm12878,auc = TRUE, direction = "<")
test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
pandaAUC <- test$estimate["AUC of roc1"]
egretAUC <- test$estimate["AUC of roc2"]
pval <- round(test$p.value,digits = 4)

pandaROC <- as.data.frame(curves$rocs[[1]]$x)
colnames(pandaROC) <- c("x")
pandaROC$y <- curves$rocs[[1]]$y
pandaROC$Method <- "PANDA"

egretROC <- as.data.frame(curves$rocs[[2]]$x)
colnames(egretROC) <- c("x")
egretROC$y <- curves$rocs[[2]]$y
egretROC$Method <- "EGRET"

roc_data <- rbind(pandaROC, egretROC)
roccurves_gm12878 <- ggplot(roc_data, aes(x=x, y=y, color=Method, group = Method)) + geom_line() +  theme_bw() + scale_color_manual(values=pallet) + xlim(0,1) + ylim(0,1) + geom_abline(intercept = 0, slope = 1, linetype = 2) + labs(title="ROC (GM12878)", x ="1 - Specificity", y = "Sensitivity") + annotate("text", x = 0.3, y = 0.8, label = paste0("Delong p: ",pval)) 


pandaPR <- as.data.frame(curves$prcs[[1]]$x)
colnames(pandaPR) <- c("x")
pandaPR$y <- curves$prcs[[1]]$y
pandaPR$Method <- "PANDA"

egretPR <- as.data.frame(curves$prcs[[2]]$x)
colnames(egretPR) <- c("x")
egretPR$y <- curves$prcs[[2]]$y
egretPR$Method <- "EGRET"

pr_data <- rbind(pandaPR, egretPR)
prcurves_gm12878 <- ggplot(pr_data, aes(x=x, y=y, color=Method, group = Method)) + geom_line() +  theme_bw() + scale_color_manual(values=pallet) + xlim(0,1) + ylim(0,1) + labs(title="PR (GM12878)", x ="Recall", y = "Precision") 
```


```{r}
top_x_table_k562 <- top_n(validation_table_k562, 35, diff_k562)[,c(4,5,6)]
scores <- join_scores(top_x_table_k562[,1],top_x_table_k562[,2])
labels <- join_labels(top_x_table_k562[,3],top_x_table_k562[,3])
curve_data <- mmdata(scores, labels,modnames = c("panda", "egret"))
curves <- evalmod(curve_data)
#autoplot(curves)
#curves
#auc(curves)

rocP <- roc(top_x_table_k562$chipseq_k562, top_x_table_k562$panda, auc = TRUE, direction = "<")
rocE <- roc(top_x_table_k562$chipseq_k562, top_x_table_k562$egret_k562,auc = TRUE, direction = "<")
test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
pandaAUC <- test$estimate["AUC of roc1"]
egretAUC <- test$estimate["AUC of roc2"]
pval <- round(test$p.value,digits = 4)

pandaROC <- as.data.frame(curves$rocs[[1]]$x)
colnames(pandaROC) <- c("x")
pandaROC$y <- curves$rocs[[1]]$y
pandaROC$Method <- "PANDA"

egretROC <- as.data.frame(curves$rocs[[2]]$x)
colnames(egretROC) <- c("x")
egretROC$y <- curves$rocs[[2]]$y
egretROC$Method <- "EGRET"

roc_data <- rbind(pandaROC, egretROC)
roccurves_k562 <- ggplot(roc_data, aes(x=x, y=y, color=Method, group = Method)) + geom_line() +  theme_bw() + scale_color_manual(values=pallet) + xlim(0,1) + ylim(0,1) + geom_abline(intercept = 0, slope = 1, linetype = 2) + labs(title="ROC (K562)", x ="1 - Specificity", y = "Sensitivity") +annotate("text", x = 0.25, y = 0.8, label = paste0("Delong p: ",pval))


pandaPR <- as.data.frame(curves$prcs[[1]]$x)
colnames(pandaPR) <- c("x")
pandaPR$y <- curves$prcs[[1]]$y
pandaPR$Method <- "PANDA"

egretPR <- as.data.frame(curves$prcs[[2]]$x)
colnames(egretPR) <- c("x")
egretPR$y <- curves$prcs[[2]]$y
egretPR$Method <- "EGRET"

pr_data <- rbind(pandaPR, egretPR)
prcurves_k562 <- ggplot(pr_data, aes(x=x, y=y, color=Method, group = Method)) + geom_line() +  theme_bw() + scale_color_manual(values=pallet) + xlim(0,1) + ylim(0,1) + labs(title="PR (K562)", x ="Recall", y = "Precision") 
```

```{r}
p3 <- grid.arrange(roccurves_gm12878 + theme(legend.position="hidden"), prcurves_gm12878 + theme(legend.position="none"),roccurves_k562 + theme(legend.position="hidden"), prcurves_k562 + theme(legend.position="none"), nrow=2, widths = c(4,4),heights = c(4,4))

p4 <- annotate_figure(p3,top = text_grob("Performance curves for top 35 disrupted edges", color = "black", face = "bold", size = 14))
pdf("../figures/performance_curves_top35_humanModels.pdf",width = 6, height = 4)
plot(p4)
dev.off()
```



# Now show that difference scores are predictive of disruption with AUCs
```{r}
colnames(validation_table_gm12878) <- c("id","tf","gene","panda","egret_gm12878","chipseq_gm12878","diff_gm12878")
get_roc_for_top_x <- function(egretNum, x) {
  
  top_x_table_gm12878 <- top_n(validation_table_gm12878, x, diff_gm12878)
  top_x_table_gm12878$disrupt <- 1 - top_x_table_gm12878$chipseq
  curve <- evalmod(scores = top_x_table_gm12878$diff_gm12878, labels = top_x_table_gm12878$disrupt, posclass = 1)
  aucs <- precrec::auc(curve)
  autoplot(curve)
  auc_gm12878 <- round(aucs$aucs[1], digits=4)
  prc_gm12878 <- round(aucs$aucs[2], digits=4)
  nn <- attr(curve$rocs,"data_info")$nn[1]
  np <- attr(curve$rocs,"data_info")$np[1]
  
  result <- c("egret",x, nn, np, auc_gm12878, prc_gm12878)
  return(result)
}

egret_priors <- c(1)
numEdges <- c(10,15,20,25,30,35,40,45,50)

disrup_auc_topn<- data.frame()

for (p in egret_priors){
  for (x in numEdges){
    result <- get_roc_for_top_x(p,x)
    df <- data.frame(t(result), stringsAsFactors = FALSE)
    disrup_auc_topn <- rbind(disrup_auc_topn,df)
  }
}

colnames(disrup_auc_topn) <- c("egret","topN", "nn", "np", "AU-ROC", "AU-PR")
disrup_auc_topn
print(xtable(disrup_auc_topn[,c(2,3,4,5,6)], type = "latex", digits = 5), file = "../tables/disruption_auc_table_allHumanModels_topn_gm12878.tex",include.rownames=FALSE,)
```

```{r}
colnames(validation_table_k562) <- c("id","tf","gene","panda","egret_k562","chipseq_gk562","diff_k562")

get_roc_for_top_x <- function(egretNum, x) {
  
  top_x_table_k562 <- top_n(validation_table_k562, x, diff_k562)
  top_x_table_k562$disrupt <- 1 - top_x_table_k562$chipseq
  curve <- evalmod(scores = top_x_table_k562$diff_k562, labels = top_x_table_k562$disrupt, posclass = 1)
  aucs <- precrec::auc(curve)
  autoplot(curve)
  auc_k562 <- round(aucs$aucs[1], digits=4)
  prc_k562 <- round(aucs$aucs[2], digits=4)
  nn <- attr(curve$rocs,"data_info")$nn[1]
  np <- attr(curve$rocs,"data_info")$np[1]
  
  result <- c("egret",x, nn, np, auc_k562, prc_k562)
  return(result)
}

egret_priors <- c(1)
numEdges <- c(10,15,20,25,30,35,40,45,50)

disrup_auc_topn<- data.frame()

for (p in egret_priors){
  for (x in numEdges){
    result <- get_roc_for_top_x(p,x)
    df <- data.frame(t(result), stringsAsFactors = FALSE)
    disrup_auc_topn <- rbind(disrup_auc_topn,df)
  }
}

colnames(disrup_auc_topn) <- c("egret","topN", "nn", "np", "AU-ROC", "AU-PR")
disrup_auc_topn
print(xtable(disrup_auc_topn[,c(2,3,4,5,6)], type = "latex", digits = 5), file = "../tables/disruption_auc_table_allHumanModels_topn_k562.tex",include.rownames=FALSE,)
```

```{r}
pallet <- c("#870E75"  ,  "#0BB19F" )
color <- "#870E75"
#GM12878
top_x_table_gm12878 <- top_n(validation_table_gm12878, 35, diff_gm12878)
top_x_table_gm12878$disrupt <- 1 - top_x_table_gm12878$chipseq
curve <- evalmod(scores = top_x_table_gm12878$diff_gm12878, labels = top_x_table_gm12878$disrupt, posclass = 1)
aucs <- precrec::auc(curve)
autoplot(curve)
auc_gm12878 <- c("GM12878","AU-ROC",round(aucs$aucs[1], digits=4))
prc_gm12878 <- c("GM12878","AU-PR",round(aucs$aucs[2], digits=4))

roc <- as.data.frame(curve$rocs[[1]]$x)
colnames(roc) <- c("x")
roc$y <- curve$rocs[[1]]$y

pr <- as.data.frame(curve$prcs[[1]]$x)
colnames(pr) <- c("x")
pr$y <- curve$prcs[[1]]$y

roccurves_gm12878 <- ggplot(roc, aes(x=x, y=y, color = color)) + geom_line() +  theme_bw() + scale_color_manual(values=c(color)) + xlim(0,1) + ylim(0,1) + geom_abline(intercept = 0, slope = 1, linetype = 2) + labs(title="ROC-disruption (GM12878)", x ="1 - Specificity", y = "Sensitivity")

prcurves_gm12878 <- ggplot(pr, aes(x=x, y=y, color=color)) + geom_line() +  theme_bw() + scale_color_manual(values=c(color)) + xlim(0,1) + ylim(0,1) + labs(title="PR-disruption (GM12878)", x ="Recall", y = "Precision") 

# K562
top_x_table_k562 <- top_n(validation_table_k562, 35, diff_k562)
top_x_table_k562$disrupt <- 1 - top_x_table_k562$chipseq
curve <- evalmod(scores = top_x_table_k562$diff_k562, labels = top_x_table_k562$disrupt, posclass = 1)
autoplot(curve)
aucs <- precrec::auc(curve)
auc_k562<- c("K562","AU-ROC",round(aucs$aucs[1],digits = 4))
prc_k562 <- c("K562","AU-PR",round(aucs$aucs[2],digits = 4))


roc <- as.data.frame(curve$rocs[[1]]$x)
colnames(roc) <- c("x")
roc$y <- curve$rocs[[1]]$y

pr <- as.data.frame(curve$prcs[[1]]$x)
colnames(pr) <- c("x")
pr$y <- curve$prcs[[1]]$y

roccurves_k562 <- ggplot(roc, aes(x=x, y=y, color = color)) + geom_line() +  theme_bw() + scale_color_manual(values=color) + xlim(0,1) + ylim(0,1) + geom_abline(intercept = 0, slope = 1, linetype = 2) + labs(title="ROC-disruption (K562)", x ="1 - Specificity", y = "Sensitivity")

prcurves_k562 <- ggplot(pr, aes(x=x, y=y, color=color)) + geom_line() +  theme_bw() + scale_color_manual(values=color) + xlim(0,1) + ylim(0,1) + labs(title="PR-disruption (K562)", x ="Recall", y = "Precision") 

p3 <- grid.arrange(roccurves_gm12878 + theme(legend.position="hidden"), prcurves_gm12878 + theme(legend.position="none"),roccurves_k562 + theme(legend.position="hidden"), prcurves_k562 + theme(legend.position="none"), nrow=2, widths = c(4,4),heights = c(4,4))

p4 <- annotate_figure(p3,top = text_grob("Performance curves predicting disruption", color = "black", face = "bold", size = 14))
pdf("../figures/disruption_performance_curves_top35_humanModels.pdf",width = 6, height = 4)
plot(p4)
dev.off()

```


```{r}
table <- rbind(auc_gm12878,prc_gm12878,auc_k562, prc_k562)
colnames(table) <- c("Cell type","Measure","Area")
print(xtable(table, type = "latex", digits = 4), file = "../tables/aucs_disruption.tex",include.rownames=FALSE,)
```

# Analysis of edited prior edges
```{r}
# load gm12878 edited prior edges
edited_gm12878 <- read.table("../outputs/edge_modifications_EGRET_na12878_08142020.txt", header = FALSE)
dim(edited_gm12878)
colnames(edited_gm12878) <- c("tf","gene","id","mod")

p1 <- ggplot(edited_gm12878, aes(x=mod)) + geom_density(alpha = 0.1, fill=pallet[3], col = pallet[3]) + theme_bw() + scale_color_manual(values=pallet) + scale_fill_manual(values=pallet) +labs(title="GM12878", x ="Prior modification", y = "Frequency")


edited_gm12878_chip <- edited_gm12878[which(edited_gm12878$tf %in% chipseq_regnet_gm12878$tf),]
edited_gm12878_chip$chipseq <- ifelse(edited_gm12878_chip$id %in% chipseq_regnet_gm12878$id, 1, 0)

# count number of 1's
dim(edited_gm12878_chip[edited_gm12878_chip$chipseq == 1,])
dim(edited_gm12878_chip[edited_gm12878_chip$chipseq == 0,])
```

# Analysis of edited prior edges
```{r}
# load k562 edited prior edges
edited_k562 <- read.table("../outputs/edge_modifications_finalEgret_v1_k562_eQTL_06182020.txt", header = FALSE)
dim(edited_k562)
colnames(edited_k562) <- c("tf","gene","id","mod")

p2 <- ggplot(edited_k562, aes(x=mod)) + geom_density(alpha = 0.1, fill=pallet[2], col = pallet[2]) + theme_bw() + scale_color_manual(values=pallet) + scale_fill_manual(values=pallet) +labs(title="K562", x ="Prior modification", y = "Frequency")
p2
p3 <- grid.arrange(p1 + theme(legend.position="none"), p2 + theme(legend.position="none"), nrow=1, widths = c(4,4))
p3

p4 <- annotate_figure(p3,top = text_grob("Prior modification distributions (only human QBiC models)", color = "black", face = "bold", size = 14))
pdf("../figures/modification_distribution_humanModels.pdf",width = 6, height = 4)
plot(p4)
dev.off()

edited_k562_chip <- edited_k562[which(edited_k562$tf %in% chipseq_regnet_k562$tf),]
edited_k562_chip$chipseq <- ifelse(edited_k562_chip$id %in% chipseq_regnet_k562$id, 1, 0)

# count number of 1's
dim(edited_k562_chip[edited_k562_chip$chipseq == 1,])
dim(edited_k562_chip[edited_k562_chip$chipseq == 0,])
```

# Subset to modified edges only
Make columns for chipseq, edge weight, prior mod and prior edge weight

```{r}
library(xtable)
library(jcolors)
library(precrec)
library(gridExtra)
library(grid)
library(tidyr)
library(plyr)
library(ggpubr)
library(dplyr)
library(pROC)
library(viridis)
library(lemon)
library(GenomicRanges)
library(ggplot2)
library(reshape2)
library(rtracklayer)

# read in the gene annotation file with gene ranges.
genes <- read.table("../annotation/ensembl_genes_from_ucsc.txt", header = TRUE,sep = "\t")
genes$promotorLeft <- ifelse(genes$strand == "+", (genes$txStart + 1 - 750), (genes$txEnd-250))
genes$promotorRight <- ifelse(genes$strand == "+", (genes$txStart + 1 + 250), (genes$txEnd+750))
gr_genes_promotor <- GRanges(seqnames = genes$chrom, ranges = IRanges(start = genes$promotorLeft, end = genes$promotorRight), strand = NULL, mcols = genes[,13:15])

# chipseq chipseq data
gr_chipseq_gm12878 <- import(chipseq_gm12878)
overlaps_gm12878 <- data.frame(findOverlaps(gr_chipseq_gm12878, gr_genes_promotor))
overlaps_gm12878$tf_name <- gr_chipseq_gm12878$name[overlaps_gm12878$queryHits]
overlaps_gm12878$gene_name <- gr_genes_promotor$mcols.name2[overlaps_gm12878$subjectHits]
validation_regnet_gm12878 <- separate(overlaps_gm12878, tf_name, c("exp", "tf", "cell line"), "\\.")
chipseq_regnet_gm12878 <- distinct(validation_regnet_gm12878[,c("tf","gene_name")])
chipseq_regnet_gm12878$id <- paste0(chipseq_regnet_gm12878$tf,chipseq_regnet_gm12878$gene)

gr_chipseq_k562 <- import(chipseq_k562)
overlaps_k562 <- data.frame(findOverlaps(gr_chipseq_k562, gr_genes_promotor))
overlaps_k562$tf_name <- gr_chipseq_k562$name[overlaps_k562$queryHits]
overlaps_k562$gene_name <- gr_genes_promotor$mcols.name2[overlaps_k562$subjectHits]
validation_regnet_k562 <- separate(overlaps_k562, tf_name, c("exp", "tf", "cell line"), "\\.")
chipseq_regnet_k562 <- distinct(validation_regnet_k562[,c("tf","gene_name")])
chipseq_regnet_k562$id <- paste0(chipseq_regnet_k562$tf,chipseq_regnet_k562$gene)

load("../outputs/validation_table_gm12878_08052020.RData")
load("../outputs/validation_table_k562_08052020.RData")
```

```{r}
edited_gm12878_chip$prior <- 1 - edited_gm12878_chip$mod
edited_gm12878_chip$egret <- validation_table_gm12878$egret_GM12878[match(edited_gm12878_chip$id, validation_table_gm12878$id)]
edited_gm12878_chip$disrup <- validation_table_gm12878$diff_egret[match(edited_gm12878_chip$id, validation_table_gm12878$id)]
edges_for_valid_gm12878 <- edited_gm12878_chip[which(!(is.na(edited_gm12878_chip$egret))),]

edited_k562_chip$prior <- 1 - edited_k562_chip$mod
edited_k562_chip$egret <- validation_table_k562$egret_k562[match(edited_k562_chip$id, validation_table_k562$id)]
edited_k562_chip$disrup <- validation_table_k562$diff_egret[match(edited_k562_chip$id, validation_table_k562$id)]
edges_for_valid_k562 <- edited_k562_chip[which(!(is.na(edited_k562_chip$egret))),]
```


boxplots - GM12878
```{r}
# egret weight - 0's have lower
test <- t.test(edges_for_valid_gm12878[which(edges_for_valid_gm12878$chipseq == 0),]$egret,edges_for_valid_gm12878[which(edges_for_valid_gm12878$chipseq == 1),]$egret, alternative="less")

ggplot(edges_for_valid_gm12878, aes(x = as.factor(chipseq), y = egret, color = as.factor(chipseq), group = as.factor(chipseq)))+ geom_boxplot() + geom_jitter(position=position_jitter(0.1))+  stat_summary(fun.y=mean, geom="point", shape=18,size=5, color="#E41A1C") + theme_bw() + scale_color_manual(values=pallet)  + theme(legend.position="right") + labs(title="GM12878", x ="ChIP-seq binding", y = "score", color  = "ChIP-seq binding") + annotate("text", x = 1.6, y = 7.5, label = paste0("t-test p= ",signif(test$p.value,4)))

# egret prior - 0's have lower
test <- t.test(edges_for_valid_gm12878[which(edges_for_valid_gm12878$chipseq == 0),]$prior,edges_for_valid_gm12878[which(edges_for_valid_gm12878$chipseq == 1),]$prior, alternative="less")

ggplot(edges_for_valid_gm12878, aes(x = as.factor(chipseq), y = prior, color = as.factor(chipseq), group = as.factor(chipseq)))+ geom_boxplot() + geom_jitter(position=position_jitter(0.1))+  stat_summary(fun.y=mean, geom="point", shape=18,size=5, color="#E41A1C") + theme_bw() + scale_color_manual(values=pallet)  + theme(legend.position="right") + labs(title="GM12878", x ="ChIP-seq binding", y = "score", color  = "ChIP-seq binding") + annotate("text", x = 1.6, y = 7.5, label = paste0("t-test p= ",signif(test$p.value,4)))

# egret mod - 1's have lower
test <- t.test(edges_for_valid_gm12878[which(edges_for_valid_gm12878$chipseq == 0),]$mod,edges_for_valid_gm12878[which(edges_for_valid_gm12878$chipseq == 1),]$mod, alternative="greater")

ggplot(edges_for_valid_gm12878, aes(x = as.factor(chipseq), y = mod, color = as.factor(chipseq), group = as.factor(chipseq)))+ geom_boxplot() + geom_jitter(position=position_jitter(0.1))+  stat_summary(fun.y=mean, geom="point", shape=18,size=5, color="#E41A1C") + theme_bw() + scale_color_manual(values=pallet)  + theme(legend.position="right") + labs(title="GM12878", x ="ChIP-seq binding", y = "score", color  = "ChIP-seq binding") + annotate("text", x = 1.6, y = 7.5, label = paste0("t-test p= ",signif(test$p.value,4)))

# disrup - 1's have lower
test <- t.test(edges_for_valid_gm12878[which(edges_for_valid_gm12878$chipseq == 0),]$disrup,edges_for_valid_gm12878[which(edges_for_valid_gm12878$chipseq == 1),]$disrup, alternative="greater")

ggplot(edges_for_valid_gm12878, aes(x = as.factor(chipseq), y = disrup, color = as.factor(chipseq), group = as.factor(chipseq)))+ geom_boxplot() + geom_jitter(position=position_jitter(0.1))+  stat_summary(fun.y=mean, geom="point", shape=18,size=5, color="#E41A1C") + theme_bw() + scale_color_manual(values=pallet)  + theme(legend.position="right") + labs(title="GM12878", x ="ChIP-seq binding", y = "score", color  = "ChIP-seq binding") + annotate("text", x = 1.6, y = 7.5, label = paste0("t-test p= ",signif(test$p.value,4)))

```

boxplots - K562
```{r}
# egret weight - 0's have lower
test <- t.test(edges_for_valid_k562[which(edges_for_valid_k562$chipseq == 0),]$egret,edges_for_valid_k562[which(edges_for_valid_k562$chipseq == 1),]$egret, alternative="less")

ggplot(edges_for_valid_k562, aes(x = as.factor(chipseq), y = egret, color = as.factor(chipseq), group = as.factor(chipseq)))+ geom_boxplot() + geom_jitter(position=position_jitter(0.1))+  stat_summary(fun.y=mean, geom="point", shape=18,size=5, color="#E41A1C") + theme_bw() + scale_color_manual(values=pallet)  + theme(legend.position="right") + labs(title="K562", x ="ChIP-seq binding", y = "score", color  = "ChIP-seq binding") + annotate("text", x = 1.6, y = 7.5, label = paste0("t-test p= ",signif(test$p.value,4)))

# egret prior - 0's have lower
test <- t.test(edges_for_valid_k562[which(edges_for_valid_k562$chipseq == 0),]$prior,edges_for_valid_k562[which(edges_for_valid_k562$chipseq == 1),]$prior, alternative="less")

ggplot(edges_for_valid_k562, aes(x = as.factor(chipseq), y = prior, color = as.factor(chipseq), group = as.factor(chipseq)))+ geom_boxplot() + geom_jitter(position=position_jitter(0.1))+  stat_summary(fun.y=mean, geom="point", shape=18,size=5, color="#E41A1C") + theme_bw() + scale_color_manual(values=pallet)  + theme(legend.position="right") + labs(title="K562", x ="ChIP-seq binding", y = "score", color  = "ChIP-seq binding") + annotate("text", x = 1.6, y = 7.5, label = paste0("t-test p= ",signif(test$p.value,4)))

# egret mod - 1's have lower
test <- t.test(edges_for_valid_k562[which(edges_for_valid_k562$chipseq == 0),]$mod,edges_for_valid_k562[which(edges_for_valid_k562$chipseq == 1),]$mod, alternative="greater")

ggplot(edges_for_valid_k562, aes(x = as.factor(chipseq), y = mod, color = as.factor(chipseq), group = as.factor(chipseq)))+ geom_boxplot() + geom_jitter(position=position_jitter(0.1))+  stat_summary(fun.y=mean, geom="point", shape=18,size=5, color="#E41A1C") + theme_bw() + scale_color_manual(values=pallet)  + theme(legend.position="right") + labs(title="K562", x ="ChIP-seq binding", y = "score", color  = "ChIP-seq binding") + annotate("text", x = 1.6, y = 7.5, label = paste0("t-test p= ",signif(test$p.value,4)))

# disrup - 1's have lower
test <- t.test(edges_for_valid_k562[which(edges_for_valid_k562$chipseq == 0),]$disrup,edges_for_valid_k562[which(edges_for_valid_k562$chipseq == 1),]$disrup, alternative="greater")

ggplot(edges_for_valid_k562, aes(x = as.factor(chipseq), y = disrup, color = as.factor(chipseq), group = as.factor(chipseq)))+ geom_boxplot() + geom_jitter(position=position_jitter(0.1))+  stat_summary(fun.y=mean, geom="point", shape=18,size=5, color="#E41A1C") + theme_bw() + scale_color_manual(values=pallet)  + theme(legend.position="right") + labs(title="K562", x ="ChIP-seq binding", y = "score", color  = "ChIP-seq binding") + annotate("text", x = 1.6, y = 7.5, label = paste0("t-test p= ",signif(test$p.value,4)))

```


predict disruption aucs

```{r}
pallet <- c("#870E75"  ,  "#0BB19F" )
color <- "#870E75"
edges_for_valid_gm12878$binding_disrup <- 1 - edges_for_valid_gm12878$chipseq
curve <- evalmod(scores = edges_for_valid_gm12878$disrup, labels = edges_for_valid_gm12878$binding_disrup, posclass = 1)
aucs <- precrec::auc(curve)
autoplot(curve)
auc_gm12878 <- c("GM12878","AU-ROC",round(aucs$aucs[1], digits=4))
prc_gm12878 <- c("GM12878","AU-PR",round(aucs$aucs[2], digits=4))

roc <- as.data.frame(curve$rocs[[1]]$x)
colnames(roc) <- c("x")
roc$y <- curve$rocs[[1]]$y

pr <- as.data.frame(curve$prcs[[1]]$x)
colnames(pr) <- c("x")
pr$y <- curve$prcs[[1]]$y

roccurves_gm12878 <- ggplot(roc, aes(x=x, y=y, color = color)) + geom_line() +  theme_bw() + scale_color_manual(values=c(color)) + xlim(0,1) + ylim(0,1) + geom_abline(intercept = 0, slope = 1, linetype = 2) + labs(title="ROC-disruption (GM12878)", x ="1 - Specificity", y = "Sensitivity")

prcurves_gm12878 <- ggplot(pr, aes(x=x, y=y, color=color)) + geom_line() +  theme_bw() + scale_color_manual(values=c(color)) + xlim(0,1) + ylim(0,1) + labs(title="PR-disruption (GM12878)", x ="Recall", y = "Precision") 

# K562
top_x_table_k562 <- top_n(validation_table_k562, 35, diff_k562)
top_x_table_k562$disrupt <- 1 - top_x_table_k562$chipseq
curve <- evalmod(scores = top_x_table_k562$diff_k562, labels = top_x_table_k562$disrupt, posclass = 1)
autoplot(curve)
aucs <- precrec::auc(curve)
auc_k562<- c("K562","AU-ROC",round(aucs$aucs[1],digits = 4))
prc_k562 <- c("K562","AU-PR",round(aucs$aucs[2],digits = 4))


roc <- as.data.frame(curve$rocs[[1]]$x)
colnames(roc) <- c("x")
roc$y <- curve$rocs[[1]]$y

pr <- as.data.frame(curve$prcs[[1]]$x)
colnames(pr) <- c("x")
pr$y <- curve$prcs[[1]]$y

roccurves_k562 <- ggplot(roc, aes(x=x, y=y, color = color)) + geom_line() +  theme_bw() + scale_color_manual(values=color) + xlim(0,1) + ylim(0,1) + geom_abline(intercept = 0, slope = 1, linetype = 2) + labs(title="ROC-disruption (K562)", x ="1 - Specificity", y = "Sensitivity")

prcurves_k562 <- ggplot(pr, aes(x=x, y=y, color=color)) + geom_line() +  theme_bw() + scale_color_manual(values=color) + xlim(0,1) + ylim(0,1) + labs(title="PR-disruption (K562)", x ="Recall", y = "Precision") 

p3 <- grid.arrange(roccurves_gm12878 + theme(legend.position="hidden"), prcurves_gm12878 + theme(legend.position="none"),roccurves_k562 + theme(legend.position="hidden"), prcurves_k562 + theme(legend.position="none"), nrow=2, widths = c(4,4),heights = c(4,4))

p4 <- annotate_figure(p3,top = text_grob("Performance curves predicting disruption", color = "black", face = "bold", size = 14))
pdf("../figures/disruption_performance_curves_top35_humanModels.pdf",width = 6, height = 4)
plot(p4)
dev.off()
```