---
title: "finalEGRET v1 preprocessing"
author: "Deborah Weighill"
date: "June 17, 2020"
output:
  html_notebook:
    toc: true
    theme: cosmo
    df_print: paged
---

# Description
This notebook preprocesses the data for EGRET to make inputs. It involves mapping motifs to promotor regions, mapping QTLs to promotor regions, making a VCF file for QTLs for running qbic and also a VCF file 

# Load libraries and set tag

```{r}
library(GenomicRanges)
library(dplyr)
library(parallel)

tag <- "finalEGRET_v1_06172020"
```

# Gene annotation

```{r}
# read in the gene annotation file with gene ranges.
genes <- read.table("../annotation/ensembl_genes_from_ucsc.txt", header = TRUE,sep = "\t")

# for UCSC gene annotations, if the gene is on the negative strand the txStart is actually the txEnd. Also with UCSC, txStart needs a +1 because of the 0-based system.
genes$promotorLeft <- ifelse(genes$strand == "+", (genes$txStart + 1 - 750), (genes$txEnd-250))
genes$promotorRight <- ifelse(genes$strand == "+", (genes$txStart + 1 + 250), (genes$txEnd+750))

# make a granges object of these promotor ranges
gr_genes_promotor <- GRanges(seqnames = genes$chrom, ranges = IRanges(start = genes$promotorLeft, end = genes$promotorRight), strand = NULL, mcols = genes[,13:15])
```


# Motif data
Now we load motif ranges for each TF from the bed files, and we overlap them with the gene promotors

## Map motif ranges to genes
```{r}
# make a list of the motif bed files to parse
files <- list.files(path="../../ca_egret/ca_egret_v3.1_LOO/motifs/motif_scan_results", pattern="*.bed", full.names=TRUE, recursive=FALSE)

# apply this function to every file in the list
mclapply(files, function(x) {
  
  # load file
  motif <- read.table(x, header = FALSE, sep = "\t")
  
  # apply function to overlap with gene promotor regions
  name <- strsplit(strsplit(x, "\\/")[[1]][[7]], "\\.")[[1]][[1]]
  motif$motifname <- rep(name, dim(motif)[1])
  gr_motif <- GRanges(seqnames = motif[,1], ranges = IRanges(start = motif[,2], end = motif[,3]), strand = NULL, mcols = motif$motifname)
  motif_promotor_overlap <- data.frame(findOverlaps(gr_motif,gr_genes_promotor))
  motif_promotor_overlap$tf <- gr_motif$mcols[motif_promotor_overlap$queryHits]
  motif_promotor_overlap$chr <- as.vector(seqnames(gr_motif))[motif_promotor_overlap$queryHits]
  motif_promotor_overlap$start <- start(gr_motif)[motif_promotor_overlap$queryHits]
  motif_promotor_overlap$end <- end(gr_motif)[motif_promotor_overlap$queryHits]
  motif_promotor_overlap$gene <- gr_genes_promotor$mcols.name2[motif_promotor_overlap$subjectHits]
  motif_gene_pairs <- distinct(motif_promotor_overlap[,c(3,4,5,6,7)])
  filename <- paste0("../motifs/mapped_motif_files/mapped_motif_promotor_",tag,"_",name,".txt")
  
  # write to file
  write.table(x = motif_gene_pairs, file = filename, quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE)
  }, mc.cores = 30)
```


## Make prior file
```{r}
# make a list of the motif bed files to parse
files <- list.files(path="../motifs/mapped_motif_files", pattern="*.txt", full.names=TRUE, recursive=FALSE)

# apply this function to every file in the list
mclapply(files, function(x) {
  
  # load file
  motif <- read.table(x, header = FALSE, sep = "\t")
  prior_pairs <- unique(motif[,c(1,5)])
  

  name <- strsplit(strsplit(strsplit(x, "\\/")[[1]][[4]], "\\.")[[1]][[1]],"_")[[1]][[7]]
  filename <- paste0("../motifs/unique_motif_prior_files/prior_",tag,"_",name,".txt")
  
  # write to file
  write.table(x = prior_pairs, file = filename, quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE)
  }, mc.cores = 30)
```

```{r}
library(tidyr)
# concatenate the individial tf-gene files
system("cat /home/ubuntu/EGRET/final_egret_v1/motifs/unique_motif_prior_files/* >/home/ubuntu/EGRET/final_egret_v1/prior/prior_motif_true1_pairs_finalEGRET_v1_06172020.txt")

#read in the concatendated file, "fill it out" with zeros making a complete bipartite graph and then save to an RData object.
motif <- read.table("/home/ubuntu/EGRET/final_egret_v1/prior/prior_motif_true1_pairs_finalEGRET_v1_06172020.txt", header = FALSE)
colnames(motif) <- c("tf","gene")
motif$edgeP <- rep(1,nrow(motif))
panda_prior <- complete(motif, tf, gene, fill = list(edgeP = 0))
save(panda_prior, file = "../inputs/panda_motif_prior_finalEGRET_v1.RData")
```


# eQTLs
```{r}
library(tidyr)

# load eQTL data and overlap with motifs
eqtl <- read.table("../pop_data/GTEx_Analysis_v7_eQTL/Cells_EBV-transformed_lymphocytes.v7.signif_variant_gene_pairs.txt", header = 1)
#split the first column to get chromosome, position and ID columns and add the "chr" tag
eqtl <- separate(eqtl, variant_id, c("chr", "pos"), "_", remove = FALSE)
eqtl$chr = paste0("chr",eqtl$chr)
eqtl <- separate(eqtl, gene_id, c("geneID"), "\\.", remove = TRUE)

#Convert the data into a GRanges object
gr_eqtl <- GRanges(seqnames = eqtl$chr, ranges = IRanges(start = as.numeric(eqtl$pos), end = as.numeric(eqtl$pos)), strand = NULL, mcols=eqtl[,c("geneID", "pval_beta","slope","pval_nominal")])

 
  # make motif-eGene network
  motifs <- list.files(path="../motifs/mapped_motif_files/", pattern="*.txt", full.names=TRUE, recursive=FALSE)
  lapply(motifs, function(y) {
    # load file
    motif <- read.table(y, header = FALSE, sep = "\t")
    motif_name <- motif[1,1]
    print(motif_name)
    colnames(motif) <- c("motifName","chr","start","stop","adjacentGene")
    gr_motif <- GRanges(seqnames = motif[,2], ranges = IRanges(start = motif[,3], end = motif[,4]), strand = NULL, mcols = motif[,c(1,5)])
  
    eqtl_motif_overlap <- data.frame(findOverlaps(gr_eqtl,gr_motif))
    eqtl_motif_overlap$tfName <- gr_motif$mcols.motifName[eqtl_motif_overlap$subjectHits]
    eqtl_motif_overlap$adjacentGene <- gr_motif$mcols.adjacentGene[eqtl_motif_overlap$subjectHits]
    eqtl_motif_overlap$eGene <- gr_eqtl$mcols.geneID[eqtl_motif_overlap$queryHits]
    eqtl_motif_overlap$snpPos <- start(gr_eqtl)[eqtl_motif_overlap$queryHits]
    eqtl_motif_overlap$chr <- as.vector(seqnames(gr_eqtl))[eqtl_motif_overlap$queryHits]
    eqtl_motif_overlap$pvalNom <- gr_eqtl$mcols.pval_nominal[eqtl_motif_overlap$queryHits]
    eqtl_motif_overlap$pvalBeta <- gr_eqtl$mcols.pval_beta[eqtl_motif_overlap$queryHits]
    eqtl_motif_overlap$beta <- gr_eqtl$mcols.slope[eqtl_motif_overlap$queryHits]
    indices <- which(as.vector(eqtl_motif_overlap$eGene) == as.vector(eqtl_motif_overlap$adjacentGene))
    eQTL_in_promotor_adjacent_egene <- eqtl_motif_overlap[indices,]
  
    filename <- paste0("../inputs/eQTL_in_motif_promotor_adjacent_egene_finalEGRET_v1_06172020.txt")
    # write to file
    write.table(x = eQTL_in_promotor_adjacent_egene[,c(3,4,6,7,10)], file = filename, quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)
  })
```


# Make vcf files for qbic

## VCF for all QTLs in motifs in promotors where the vcf file (genotype(s)) has an alternate allele. 
Thus, we don't do qbic on those where there is no alternate allele. Do this for K562 and GM12878.


First GM12878:

```{r}
motif_promotor_qtls <- read.table("../inputs/eQTL_in_motif_promotor_adjacent_egene_finalEGRET_v1_06172020.txt", header = FALSE)
colnames(motif_promotor_qtls) <- c("tf","gene","snpPos","chrom","effect")
motif_promotor_qtls$chromPos <- paste0(motif_promotor_qtls$chrom,"_",motif_promotor_qtls$snpPos)

genotype <- read.table("../genotype_data/NA12878.vcf", sep = "\t" ,header = FALSE)
colnames(genotype) <- c("CHROM", "POS","ID", "REF", "ALT", "QUAL", "FILTER", "INFO", "FORMAT", "GENO")
genotype$chromPos <- paste0(genotype$CHROM,"_",genotype$POS)
  
#select QTLs where there is a SNP (alternate allele) in the vcf file
genotype_qtls <- motif_promotor_qtls[which(motif_promotor_qtls$chromPos  %in% genotype$chromPos),]
# fill in the reference and alternate alleles
genotype_qtls$ref <- genotype$REF[match(genotype_qtls$chromPos,genotype$chromPos)]
genotype_qtls$alt <- genotype$ALT[match(genotype_qtls$chromPos,genotype$chromPos)]
# remove indels
motif_promotor_qtls_noIndels <- genotype_qtls[(nchar(as.character(genotype_qtls$ref)) == 1) & (nchar(as.character(genotype_qtls$alt)) == 1),c(4,3,2,7,8)]
  
qtl_vcf <- paste0("../genotype_data/eQTLs_motif_promotor_finalEGRET_v1_06172020_GM12878.vcf")
write.table(unique(motif_promotor_qtls_noIndels), file =qtl_vcf, col.names = FALSE, sep = "\t", quote = FALSE, row.names = FALSE)
```

Now K562:

```{r}
motif_promotor_qtls <- read.table("../inputs/eQTL_in_motif_promotor_adjacent_egene_finalEGRET_v1_06172020.txt", header = FALSE)
colnames(motif_promotor_qtls) <- c("tf","gene","snpPos","chrom","effect")
motif_promotor_qtls$chromPos <- paste0(motif_promotor_qtls$chrom,"_",motif_promotor_qtls$snpPos)

genotype <- read.table("../genotype_data/k562.vcf", sep = "\t" ,header = FALSE)
colnames(genotype) <- c("CHROM", "POS","ID", "REF", "ALT", "QUAL", "FILTER", "INFO", "FORMAT", "GENO")
genotype$chromPos <- paste0(genotype$CHROM,"_",genotype$POS)
  
#select QTLs where there is a SNP (alternate allele) in the vcf file
genotype_qtls <- motif_promotor_qtls[which(motif_promotor_qtls$chromPos  %in% genotype$chromPos),]
# fill in the reference and alternate alleles
genotype_qtls$ref <- genotype$REF[match(genotype_qtls$chromPos,genotype$chromPos)]
genotype_qtls$alt <- genotype$ALT[match(genotype_qtls$chromPos,genotype$chromPos)]
# remove indels
motif_promotor_qtls_noIndels <- genotype_qtls[(nchar(as.character(genotype_qtls$ref)) == 1) & (nchar(as.character(genotype_qtls$alt)) == 1),c(4,3,2,7,8)]
  
qtl_vcf <- paste0("../genotype_data/eQTLs_motif_promotor_finalEGRET_v1_06172020_K562.vcf")
write.table(unique(motif_promotor_qtls_noIndels), file =qtl_vcf, col.names = FALSE, sep = "\t", quote = FALSE, row.names = FALSE)
```

Now let's combine the vcfs (non-redundantly) for K562 and GM12878 so that we only have to run qbic once.
```{r}
system("cat /home/ubuntu/EGRET/final_egret_v1/genotype_data/eQTLs_motif_promotor_finalEGRET_v1_06172020_K562.vcf /home/ubuntu/EGRET/final_egret_v1/genotype_data/eQTLs_motif_promotor_finalEGRET_v1_06172020_GM12878.vcf | sort -u >/home/ubuntu/EGRET/final_egret_v1/genotype_data/unique_eQTLs_motif_promotor_finalEGRET_v1_06172020_GM12878_K562.vcf")
```


## VCF for all SNPs in motifs in promotors in vcf file
First GM12878:
```{r}
library(tidyr)
library(GenomicRanges)
# Now load the genotype data 
genotype <- read.table("../genotype_data/NA12878.vcf", sep = "\t", header = FALSE)
colnames(genotype) <- c("chr","pos","ID","ref","alt","QUAL",	"FILTER",	"INFO",	"FORMAT",	"NA12878")
genotype_gr <-  GRanges(seqnames = genotype$chr, ranges = IRanges(start = genotype$pos, end = genotype$pos), strand = NULL, mcols = genotype[,4:5])


#map to motifs
  motifs <- list.files(path="../motifs/mapped_motif_files/", pattern="*.txt", full.names=TRUE, recursive=FALSE)
  lapply(motifs, function(y) {
    # load file
    motif <- read.table(y, header = FALSE, sep = "\t")
    motif_name <- motif[1,1]
    #print(motif_name)
    colnames(motif) <- c("motifName","chr","start","stop","adjacentGene")
    gr_motif <- GRanges(seqnames = motif[,2], ranges = IRanges(start = motif[,3], end = motif[,4]), strand = NULL, mcols = motif[,c(1,5)])
  
    genotype_motif_overlap <- data.frame(findOverlaps(genotype_gr,gr_motif))
    genotype_motif_overlap$tfName <- gr_motif$mcols.motifName[genotype_motif_overlap$subjectHits]
    genotype_motif_overlap$adjacentGene <- gr_motif$mcols.adjacentGene[genotype_motif_overlap$subjectHits]
    genotype_motif_overlap$chr <- as.vector(seqnames(genotype_gr))[genotype_motif_overlap$queryHits]
    genotype_motif_overlap$snpPos <- start(genotype_gr)[genotype_motif_overlap$queryHits]
    genotype_motif_overlap$ref <- genotype_gr$mcols.ref[genotype_motif_overlap$queryHits]
    genotype_motif_overlap$alt <- genotype_gr$mcols.alt[genotype_motif_overlap$queryHits]  
    genotype_motif_overlap_noIndels <- genotype_motif_overlap[(nchar(as.character(genotype_motif_overlap$ref)) == 1) & (nchar(as.character(genotype_motif_overlap$alt)) == 1),c(5,6,4,7,8)]
    filename <- paste0("../genotype_data/NA12878_snps_in_motifs_in_promotor_finalEGRET_v1_06172020.txt")
    write.table(x = genotype_motif_overlap_noIndels, file = filename, quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)
})
```

Now K562:
```{r}
library(tidyr)
library(GenomicRanges)
# Now load the genotype data 
genotype <- read.table("../genotype_data/k562.vcf", sep = "\t", header = FALSE)
colnames(genotype) <- c("chr","pos","ID","ref","alt","QUAL",	"FILTER",	"INFO",	"FORMAT",	"NA12878")
genotype_gr <-  GRanges(seqnames = genotype$chr, ranges = IRanges(start = genotype$pos, end = genotype$pos), strand = NULL, mcols = genotype[,4:5])


#map to motifs
  motifs <- list.files(path="../motifs/mapped_motif_files/", pattern="*.txt", full.names=TRUE, recursive=FALSE)
  lapply(motifs, function(y) {
    # load file
    motif <- read.table(y, header = FALSE, sep = "\t")
    motif_name <- motif[1,1]
    #print(motif_name)
    colnames(motif) <- c("motifName","chr","start","stop","adjacentGene")
    gr_motif <- GRanges(seqnames = motif[,2], ranges = IRanges(start = motif[,3], end = motif[,4]), strand = NULL, mcols = motif[,c(1,5)])
  
    genotype_motif_overlap <- data.frame(findOverlaps(genotype_gr,gr_motif))
    genotype_motif_overlap$tfName <- gr_motif$mcols.motifName[genotype_motif_overlap$subjectHits]
    genotype_motif_overlap$adjacentGene <- gr_motif$mcols.adjacentGene[genotype_motif_overlap$subjectHits]
    genotype_motif_overlap$chr <- as.vector(seqnames(genotype_gr))[genotype_motif_overlap$queryHits]
    genotype_motif_overlap$snpPos <- start(genotype_gr)[genotype_motif_overlap$queryHits]
    genotype_motif_overlap$ref <- genotype_gr$mcols.ref[genotype_motif_overlap$queryHits]
    genotype_motif_overlap$alt <- genotype_gr$mcols.alt[genotype_motif_overlap$queryHits]  
    genotype_motif_overlap_noIndels <- genotype_motif_overlap[(nchar(as.character(genotype_motif_overlap$ref)) == 1) & (nchar(as.character(genotype_motif_overlap$alt)) == 1),c(5,6,4,7,8)]
    filename <- paste0("../genotype_data/K562_snps_in_motifs_in_promotor_finalEGRET_v1_06172020.txt")
    write.table(x = genotype_motif_overlap_noIndels, file = filename, quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)
})
```

Uniquefy SNPs so that we don't run qbic more than we have to. Some SNPs appear more than once (even assigned to the same gene id) because they are mapping to multiple motifs.
```{r}
system("cat /home/ubuntu/EGRET/final_egret_v1/genotype_data/K562_snps_in_motifs_in_promotor_finalEGRET_v1_06172020.txt /home/ubuntu/EGRET/final_egret_v1/genotype_data/NA12878_snps_in_motifs_in_promotor_finalEGRET_v1_06172020.txt | sort -u >/home/ubuntu/EGRET/final_egret_v1/genotype_data/NA12878_K562_snps_in_motifs_in_promotor_finalEGRET_v1_06172020.vcf")
```


# Make expression matrix

```{r}
library(data.table)
#let's extract our LCL sample names
system("grep Cells /home/ubuntu/EGRET/final_egret_v1/pop_data/GTEx_v7_Annotations_SampleAttributesDS.txt | grep EBV | cut -f1 >/home/ubuntu/EGRET/final_egret_v1/pop_data/LCL_samples")

expr <- fread("../pop_data/GTEx_Analysis_2016-01-15_v7_RNASeQCv1.1.8_gene_tpm.gct", header = TRUE, skip = 2, data.table = TRUE)
lcl_samples <-read.table("../pop_data/LCL_samples", header = FALSE)
expr$Name<-sub("\\.[0-9]","", expr$Name)
lcl_expr <- expr[,union("Name", intersect(lcl_samples[1:149,], colnames(expr))), with=FALSE]
lcl_expr_df <- data.frame(lcl_expr, row.names = 1)
clean_data <- lcl_expr_df[(rowSums(lcl_expr_df != 0) > 50),]
save(clean_data, file = "../inputs/expr_data_finalEGRET_v1_06172020.RData")
```

# Map allele-specific expression (ASE) SNPs to motifs in promotors

Get the data (downloaded 09/01/2020)

wget https://genome.cshlp.org/content/suppl/2018/10/17/gr.237354.118.DC1/Supplemental_Table_S1_.txt
```{r}
# load ase data
ase <- read.table("../ase/Supplemental_Table_S1_.txt", sep = " ", header = TRUE)
ase_gr <- GRanges(seqnames = ase$chr, ranges = IRanges(start = ase$pos1, end = ase$pos1), strand = NULL, mcols = ase[,])


# make motif-eGene network
  motifs <- list.files(path="../motifs/mapped_motif_files/", pattern="*.txt", full.names=TRUE, recursive=FALSE)
  lapply(motifs, function(y) {
    # load file
    motif <- read.table(y, header = FALSE, sep = "\t")
    motif_name <- motif[1,1]
    print(motif_name)
    colnames(motif) <- c("motifName","chr","start","stop","adjacentGene")
    gr_motif <- GRanges(seqnames = motif[,2], ranges = IRanges(start = motif[,3], end = motif[,4]), strand = NULL, mcols = motif[,c(1,5)])
    
  # overlap bitbundle snps with motifs 
  ase_in_motif <- data.frame(findOverlaps(ase_gr, gr_motif))
  ase_in_motif$tf <- gr_motif$mcols.motifName[ase_in_motif$subjectHits]
  ase_in_motif$gene <- gr_motif$mcols.adjacentGene[ase_in_motif$subjectHits]
  ase_in_motif$fdr <- ase_gr$mcols.padj_comb[ase_in_motif$queryHits]
  ase_in_motif$pval <- ase_gr$mcols.p_comb[ase_in_motif$queryHits]
  ase_in_motif$chr <- ase_gr$mcols.chr[ase_in_motif$queryHits]
  ase_in_motif$pos <- ase_gr$mcols.pos1[ase_in_motif$queryHits]
  write.table(ase_in_motif[,c(3:8)], file = "../ase/ase_mapped_motifs_09012020.txt", sep = "\t", quote = FALSE, append = TRUE, row.names =FALSE, col.names = FALSE)
  })
```

Map only to promotor regions
```{r}
# load ase data
ase <- read.table("../ase/Supplemental_Table_S1_.txt", sep = " ", header = TRUE)
ase_gr <- GRanges(seqnames = ase$chr, ranges = IRanges(start = ase$pos1, end = ase$pos1), strand = NULL, mcols = ase[,])


# map to promotor
    
  # overlap bitbundle snps with motifs 
  ase_in_promotor <- data.frame(findOverlaps(ase_gr, gr_genes_promotor))
  ase_in_promotor$gene <- gr_genes_promotor$mcols.name2[ase_in_promotor$subjectHits]
  ase_in_promotor$fdr <- ase_gr$mcols.padj_comb[ase_in_promotor$queryHits]
  ase_in_promotor$pval <- ase_gr$mcols.p_comb[ase_in_promotor$queryHits]
  ase_in_promotor$chr <- ase_gr$mcols.chr[ase_in_promotor$queryHits]
  ase_in_promotor$pos <- ase_gr$mcols.pos1[ase_in_promotor$queryHits]
  write.table(unique(ase_in_promotor[,c(3:7)]), file = "../ase/ase_mapped_promotor_09032020.txt", sep = "\t", quote = FALSE, append = TRUE, row.names =FALSE, col.names = FALSE)
```

# Map allele-specific binding (ASB) SNPs for NFKB1 to promotors
```{r}
asb <- read.table("../ase/Supplemental_Table_S11.txt", header = TRUE)

map <- read.table("../ase/Supplemental_Table_S1_.txt", header = TRUE, sep = " ")

asb$pos <- map$pos1[match(asb$rsID, map$rsID)]
asb$chr <- map$chr[match(asb$rsID, map$rsID)]
mapped <- asb[which(!(is.na(asb$pos))),]
asb_gr <- GRanges(seqnames = mapped$chr, ranges = IRanges(start = mapped$pos, end = mapped $pos), strand = NULL, mcols =  mapped[,])


asb_in_promotor <- data.frame(findOverlaps(asb_gr, gr_genes_promotor))
  asb_in_promotor$gene <- gr_genes_promotor$mcols.name2[asb_in_promotor$subjectHits]
  asb_in_promotor$fdr <- asb_gr$mcols.padj[asb_in_promotor$queryHits]
  asb_in_promotor$pval <- asb_gr$mcols.p_all[asb_in_promotor$queryHits]
  asb_in_promotor$chr <- asb_gr$mcols.chr[asb_in_promotor$queryHits]
    asb_in_promotor$pos <- asb_gr$mcols.pos[asb_in_promotor$queryHits]
  write.table(unique(asb_in_promotor[,]), file = "../ase/asb_mapped_promotor_09042020.txt", sep = "\t", quote = FALSE, append = FALSE, row.names =FALSE, col.names = FALSE)
```

# Map caQTLs to promotor regions
```{r}
caQTL <- read.table("../caqtl/cht_results_full_LCL.txt", header = TRUE)

  caqtl_gr <- GRanges(seqnames = caQTL$TEST.SNP.CHROM, ranges = IRanges(start = caQTL$TEST.SNP.POS, end = caQTL$TEST.SNP.POS), strand = NULL, mcols =  caQTL[,])


caqtl_in_promotor <- data.frame(findOverlaps(caqtl_gr, gr_genes_promotor))
  caqtl_in_promotor$gene <- gr_genes_promotor$mcols.name2[caqtl_in_promotor$subjectHits]
  caqtl_in_promotor$pval <- caqtl_gr$mcols.P.VALUE[caqtl_in_promotor$queryHits]
  caqtl_in_promotor$chr <- caqtl_gr$mcols.TEST.SNP.CHROM[caqtl_in_promotor$queryHits]
    caqtl_in_promotor$pos <- caqtl_gr$mcols.TEST.SNP.POS[caqtl_in_promotor$queryHits]
  write.table(unique(caqtl_in_promotor[,]), file = "../caqtl/caqtl_mapped_promotor_09092020.txt", sep = "\t", quote = FALSE, append = FALSE, row.names =FALSE, col.names = FALSE)
```

# map caqtl to motifs in promotors
```{r}
caQTL <- read.table("../caqtl/cht_results_full_LCL.txt", header = TRUE)
caqtl_gr <- GRanges(seqnames = caQTL$TEST.SNP.CHROM, ranges = IRanges(start = caQTL$TEST.SNP.POS, end = caQTL$TEST.SNP.POS), strand = NULL, mcols =  caQTL[,])

# make motif-eGene network
  motifs <- list.files(path="../motifs/mapped_motif_files/", pattern="*.txt", full.names=TRUE, recursive=FALSE)
  lapply(motifs, function(y) {
    # load file
    motif <- read.table(y, header = FALSE, sep = "\t")
    motif_name <- motif[1,1]
    print(motif_name)
    colnames(motif) <- c("motifName","chr","start","stop","adjacentGene")
    gr_motif <- GRanges(seqnames = motif[,2], ranges = IRanges(start = motif[,3], end = motif[,4]), strand = NULL, mcols = motif[,c(1,5)])
    
  # overlap bitbundle snps with motifs 
  caqtl_in_motif <- data.frame(findOverlaps(caqtl_gr, gr_motif))
  caqtl_in_motif$tf <- gr_motif$mcols.motifName[caqtl_in_motif$subjectHits]
  caqtl_in_motif$gene <- gr_motif$mcols.adjacentGene[caqtl_in_motif$subjectHits]
  caqtl_in_motif$chr <- caqtl_gr$mcols.TEST.SNP.CHROM[caqtl_in_motif$queryHits]
    caqtl_in_motif$pos <- caqtl_gr$mcols.TEST.SNP.POS[caqtl_in_motif$queryHits]
  caqtl_in_motif$pval <- caqtl_gr$mcols.P.VALUE[caqtl_in_motif$queryHits]
  write.table(caqtl_in_motif[,], file = "../caqtl/caqtl_mapped_motifs_09092020.txt", sep = "\t", quote = FALSE, append = TRUE, row.names =FALSE, col.names = FALSE)
  })
```