---
title: "Run finalEgret v1 - GM12878"
author: "Deborah Weighill"
date: "June 19, 2020"
output:
  html_notebook:
    toc: true
    theme: cosmo
    df_print: paged
---
___
# Description
___

Running finalEgret v1 (*with only QTL data, i.e. no qbic, no alternate allele count, no genotype*) on GM12878.

# Libraries and tags
```{r libraries_and_tags}
Sys.setenv(RSTUDIO_PANDOC="/usr/lib/rstudio-server/bin/pandoc")

# load the libraries we will need.
library(GenomicRanges)
library(IRanges)
library(tidyr)
library(data.table)
library(dplyr)
library(pandaR)

# this tag should indicate the genotype and the date
tag <- "finalEGRET_v1_onlyQTL_na12878_06192020"
```

# Inputs

Individual inputs: leave out for this LOO
#```{r individual_inputs}

#individual qbic results for caQTL and eQTL - these are qbic results of qtls which are (1) in motifs within the promotor region of genes (2) for eQTLs, the eGene is the adjacent gene, for caQTLs, the accessibility region overlaps the motif, (3) the vcf file of the individual(s) in question contains the QTL as an alternate allele and (4) the qbic effect is negative, is predicted from human pbms and predicts a change in binding.
qbic <- read.table(file = "../inputs/qbic_qtls_gm12878_caEGRET_v3.1_06112020.txt", header = FALSE)

  
# individual genotype
vcf <- read.table("../genotype_data/NA12878.vcf", header = FALSE, sep = "\t")
#```

General inputs:
```{r general_inputs}

# general motif prior
motif <- load("../inputs/panda_motif_prior_finalEGRET_v1.RData")

# population-level expression data
expr <- load("../inputs/expr_data_finalEGRET_v1_06172020.RData")

# general ppi prior
ppi <- read.table("../inputs/ppi.txt", header = FALSE, sep = "\t")

# population-level QTL data
qtl <- read.table("../inputs/eQTL_in_motif_promotor_adjacent_egene_finalEGRET_v1_06172020.txt", header = FALSE)

```

## PPI prior
Load and filter the ppi prior (only include proteins whose genes are actually expressed.)
```{r filter_ppi}
# filter ppi prior to only include transcripts that are expressed
nameGeneMap <- read.table("../annotation/geneID_name_map.txt", header = FALSE)
colnames(nameGeneMap) <- c("gene","name")
#names of expressed genes
expressedGeneNames <- nameGeneMap$name[which(nameGeneMap$gene %in% rownames(clean_data))]
#only include edges in ppi where both of the tfs are expressed (i.e. in the above list)
ppiFiltered <- ppi[((ppi$V1 %in% expressedGeneNames) & (ppi$V2 %in% expressedGeneNames)),]
```

## QTL data
```{r parse_qtl}
#these QTLs are those which exist within motifs within promotors.
colnames(qtl) <- c("tf",	"gene", "snpPos",	"chr",	"effect")
qtl$snpID <- paste0(qtl$chr,"_",qtl$snpPos)
```

## qbic data for QTLs - leave out for this LOO
##```{r parse_qbic_eqtl}
colnames(qbic) <- c("snpID",	"tf",	"gene",	"qbicEffectSize","qbicEffect")

# aggregate the qbic data per SNP -> function to return the maximum effect size of a qbic SNP 
maxabs <- function(x){
  location <- which(abs(x) == max(abs(x)))
  return(x[location])
}
qbic_uniq <- distinct(qbic)
qbic_uniq$catid <- paste0(qbic_uniq$snpID,qbic_uniq$tf,qbic_uniq$gene,qbic_uniq$qbicEffectSize)

qbic_ag <- aggregate(qbic_uniq$qbicEffectSize, by = list(qbic_uniq$snpID,qbic_uniq$tf,qbic_uniq$gene), FUN = maxabs)
colnames(qbic_ag) <- c("snpID","tf","gene","qbicEffectSize")
qbic_ag$catid <- paste0(qbic_ag$snpID,qbic_ag$tf,qbic_ag$gene,qbic_ag$qbicEffectSize)
qbic_ag$qbicEffect <- qbic_uniq$qbicEffect[match(qbic_ag$catid,qbic_uniq$catid)]

#absQbicEffectSize - absolute value of the qbic effect (which is negative)
qbic_ag$absQbicEffectSize <- abs(qbic_ag$qbicEffectSize)
#```


## Calculating alt allele count for each qbic SNP.
Label each eqtl snp with the alternate allele count for the genotype in question (here NA12878)
#```{r parse_genotype}
colnames(vcf) <- c("CHROM",  "POS"  ,   "ID"  ,    "REF"  ,   "ALT"   ,  "QUAL"   , "FILTER" , "INFO"   , "FORMAT", "NA12878")
snp_ids <- paste0(vcf$CHROM,"_",vcf$POS)
rownames(vcf) <- snp_ids
vcf <- separate(vcf, NA12878, c("allele1", "allele2"), "\\|", remove = TRUE)
vcf$alt_allele_count <- as.numeric(vcf$allele1) + as.numeric(vcf$allele2)
vcf$snp_id <- snp_ids

#assign alt allele count of the individual to the qbic table
qbic_ag$alt_allele_count <- vcf$alt_allele_count[match(qbic_ag$snpID, vcf$snp_id)]

#assign alt allele count of the individual to the QTL table
qtl$alt_allele_count <- vcf$alt_allele_count[match(qtl$snpID, vcf$snp_id)]
QTL_tf_gene_pairs <- distinct(qtl[,c(1:7)])
QTL_tf_gene_pairs$edgeE <- rep(1,nrow(QTL_tf_gene_pairs))
QTL_tf_gene_pairs$alt_allele_count[is.na(QTL_tf_gene_pairs$alt_allele_count)] <- 0
QTL_tf_gene_pairs$qtlTF <- paste0(QTL_tf_gene_pairs$tf,QTL_tf_gene_pairs$snpID)
#```


# Prior Modifiers

## Construct modifiers

Now we need to annotate each QTL with its qbic effect (some will be zero as they have no qbic effect.) Add the qbic effects to the qtl table.

```{r combine_qbic_eqtl}
QTL_tf_gene_pairs <- distinct(qtl[,c(1:6)])
#qbic_ag$snpTF <- paste0(qbic_ag$tf,qbic_ag$snpID)
# we can match like this because the qbic effect of a SNP on a tf is the same no matter what nearby gene it is assigned to.
#QTL_tf_gene_pairs$qbicEffectSize <- qbic_ag$qbicEffectSize[match(QTL_tf_gene_pairs$qtlTF, qbic_ag$snpTF)]
#QTL_tf_gene_pairs$qbicEffect <- qbic_ag$qbicEffect[match(QTL_tf_gene_pairs$qtlTF, qbic_ag$snpTF)]
#QTL_tf_gene_pairs$absQbicEffectSize <- qbic_ag$absQbicEffectSize[match(QTL_tf_gene_pairs$qtlTF, qbic_ag$snpTF)]
#QTL_tf_gene_pairs[is.na(QTL_tf_gene_pairs)] <- 0

# now we make our "per SNP" combinations (i.e. products of eQTL and qbic variables)
QTL_tf_gene_pairs$absQtlQffect <- abs(QTL_tf_gene_pairs$effect)

```

Now we have a matrix of QTLs (and their associated TF-gene edge) as well as their qbic effect (which can be zero if the qbic effect of the SNP was not significant for that TF) and the combinations of QTL and qbic data.

Now, aggregate the eqtl/caqtl/qbic data per tf-gene pair into a matrix of modifiers. 

```{r aggregate_modifiers}
#make an edge id of TG-gene - this will be what we aggregate by
QTL_tf_gene_pairs$edgeID <- paste0(QTL_tf_gene_pairs$tf,QTL_tf_gene_pairs$gene)

#this is where we will store the modifiers aggregated by tf-gene edge
mods <- unique(QTL_tf_gene_pairs[,c(1,2)])
mods$edgeID <- paste0(mods$tf,mods$gene)

# aggregated modifiers from qbic and qtl variables by tf-gene pair
absQtlQffect <- aggregate(absQtlQffect ~ edgeID, data=QTL_tf_gene_pairs, FUN=sum)


# put into mods matrix
mods$absQtlQffect <- absQtlQffect$absQtlQffect[match(mods$edgeID, absQtlQffect$edgeID)]
```


## Construct priors

E = Eqtl in edge
A = Alt allele count
qes = qbic effect size
qe = qbic effect
qb = qbic binary
B = beta

Merge the panda prior with the modifier matrix
```{r combine_panda_and_mods}
combined <- merge(as.data.frame(panda_prior),mods, all.x=TRUE, by.x = c(1,2),by.y = c(1,2))
combined[is.na(combined)] <- 0
```

Make the three priors by subtracting the modifier from the prior edge.
```{r make_priors}
# construct the egret prior
combined$egretPrior <- combined$edgeP - combined$absQtlQffect

egretPrior <- combined[,c(1,2,6)]

priorfile <- paste0("../outputs/priors_",tag,".txt")
write.table(combined, file = priorfile, col.names = TRUE, row.names = FALSE, sep = "\t", quote = FALSE)
```


# Run PANDA Message Passing 

```{r run_message_passing}
processes <- c(1)
prior_list <- list(egretPrior)
prior_names <- c("egret")

mclapply(processes, function(p) {
  system.time(results <- panda(as.data.frame(prior_list[p]), expr=clean_data, ppi=ppiFiltered,progress=TRUE, remove.missing.ppi = TRUE, remove.missing.motif = TRUE, remove.missing.genes = TRUE))
  filename <- paste0("../outputs/",tag,"_",prior_names[p],".RData")
  regnet <- results@regNet
  save(regnet, file = filename)
d}, mc.cores = 1,mc.preschedule=TRUE)
```
