---
title: "Run EGRET with only eQTL modifiers - GM12878"
author: "Deborah Weighill"
output:
  html_notebook:
    toc: true
    theme: cosmo
    df_print: paged
---
___
# Description
___

Running EGRET (*with only QTL data, i.e. no qbic, no alternate allele count, no genotype*) on GM12878.

# Libraries and tags
```{r libraries_and_tags}
Sys.setenv(RSTUDIO_PANDOC="/usr/lib/rstudio-server/bin/pandoc")

# load the libraries we will need.
library(GenomicRanges)
library(IRanges)
library(tidyr)
library(data.table)
library(dplyr)
library(pandaR)

# this tag should indicate the genotype and the date
tag <- "finalEGRET_v1_onlyQTL_gm12878"
```

# Inputs

General inputs:
```{r general_inputs}

# general motif prior
motif <- load("../inputs/panda_motif_prior_finalEGRET_v1.RData")

# population-level expression data
expr <- load("../inputs/expr_data_finalEGRET_v1_06172020.RData")

# general ppi prior
ppi <- read.table("../inputs/ppi.txt", header = FALSE, sep = "\t")

# population-level QTL data
qtl <- read.table("../inputs/eQTL_in_motif_promotor_adjacent_egene_finalEGRET_v1_06172020.txt", header = FALSE)

```

## PPI prior
Load and filter the ppi prior (only include proteins whose genes are actually expressed.)
```{r filter_ppi}
# filter ppi prior to only include transcripts that are expressed
nameGeneMap <- read.table("../annotation/geneID_name_map.txt", header = FALSE)
colnames(nameGeneMap) <- c("gene","name")
#names of expressed genes
expressedGeneNames <- nameGeneMap$name[which(nameGeneMap$gene %in% rownames(clean_data))]
#only include edges in ppi where both of the tfs are expressed (i.e. in the above list)
ppiFiltered <- ppi[((ppi$V1 %in% expressedGeneNames) & (ppi$V2 %in% expressedGeneNames)),]
```

## QTL data
```{r parse_qtl}
#these QTLs are those which exist within motifs within promotors.
colnames(qtl) <- c("tf",	"gene", "snpPos",	"chr",	"effect")
qtl$snpID <- paste0(qtl$chr,"_",qtl$snpPos)
```

# Prior Modifiers

## Construct modifiers

Now we need to annotate each QTL with its qbic effect (some will be zero as they have no qbic effect.) Add the qbic effects to the qtl table.

```{r combine_qbic_eqtl}
QTL_tf_gene_pairs <- distinct(qtl[,c(1:6)])
QTL_tf_gene_pairs$absQtlQffect <- abs(QTL_tf_gene_pairs$effect)
```


Now, aggregate the eqtl/caqtl/qbic data per tf-gene pair into a matrix of modifiers. 

```{r aggregate_modifiers}
#make an edge id of TG-gene - this will be what we aggregate by
QTL_tf_gene_pairs$edgeID <- paste0(QTL_tf_gene_pairs$tf,QTL_tf_gene_pairs$gene)

#this is where we will store the modifiers aggregated by tf-gene edge
mods <- unique(QTL_tf_gene_pairs[,c(1,2)])
mods$edgeID <- paste0(mods$tf,mods$gene)

# aggregated modifiers from qbic and qtl variables by tf-gene pair
absQtlQffect <- aggregate(absQtlQffect ~ edgeID, data=QTL_tf_gene_pairs, FUN=sum)


# put into mods matrix
mods$absQtlQffect <- absQtlQffect$absQtlQffect[match(mods$edgeID, absQtlQffect$edgeID)]
```


## Construct priors

E = Eqtl in edge
A = Alt allele count
qes = qbic effect size
qe = qbic effect
qb = qbic binary
B = beta

Merge the panda prior with the modifier matrix
```{r combine_panda_and_mods}
combined <- merge(as.data.frame(panda_prior),mods, all.x=TRUE, by.x = c(1,2),by.y = c(1,2))
combined[is.na(combined)] <- 0
```

Make the three priors by subtracting the modifier from the prior edge.
```{r make_priors}
# construct the egret prior
combined$egretPrior <- combined$edgeP - combined$absQtlQffect

egretPrior <- combined[,c(1,2,6)]

priorfile <- paste0("../outputs/priors_",tag,".txt")
write.table(combined, file = priorfile, col.names = TRUE, row.names = FALSE, sep = "\t", quote = FALSE)
```


# Run PANDA Message Passing 

```{r run_message_passing}
processes <- c(1)
prior_list <- list(egretPrior)
prior_names <- c("egret")

mclapply(processes, function(p) {
  system.time(results <- panda(as.data.frame(prior_list[p]), expr=clean_data, ppi=ppiFiltered,progress=TRUE, remove.missing.ppi = TRUE, remove.missing.motif = TRUE, remove.missing.genes = TRUE))
  filename <- paste0("../outputs/",tag,"_",prior_names[p],".RData")
  regnet <- results@regNet
  save(regnet, file = filename)
d}, mc.cores = 1,mc.preschedule=TRUE)
```
