---
title: "Run EGRET with only QBiC modifiers -  GM12878"
author: "Deborah Weighill"
output:
  html_notebook:
    toc: true
    theme: cosmo
    df_print: paged
---
___
# Description
___

Running finalEgret v1 (*with only qbic on alternate alleles*) on GM12878.

# Libraries and tags
```{r libraries_and_tags}
Sys.setenv(RSTUDIO_PANDOC="/usr/lib/rstudio-server/bin/pandoc")

# load the libraries we will need.
library(GenomicRanges)
library(IRanges)
library(tidyr)
library(data.table)
library(dplyr)
library(pandaR)

# this tag should indicate the genotype and the date
tag <- "finalEgret_v1_onlyQbic_gm12878"
```

# Inputs

Individual inputs:
```{r individual_inputs}
qbic <- read.table(file = "../inputs/qbic_allSNPS_GM12878_K562.txt", header = FALSE)

# individual genotype
vcf <- read.table("../genotype_data/NA12878.vcf", header = FALSE, sep = "\t")
```

General inputs:
```{r general_inputs}

# general motif prior
motif <- load("../inputs/panda_motif_prior_finalEGRET_v1.RData")

# population-level expression data
expr <- load("../inputs/expr_data_finalEGRET_v1_06172020.RData")

# general ppi prior
ppi <- read.table("../inputs/ppi.txt", header = FALSE, sep = "\t")
```

## PPI prior
Load and filter the ppi prior (only include proteins whose genes are actually expressed.)
```{r filter_ppi}
# filter ppi prior to only include transcripts that are expressed
nameGeneMap <- read.table("../annotation/geneID_name_map.txt", header = FALSE)
colnames(nameGeneMap) <- c("gene","name")
#names of expressed genes
expressedGeneNames <- nameGeneMap$name[which(nameGeneMap$gene %in% rownames(clean_data))]
#only include edges in ppi where both of the tfs are expressed (i.e. in the above list)
ppiFiltered <- ppi[((ppi$V1 %in% expressedGeneNames) & (ppi$V2 %in% expressedGeneNames)),]
```


## qbic data for SNPs
```{r parse_qbic_eqtl}
colnames(qbic) <- c("snpID",	"tf",	"gene",	"qbicEffectSize","qbicEffect")

# aggregate the qbic data per SNP -> function to return the maximum effect size of a qbic SNP 
maxabs <- function(x){
  location <- which(abs(x) == max(abs(x)))
  return(x[location])
}
qbic_uniq <- distinct(qbic)
qbic_uniq$catid <- paste0(qbic_uniq$snpID,qbic_uniq$tf,qbic_uniq$gene,qbic_uniq$qbicEffectSize)

qbic_ag <- aggregate(qbic_uniq$qbicEffectSize, by = list(qbic_uniq$snpID,qbic_uniq$tf,qbic_uniq$gene), FUN = maxabs)
colnames(qbic_ag) <- c("snpID","tf","gene","qbicEffectSize")
qbic_ag$catid <- paste0(qbic_ag$snpID,qbic_ag$tf,qbic_ag$gene,qbic_ag$qbicEffectSize)
qbic_ag$qbicEffect <- qbic_uniq$qbicEffect[match(qbic_ag$catid,qbic_uniq$catid)]

#absQbicEffectSize - absolute value of the qbic effect (which is negative)
qbic_ag$absQbicEffectSize <- abs(qbic_ag$qbicEffectSize)
```


## Calculating alt allele count for each qbic SNP.
Label each eqtl snp with the alternate allele count for the genotype in question (here NA12878)
```{r parse_genotype}
colnames(vcf) <- c("CHROM",  "POS"  ,   "ID"  ,    "REF"  ,   "ALT"   ,  "QUAL"   , "FILTER" , "INFO"   , "FORMAT", "NA12878")
snp_ids <- paste0(vcf$CHROM,"_",vcf$POS)
rownames(vcf) <- snp_ids
vcf <- separate(vcf, NA12878, c("allele1", "allele2"), "\\|", remove = TRUE)
vcf$alt_allele_count <- as.numeric(vcf$allele1) + as.numeric(vcf$allele2)
vcf$snp_id <- snp_ids

#assign alt allele count of the individual to the qbic table
qbic_ag$alt_allele_count <- vcf$alt_allele_count[match(qbic_ag$snpID, vcf$snp_id)]

```


# Prior Modifiers

## Construct modifiers

Now we need to make per-snp combinations of qbic and alt allele count

```{r combine_qbic_eqtl}

qbic_ag$altAlleleCount_qbicEffect <- qbic_ag$alt_allele_count * qbic_ag$qbicEffect

```



Now, aggregate the qbic data per tf-gene pair into a matrix of modifiers. 

```{r aggregate_modifiers}
#make an edge id of TG-gene - this will be what we aggregate by
qbic_ag$edgeID <- paste0(qbic_ag$tf,qbic_ag$gene)

#this is where we will store the modifiers aggregated by tf-gene edge
mods <- unique(qbic_ag[,c(2,3)])
mods$edgeID <- paste0(mods$tf,mods$gene)

# aggregated modifiers from qbic variables by tf-gene pair
altAlleleCount_qbicEffect <- aggregate(altAlleleCount_qbicEffect ~ edgeID, data=qbic_ag, FUN=sum)


# put into mods matrix
mods$altAlleleCount_qbicEffect <- altAlleleCount_qbicEffect$altAlleleCount_qbicEffect[match(mods$edgeID, altAlleleCount_qbicEffect$edgeID)]
```


## Construct priors

E = Eqtl in edge
A = Alt allele count
qes = qbic effect size
qe = qbic effect
qb = qbic binary
B = beta

Merge the panda prior with the modifier matrix
```{r combine_panda_and_mods}
combined <- merge(as.data.frame(panda_prior),mods, all.x=TRUE, by.x = c(1,2),by.y = c(1,2))
combined[is.na(combined)] <- 0
```

Make the three priors by subtracting the modifier from the prior edge.
```{r make_priors}
# construct the egret prior
combined$egretPrior <- combined$edgeP - combined$altAlleleCount_qbicEffect

egretPrior <- combined[,c(1,2,6)]

priorfile <- paste0("../outputs/priors_",tag,".txt")
write.table(combined, file = priorfile, col.names = TRUE, row.names = FALSE, sep = "\t", quote = FALSE)
```


# Run PANDA Message Passing 

```{r run_message_passing}
processes <- c(1)
prior_list <- list(egretPrior)
prior_names <- c("egret")

mclapply(processes, function(p) {
  system.time(results <- panda(as.data.frame(prior_list[p]), expr=clean_data, ppi=ppiFiltered,progress=TRUE, remove.missing.ppi = TRUE, remove.missing.motif = TRUE, remove.missing.genes = TRUE))
  filename <- paste0("../outputs/",tag,"_",prior_names[p],".RData")
  regnet <- results@regNet
  save(regnet, file = filename)
d}, mc.cores = 1,mc.preschedule=TRUE)
```
