---
title: "Plots for GM12878 and K562 - human models threhsolded at 1e-10"
author: "Deborah Weighill"
date: "July 3, 2020"
output:
  html_notebook:
    toc: true
    theme: cosmo
    df_print: paged
---



# Load validation tables for K562 and GM12878 for no PPI/noExp versions
```{r}
library(plyr)
# GM12878
load("../outputs/validation_table_finalEgret_v1_GM12878_06182020.RData")
table_gm12878 <- validation_table
colnames(table_gm12878) <- c("id","tf","gene","panda","egret_gm12878","chipseq_gm12878","diff_gm12878")

# K562
load("../outputs/validation_table_finalEgret_v1_K562_human-10_07032020.RData")
table_k562 <- validation_table
colnames(table_k562) <- c("id","tf","gene","panda","egret_k562","chipseq_k562","diff_k562")
combined_table <- join(table_gm12878,table_k562,by = c("id","tf","gene"),type="inner")
```

# Plot of distributions of differences

# Select examples for visualizing in genome browser

```{r}
combined_table[which((combined_table$chipseq_k562 != combined_table$chipseq_gm12878) & (combined_table$diff_k562 > 0.5)),c(1:11)]
```




```{r}
library(ggpubr)
g <- table_gm12878[which(table_gm12878$diff_gm12878 > 0.25),]

p <- ggboxplot(g, x = "chipseq_gm12878", y = "diff_gm12878", color = "chipseq_gm12878", palette = "Dark2", add = "jitter") + stat_compare_means(method = "t.test", method.args = list(alternative = "two.sided"))  + theme(legend.position="right") + ggtitle(paste0("Header "))
p

```

```{r}
g <- table_k562[which(table_k562$diff_k562>0.25),]

p <- ggboxplot(g, x = "chipseq_k562", y = "diff_k562", color = "chipseq_k562", palette = "Dark2", add = "jitter") + stat_compare_means(method = "t.test", method.args = list(alternative = "two.sided"))  + theme(legend.position="right") + ggtitle(paste0("Header "))
p


```

```{r}
combined_table[which(combined_table$diff_gm12878 > 0.3),]
```





# Box plots of edge differences for top difference edges
```{r}
library(plyr)
# GM12878
load("../outputs/validation_table_finalEgret_v1_GM12878_06182020.RData")
table_gm12878 <- validation_table
colnames(table_gm12878) <- c("id","tf","gene","panda","egret_gm12878","chipseq_gm12878","diff_gm12878")

# K562
load("../outputs/validation_table_finalEgret_v1_K562_06182020.RData")
table_k562 <- validation_table
colnames(table_k562) <- c("id","tf","gene","panda","egret_k562","chipseq_k562","diff_k562")
combined_table <- join(table_gm12878,table_k562,by = c("id","tf","gene"),type="inner")
```

Make a box plot showing the distribution in egret edge weights for top diff edges. Also show distribution of differences, each time grouped by chipseq +/-. Remember, top difference here is not chosen as top difference in chipseq or anything, its chosen as top difference between EGRET and PANDA. We're trying to isolate the effect of the variants.

```{r}
library(ggpubr)
library(dplyr)

g <- top_n(table_gm12878, 35, diff_gm12878)
p <- ggboxplot(g, x = "chipseq_gm12878", y = "diff_gm12878", color = "chipseq_gm12878", palette = c("royalblue3", "firebrick2"), add = "jitter") + stat_compare_means(method = "t.test", method.args = list(alternative = "two.sided"))  + theme(legend.position="right") + labs(title="Top 35 disruption scores - GM12878", x ="ChIP-seq binding", y = "Edge Disruption Score") 
pdf("gm12878_top35_diffvalues_boxplot.pdf",width = 5, height = 3.5)
plot(p)
dev.off()
```


```{r}
g <- top_n(table_k562, 35, diff_k562)
p <- ggboxplot(g, x = "chipseq_k562", y = "diff_k562", color = "chipseq_k562", palette = c("royalblue3", "firebrick2"), add = "jitter") + stat_compare_means(method = "t.test", method.args = list(alternative = "two.sided"))  + theme(legend.position="right") + labs(title="Top 35 disruption scores - K562", x ="ChIP-seq binding", y = "Edge Disruption Score") 
pdf("k562_top35_diffvalues_boxplot.pdf",width = 5, height = 3.5)
plot(p)
dev.off()
```


# Now plot AUCs/AU-PRs for top35 edges
```{r}
library(precrec)
library(gridExtra)
library(pROC)
top_x_table_gm12878 <- top_n(table_gm12878, 35, diff_gm12878)[,c(4,5,6)]
scores <- join_scores(top_x_table_gm12878[,1],top_x_table_gm12878[,2])
labels <- join_labels(top_x_table_gm12878[,3],top_x_table_gm12878[,3])
curve_data <- mmdata(scores, labels,modnames = c("panda", "egret"))
curves <- evalmod(curve_data)
#autoplot(curves)
#curves
#auc(curves)

rocP <- roc(top_x_table_gm12878$chipseq_gm12878, top_x_table_gm12878$panda, auc = TRUE, direction = "<")
rocE <- roc(top_x_table_gm12878$chipseq_gm12878, top_x_table_gm12878$egret_gm12878,auc = TRUE, direction = "<")
test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
pandaAUC <- test$estimate["AUC of roc1"]
egretAUC <- test$estimate["AUC of roc2"]
pval <- round(test$p.value,digits = 4)

pandaROC <- as.data.frame(curves$rocs[[1]]$x)
colnames(pandaROC) <- c("x")
pandaROC$y <- curves$rocs[[1]]$y
pandaROC$Method <- "PANDA"

egretROC <- as.data.frame(curves$rocs[[2]]$x)
colnames(egretROC) <- c("x")
egretROC$y <- curves$rocs[[2]]$y
egretROC$Method <- "EGRET"

roc_data <- rbind(pandaROC, egretROC)
roccurves <- ggplot(roc_data, aes(x=x, y=y, color=Method, group = Method)) + geom_line() +  theme_bw(base_size = 15) + scale_color_manual(values=c("royalblue1","tomato1")) + xlim(0,1) + ylim(0,1) + geom_abline(intercept = 0, slope = 1, linetype = 2) + labs(title="ROC curves - GM12878", x ="1 - Specificity", y = "Sensitivity") + annotate("text", x = 0.3, y = 0.8, label = paste0("Delong p: ",pval))


pandaPR <- as.data.frame(curves$prcs[[1]]$x)
colnames(pandaPR) <- c("x")
pandaPR$y <- curves$prcs[[1]]$y
pandaPR$Method <- "PANDA"

egretPR <- as.data.frame(curves$prcs[[2]]$x)
colnames(egretPR) <- c("x")
egretPR$y <- curves$prcs[[2]]$y
egretPR$Method <- "EGRET"

pr_data <- rbind(pandaPR, egretPR)
prcurves <- ggplot(pr_data, aes(x=x, y=y, color=Method, group = Method)) + geom_line() +  theme_bw(base_size = 15) + scale_color_manual(values=c("royalblue1","tomato1")) + xlim(0,1) + ylim(0,1) + labs(title="PR curves - GM12878", x ="Recall", y = "Precision") 
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend<-g_legend(prcurves)


p3 <- grid.arrange(arrangeGrob(roccurves + theme(legend.position="none"), prcurves + theme(legend.position="none"), nrow=1, widths = c(4,4)), mylegend, nrow=1, widths = c(8,2), newpage = FALSE)
pdf("gm12878_top35_curves.pdf",width = 9, height = 3)
plot(p3)
dev.off()
```


```{r}
library(precrec)
library(gridExtra)
top_x_table_k562 <- top_n(table_k562, 17, diff_k562)[,c(4,5,6)]
scores <- join_scores(top_x_table_k562[,1],top_x_table_k562[,2])
labels <- join_labels(top_x_table_k562[,3],top_x_table_k562[,3])
curve_data <- mmdata(scores, labels,modnames = c("panda", "egret"))
curves <- evalmod(curve_data)
#autoplot(curves)
#curves
#auc(curves)

rocP <- roc(top_x_table_k562$chipseq_k562, top_x_table_k562$panda, auc = TRUE, direction = "<")
rocE <- roc(top_x_table_k562$chipseq_k562, top_x_table_k562$egret_k562,auc = TRUE, direction = "<")
test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
pandaAUC <- test$estimate["AUC of roc1"]
egretAUC <- test$estimate["AUC of roc2"]
pval <- round(test$p.value,digits = 4)

pandaROC <- as.data.frame(curves$rocs[[1]]$x)
colnames(pandaROC) <- c("x")
pandaROC$y <- curves$rocs[[1]]$y
pandaROC$Method <- "PANDA"

egretROC <- as.data.frame(curves$rocs[[2]]$x)
colnames(egretROC) <- c("x")
egretROC$y <- curves$rocs[[2]]$y
egretROC$Method <- "EGRET"

roc_data <- rbind(pandaROC, egretROC)
roccurves <- ggplot(roc_data, aes(x=x, y=y, color=Method, group = Method)) + geom_line() +  theme_bw(base_size = 15) + scale_color_manual(values=c("royalblue1","tomato1")) + xlim(0,1) + ylim(0,1) + geom_abline(intercept = 0, slope = 1, linetype = 2) + labs(title="ROC curves - K562", x ="1 - Specificity", y = "Sensitivity") +annotate("text", x = 0.2, y = 0.8, label = paste0("Delong p: ",pval))


pandaPR <- as.data.frame(curves$prcs[[1]]$x)
colnames(pandaPR) <- c("x")
pandaPR$y <- curves$prcs[[1]]$y
pandaPR$Method <- "PANDA"

egretPR <- as.data.frame(curves$prcs[[2]]$x)
colnames(egretPR) <- c("x")
egretPR$y <- curves$prcs[[2]]$y
egretPR$Method <- "EGRET"

pr_data <- rbind(pandaPR, egretPR)
prcurves <- ggplot(pr_data, aes(x=x, y=y, color=Method, group = Method)) + geom_line() +  theme_bw(base_size = 15) + scale_color_manual(values=c("royalblue1","tomato1")) + xlim(0,1) + ylim(0,1) + labs(title="PR curves - K562", x ="Recall", y = "Precision") 
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend<-g_legend(prcurves)


p3 <- grid.arrange(arrangeGrob(roccurves + theme(legend.position="none"), prcurves + theme(legend.position="none"), nrow=1, widths = c(4,4)), mylegend, nrow=1, widths = c(8,2), newpage = FALSE)
pdf("k562_top17_curves.pdf",width = 9, height = 3)
plot(p3)
dev.off()
```



# Now show that difference scores are predictive of disruption with AUCs
```{r}
top_x_table_gm12878 <- top_n(table_gm12878, 35, diff_gm12878)
top_x_table_gm12878$disrupt <- 1 - top_x_table_gm12878$chipseq
curve <- evalmod(scores = top_x_table_gm12878$diff_gm12878, labels = top_x_table_gm12878$disrupt, posclass = 1)
autoplot(curve)

roc <- as.data.frame(curve$rocs[[1]]$x)
colnames(roc) <- c("x")
roc$y <- curve$rocs[[1]]$y


pr <- as.data.frame(curve$prcs[[1]]$x)
colnames(pr) <- c("x")
pr$y <- curve$prcs[[1]]$y

roccurves <- ggplot(roc, aes(x=x, y=y, color = "darkorange1")) + geom_line() +  theme_bw(base_size = 15) + scale_color_manual(values=c("darkorange1")) + xlim(0,1) + ylim(0,1) + geom_abline(intercept = 0, slope = 1, linetype = 2) + labs(title="ROC-disruption (GM12878)", x ="1 - Specificity", y = "Sensitivity")

prcurves <- ggplot(pr, aes(x=x, y=y, color="darkorange1")) + geom_line() +  theme_bw(base_size = 15) + scale_color_manual(values=c("darkorange1")) + xlim(0,1) + ylim(0,1) + labs(title="PR-disruption (GM12878)", x ="Recall", y = "Precision") 
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend<-g_legend(prcurves)


p3 <- grid.arrange(arrangeGrob(roccurves + theme(legend.position="none"), prcurves + theme(legend.position="none"), nrow=1, widths = c(4,4)), newpage = FALSE)
pdf("gm12878_top35_disruptionprediction_curves.pdf",width = 9, height = 3)
plot(p3)
dev.off()

```



```{r}
top_x_table_k562 <- top_n(table_k562, 35, diff_k562)
top_x_table_k562$disrupt <- 1 - top_x_table_k562$chipseq
curve <- evalmod(scores = top_x_table_k562$diff_k562, labels = top_x_table_k562$disrupt, posclass = 1)
autoplot(curve)

roc <- as.data.frame(curve$rocs[[1]]$x)
colnames(roc) <- c("x")
roc$y <- curve$rocs[[1]]$y


pr <- as.data.frame(curve$prcs[[1]]$x)
colnames(pr) <- c("x")
pr$y <- curve$prcs[[1]]$y

roccurves <- ggplot(roc, aes(x=x, y=y, color = "darkorange1")) + geom_line() +  theme_bw(base_size = 15) + scale_color_manual(values=c("darkorange1")) + xlim(0,1) + ylim(0,1) + geom_abline(intercept = 0, slope = 1, linetype = 2) + labs(title="ROC-disruption (K562)", x ="1 - Specificity", y = "Sensitivity")

prcurves <- ggplot(pr, aes(x=x, y=y, color="darkorange1")) + geom_line() +  theme_bw(base_size = 15) + scale_color_manual(values=c("darkorange1")) + xlim(0,1) + ylim(0,1) + labs(title="PR-disruption (K562)", x ="Recall", y = "Precision") 
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend<-g_legend(prcurves)


p3 <- grid.arrange(arrangeGrob(roccurves + theme(legend.position="none"), prcurves + theme(legend.position="none"), nrow=1, widths = c(4,4)), newpage = FALSE)
pdf("k562_top35_disruptionprediction_curves.pdf",width = 9, height = 3)
plot(p3)
dev.off()
```