---
title: "Plots for GM12878 and K562"
author: "Deborah Weighill"
date: "June 25, 2020"
output:
  html_notebook:
    toc: true
    theme: cosmo
    df_print: paged
---

# Load libraries
```{r}
library(xtable)
library(jcolors)
library(precrec)
library(dplyr)
library(pROC)
pallet <- c("#870E75"  ,  "#0BB19F")
#pallet2<- c("slateblue3"  ,  "darkgoldenrod1")
library(viridis)
library(lemon)
library(ggplot2)
```

# Load validation tables for K562 and GM12878 for no PPI/noExp versions
```{r}
# GM12878
load("../outputs/validation_table_finalEgret_v1_GM12878_06182020.RData")
table_gm12878 <- validation_table
colnames(table_gm12878) <- c("id","tf","gene","panda","egret_gm12878","chipseq_gm12878","diff_gm12878")

# K562
load("../outputs/validation_table_finalEgret_v1_K562_06182020.RData")
table_k562 <- validation_table
colnames(table_k562) <- c("id","tf","gene","panda","egret_k562","chipseq_k562","diff_k562")
combined_table <- join(table_gm12878,table_k562,by = c("id","tf","gene"),type="inner")
```

# Calculate AUCs, pvals and print to latex table

## K562 threshold
```{r,results='hide',fig.show=TRUE}
names <- c("egret_k562")
diff_cols <- c(7)
egret_net_nums <- c(5)
panda_net_nums <- c(4)
chipseq_col <- 6
get_roc_for_diff_thresh <- function(egretNum, thresh) {
  diffcolnum <- diff_cols[egretNum]
  egretcol <- egret_net_nums[egretNum]
  panda_col <- panda_net_nums[egretNum]
  
  top_diff_table <- table_k562[(table_k562[,c(diffcolnum)]>thresh),c(panda_col,egretcol,chipseq_col)]
  scores <- join_scores(top_diff_table[,1],top_diff_table[,2])
  labels <- join_labels(top_diff_table[,3],top_diff_table[,3])
  specs <- paste0(names[egretNum],", thresh: ",thresh)
  
  rocP <- roc(top_diff_table$chipseq, top_diff_table[,1], auc = TRUE, direction = "<")
  rocE <- roc(top_diff_table$chipseq, top_diff_table[,2],auc = TRUE, direction = "<")
  testD <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
  pandaAUC <- testD$estimate["AUC of roc1"]
  egretAUC <- testD$estimate["AUC of roc2"]
  pvalD <- testD$p.value
  print(paste(egretNum,thresh,pvalD))
  
  curve_data <- mmdata(scores, labels,modnames = c("panda", specs), dsids = c(1, 2))
  curves <- evalmod(curve_data)
  nn <- attr(curves$rocs,"data_info")$nn[1]
  np <- attr(curves$rocs,"data_info")$np[1]
  result <- c(egretNum, names[egretNum], thresh, nn, np, pandaAUC, egretAUC, pvalD)
  return(result)
  autoplot(curves)
}

egret_priors <- c(1)
thresholds <- c(0.1, 0.15, 0.2, 0.25, 0.3, 0.35, 0.4, 0.45, 0.5, 0.55, 0.6, 0.65, 0.7, 0.75, 0.8, 0.85, 0.9,1,2,3,4)

egret_auc_pvals<- data.frame()

for (p in egret_priors){
  for (t in thresholds){
    result <- get_roc_for_diff_thresh(p,t)
    df <- data.frame(t(result), stringsAsFactors = FALSE)
    egret_auc_pvals <- rbind(egret_auc_pvals,df)
  }
}

#filename <- paste0("../outputs/AUC_pvals_.txt")
colnames(egret_auc_pvals) <- c("egretNum", "name", "thresh", "nn", "np", "pandaAUC", "egretAUC", "pvalD")
#write.table(egret_auc_pvals, file = filename, sep = "\t", col.names = TRUE, quote=FALSE)
egret_auc_pvals
egret_auc_pvals_table <- egret_auc_pvals[,c(3,4,5)]
egret_auc_pvals_table$pandaAUC <- signif(as.numeric(egret_auc_pvals$pandaAUC, digits = 4))
egret_auc_pvals_table$egretAUC <- signif(as.numeric(egret_auc_pvals$egretAUC, digits = 4))
egret_auc_pvals_table$pvalue <- signif(as.numeric(egret_auc_pvals$pvalD, digits = 4))
print(xtable(egret_auc_pvals_table, type = "latex"), file = "pval_table_allHumanModels_thresh_k562.tex", include.rownames=FALSE)
```

## K562 top_n
```{r,results='hide',fig.show=TRUE}
names <- c("egret_k562")
diff_cols <- c(7)
egret_net_nums <- c(5)
panda_net_nums <- c(4)
chipseq_col <- 6
get_roc_for_top_x <- function(egretNum, x) {
  egretcol <- egret_net_nums[egretNum]
  panda_col <- panda_net_nums[egretNum]
  
  top_x_table <- top_n(table_k562,x,diff_k562)[,c(panda_col,egretcol,chipseq_col)] 
  scores <- join_scores(top_x_table[,1],top_x_table[,2])
  labels <- join_labels(top_x_table[,3],top_x_table[,3])
  specs <- paste0(names[egretNum],", top ",x)
  
  rocP <- roc(top_x_table$chipseq, top_x_table[,1], auc = TRUE, direction = "<")
  rocE <- roc(top_x_table$chipseq, top_x_table[,2],auc = TRUE, direction = "<")
  test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
  pandaAUC <- test$estimate["AUC of roc1"]
  egretAUC <- test$estimate["AUC of roc2"]
  pval <- test$p.value
  print(paste(egretNum,x,pval))
  
  curve_data <- mmdata(scores, labels,modnames = c("panda", specs), dsids = c(1, 2))
  curves <- evalmod(curve_data)
  nn <- attr(curves$rocs,"data_info")$nn[1]
  np <- attr(curves$rocs,"data_info")$np[1]
  result <- c(egretNum, names[egretNum], x, nn, np, pandaAUC, egretAUC, pval)
  return(result)
  autoplot(curves)
}

egret_priors <- c(1)
numEdges <- c(10,15,20,25,30,35,40,45,50,55,60,65,70,75,80,85,90,95,100)

egret_auc_pvals_topn<- data.frame()

for (p in egret_priors){
  for (x in numEdges){
    result <- get_roc_for_top_x(p,x)
    df <- data.frame(t(result), stringsAsFactors = FALSE)
    egret_auc_pvals_topn <- rbind(egret_auc_pvals_topn,df)
  }
}

#filename <- paste0("../outputs/AUC_pvals_topN_",tag,".txt")
colnames(egret_auc_pvals_topn) <- c("egretNum", "name", "topN", "nn", "np", "pandaAUC", "egretAUC", "pval")
#write.table(egret_auc_pvals_topn, file = filename, sep = "\t", col.names = TRUE, quote=FALSE)
egret_auc_pvals_topn
egret_auc_pvals_table <- egret_auc_pvals_topn[,c(3,4,5)]
egret_auc_pvals_table$pandaAUC <- signif(as.numeric(egret_auc_pvals_topn$pandaAUC), digits = 5)
egret_auc_pvals_table$egretAUC <- signif(as.numeric(egret_auc_pvals_topn$egretAUC), digits = 5)
egret_auc_pvals_table$pvalue <- signif(as.numeric(egret_auc_pvals_topn$pval), digits = 5)
print(xtable(egret_auc_pvals_table, type = "latex", digits = 5), file= "pval_table_allHumanModels_topn_k562.tex",include.rownames=FALSE)
```


## GM12878 threshold
```{r,results='hide',fig.show=TRUE}
names <- c("egret_GM12878")
diff_cols <- c(7)
egret_net_nums <- c(5)
panda_net_nums <- c(4)
chipseq_col <- 6
get_roc_for_diff_thresh <- function(egretNum, thresh) {
  diffcolnum <- diff_cols[egretNum]
  egretcol <- egret_net_nums[egretNum]
  panda_col <- panda_net_nums[egretNum]
  
  top_diff_table <- table_gm12878[(table_gm12878[,c(diffcolnum)]>thresh),c(panda_col,egretcol,chipseq_col)]
  scores <- join_scores(top_diff_table[,1],top_diff_table[,2])
  labels <- join_labels(top_diff_table[,3],top_diff_table[,3])
  specs <- paste0(names[egretNum],", thresh: ",thresh)
  
  rocP <- roc(top_diff_table$chipseq, top_diff_table[,1], auc = TRUE, direction = "<")
  rocE <- roc(top_diff_table$chipseq, top_diff_table[,2],auc = TRUE, direction = "<")
  test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
  pandaAUC <- test$estimate["AUC of roc1"]
  egretAUC <- test$estimate["AUC of roc2"]
  pval <- test$p.value
  print(paste(egretNum,thresh,pval))
  
  curve_data <- mmdata(scores, labels,modnames = c("panda", specs), dsids = c(1, 2))
  curves <- evalmod(curve_data)
  nn <- attr(curves$rocs,"data_info")$nn[1]
  np <- attr(curves$rocs,"data_info")$np[1]
  result <- c(egretNum, names[egretNum], thresh, nn, np, pandaAUC, egretAUC, pval)
  return(result)
  autoplot(curves)
}

egret_priors <- c(1)
thresholds <- c(0.1, 0.15, 0.2, 0.25, 0.3, 0.35, 0.4, 0.45, 0.5, 0.55, 0.6, 0.65, 0.7, 0.75, 0.8, 0.85, 0.9, 1)

egret_auc_pvals<- data.frame()

for (p in egret_priors){
  for (t in thresholds){
    result <- get_roc_for_diff_thresh(p,t)
    df <- data.frame(t(result), stringsAsFactors = FALSE)
    egret_auc_pvals <- rbind(egret_auc_pvals,df)
  }
}

#filename <- paste0("../outputs/AUC_pvals_",tag,".txt")
colnames(egret_auc_pvals) <- c("egretNum", "name", "thresh", "nn", "np", "pandaAUC", "egretAUC", "pval")
#write.table(egret_auc_pvals, file = filename, sep = "\t", col.names = TRUE, quote=FALSE)
egret_auc_pvals
print(xtable(egret_auc_pvals[,c(3:8)], type = "latex"), file = "pval_table_allHumanModels_thresh_gm12878.tex",include.rownames=FALSE)
```

# Make some tables
## GM12878 top_n

```{r,results='hide',fig.show=TRUE}
names <- c("egret_GM12878")
diff_cols <- c(7)
egret_net_nums <- c(5)
panda_net_nums <- c(4)
chipseq_col <- 6
get_roc_for_top_x <- function(egretNum, x) {
  egretcol <- egret_net_nums[egretNum]
  panda_col <- panda_net_nums[egretNum]
  
  top_x_table <- top_n(table_gm12878,x,diff_gm12878)[,c(panda_col,egretcol,chipseq_col)] 
  scores <- join_scores(top_x_table[,1],top_x_table[,2])
  labels <- join_labels(top_x_table[,3],top_x_table[,3])
  specs <- paste0(names[egretNum],", top ",x)
  
  rocP <- roc(top_x_table$chipseq, top_x_table[,1], auc = TRUE, direction = "<")
  rocE <- roc(top_x_table$chipseq, top_x_table[,2],auc = TRUE, direction = "<")
  test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
  pandaAUC <- test$estimate["AUC of roc1"]
  egretAUC <- test$estimate["AUC of roc2"]
  pval <- test$p.value
  print(paste(egretNum,x,pval))
  
  curve_data <- mmdata(scores, labels,modnames = c("panda", specs), dsids = c(1, 2))
  curves <- evalmod(curve_data)
  nn <- attr(curves$rocs,"data_info")$nn[1]
  np <- attr(curves$rocs,"data_info")$np[1]
  result <- c(egretNum, names[egretNum], x, nn, np, pandaAUC, egretAUC, pval)
  return(result)
  autoplot(curves)
}

egret_priors <- c(1)
numEdges <- c(10,15,20,25,30,35,40,45,50,55,60,65,70,75,80,85,90,95,100)

egret_auc_pvals_topn<- data.frame()

for (p in egret_priors){
  for (x in numEdges){
    result <- get_roc_for_top_x(p,x)
    df <- data.frame(t(result), stringsAsFactors = FALSE)
    egret_auc_pvals_topn <- rbind(egret_auc_pvals_topn,df)
  }
}

#filename <- paste0("../outputs/AUC_pvals_topN_",tag,".txt")
colnames(egret_auc_pvals_topn) <- c("egretNum", "name", "topN", "nn", "np", "pandaAUC", "egretAUC", "pval")
#write.table(egret_auc_pvals_topn, file = filename, sep = "\t", col.names = TRUE, quote=FALSE)
egret_auc_pvals_topn
egret_auc_pvals_table <- egret_auc_pvals_topn[,c(3,4,5)]
egret_auc_pvals_table$pandaAUC <- signif(as.numeric(egret_auc_pvals_topn$pandaAUC), digits = 5)
egret_auc_pvals_table$egretAUC <- signif(as.numeric(egret_auc_pvals_topn$egretAUC), digits = 5)
egret_auc_pvals_table$pvalue <- signif(as.numeric(egret_auc_pvals_topn$pval), digits = 5)
print(xtable(egret_auc_pvals_table, type = "latex", digits = 5), file = "pval_table_allHumanModels_topn_gm12878.tex",include.rownames=FALSE,)
```

# Select examples for visualizing in genome browser

```{r}
combined_table[which((combined_table$chipseq_k562 != combined_table$chipseq_gm12878) & (combined_table$diff_k562 > 0.5)),c(1:11)]
```




```{r}
library(ggpubr)
g <- table_gm12878[which(table_gm12878$diff_gm12878 > 0.25),]

p <- ggboxplot(g, x = "chipseq_gm12878", y = "diff_gm12878", color = "chipseq_gm12878", palette = "Dark2", add = "jitter") + stat_compare_means(method = "t.test", method.args = list(alternative = "two.sided"))  + theme(legend.position="right") + ggtitle(paste0("Header "))
p

```

```{r}
g <- table_k562[which(table_k562$diff_k562>0.25),]

p <- ggboxplot(g, x = "chipseq_k562", y = "diff_k562", color = "chipseq_k562", palette = "Dark2", add = "jitter") + stat_compare_means(method = "t.test", method.args = list(alternative = "two.sided"))  + theme(legend.position="right") + ggtitle(paste0("Header "))
p


```

```{r}
combined_table[which(combined_table$diff_gm12878 > 0.3),]
```





# Box plots of edge differences for top difference edges
```{r}
library(plyr)
# GM12878
load("../outputs/validation_table_finalEgret_v1_GM12878_06182020.RData")
table_gm12878 <- validation_table
colnames(table_gm12878) <- c("id","tf","gene","panda","egret_gm12878","chipseq_gm12878","diff_gm12878")

# K562
load("../outputs/validation_table_finalEgret_v1_K562_06182020.RData")
table_k562 <- validation_table
colnames(table_k562) <- c("id","tf","gene","panda","egret_k562","chipseq_k562","diff_k562")

#combined_table <- join(table_gm12878,table_k562,by = c("id","tf","gene"),type="inner")

#save(combined_table,file = "../outputs/combined_table_finalEgret_v1_K562_gm12878_06182020.RData")
#load("../outputs/combined_table_finalEgret_v1_K562_gm12878_06182020.RData")
```

Make a box plot showing the distribution in egret edge weights for top diff edges. Also show distribution of differences, each time grouped by chipseq +/-. Remember, top difference here is not chosen as top difference in chipseq or anything, its chosen as top difference between EGRET and PANDA. We're trying to isolate the effect of the variants.
c("#570D32", "#E48F1B")



```{r}
library(ggpubr)
library(dplyr)

g <- top_n(table_gm12878, 35, diff_gm12878)

test <- t.test(g[which(g$chipseq_gm12878 == 0),]$diff_gm12878,g[which(g$chipseq_gm12878 == 1),]$diff_gm12878, alternative="greater")


p <- ggplot(g, aes(x = as.factor(chipseq_gm12878), y = diff_gm12878, color = as.factor(chipseq_gm12878), group = as.factor(chipseq_gm12878)))+ geom_boxplot() + geom_jitter(position=position_jitter(0.1))+  stat_summary(fun.y=mean, geom="point", shape=18,size=5, color="#E41A1C") + theme_bw() + scale_color_manual(values=pallet)  + theme(legend.position="right") + labs(title="Top 35 disruption scores - GM12878", x ="ChIP-seq binding", y = "Edge disruption score", color  = "ChIP-seq binding") + annotate("text", x = 1.6, y = 7.5, label = paste0("t-test p= ",signif(test$p.value,4)))
pdf("gm12878_top35_diffvalues_boxplot.pdf",width = 5, height = 3.5)
plot(p)
dev.off()
```


```{r}
g <- top_n(table_k562, 35, diff_k562)

test <- t.test(g[which(g$chipseq_k562 == 0),]$diff_k562,g[which(g$chipseq_k562 == 1),]$diff_k562, alternative="greater")


p <- ggplot(g, aes(x = as.factor(chipseq_k562), y = diff_k562, color = as.factor(chipseq_k562), group = as.factor(chipseq_k562)))+ geom_boxplot() + geom_jitter(position=position_jitter(0.1))+  stat_summary(fun.y=mean, geom="point", shape=18,size=5, color="#E41A1C") + theme_bw() + scale_color_manual(values=pallet)  + theme(legend.position="right") + labs(title="Top 35 disruption scores - K562", x ="ChIP-seq binding", y = "Edge disruption score", color  = "ChIP-seq binding") + annotate("text", x = 1.7, y = 10, label = paste0("t-test p= ",signif(test$p.value,4)))
pdf("k562_top35_diffvalues_boxplot.pdf",width = 5, height = 3.5)
plot(p)
dev.off()



```


# Now plot AUCs/AU-PRs for top35 edges
```{r}
#pallet <- c("#009ACD","#FF4500")
#pallet <- c("#08585A","#D5006A")

library(precrec)
library(gridExtra)
library(pROC)
top_x_table_gm12878 <- top_n(table_gm12878, 35, diff_gm12878)[,c(4,5,6)]
scores <- join_scores(top_x_table_gm12878[,1],top_x_table_gm12878[,2])
labels <- join_labels(top_x_table_gm12878[,3],top_x_table_gm12878[,3])
curve_data <- mmdata(scores, labels,modnames = c("panda", "egret"))
curves <- evalmod(curve_data)
#autoplot(curves)
#curves
#auc(curves)

rocP <- roc(top_x_table_gm12878$chipseq_gm12878, top_x_table_gm12878$panda, auc = TRUE, direction = "<")
rocE <- roc(top_x_table_gm12878$chipseq_gm12878, top_x_table_gm12878$egret_gm12878,auc = TRUE, direction = "<")
test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
pandaAUC <- test$estimate["AUC of roc1"]
egretAUC <- test$estimate["AUC of roc2"]
pval <- round(test$p.value,digits = 4)

pandaROC <- as.data.frame(curves$rocs[[1]]$x)
colnames(pandaROC) <- c("x")
pandaROC$y <- curves$rocs[[1]]$y
pandaROC$Method <- "PANDA"

egretROC <- as.data.frame(curves$rocs[[2]]$x)
colnames(egretROC) <- c("x")
egretROC$y <- curves$rocs[[2]]$y
egretROC$Method <- "EGRET"

roc_data <- rbind(pandaROC, egretROC)
roccurves <- ggplot(roc_data, aes(x=x, y=y, color=Method, group = Method)) + geom_line() +  theme_bw(base_size = 15) + scale_color_manual(values=pallet) + xlim(0,1) + ylim(0,1) + geom_abline(intercept = 0, slope = 1, linetype = 2) + labs(title="ROC curves - GM12878", x ="1 - Specificity", y = "Sensitivity") + annotate("text", x = 0.3, y = 0.8, label = paste0("Delong p: ",pval)) 


pandaPR <- as.data.frame(curves$prcs[[1]]$x)
colnames(pandaPR) <- c("x")
pandaPR$y <- curves$prcs[[1]]$y
pandaPR$Method <- "PANDA"

egretPR <- as.data.frame(curves$prcs[[2]]$x)
colnames(egretPR) <- c("x")
egretPR$y <- curves$prcs[[2]]$y
egretPR$Method <- "EGRET"

pr_data <- rbind(pandaPR, egretPR)
prcurves <- ggplot(pr_data, aes(x=x, y=y, color=Method, group = Method)) + geom_line() +  theme_bw(base_size = 15) + scale_color_manual(values=pallet) + xlim(0,1) + ylim(0,1) + labs(title="PR curves - GM12878", x ="Recall", y = "Precision") 
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend<-g_legend(prcurves)


p3 <- grid.arrange(roccurves + theme(legend.position="hidden"), prcurves + theme(legend.position="right"), nrow=1, widths = c(4,5.5))
pdf("gm12878_top35_curves.pdf",width = 8, height = 3)
plot(p3)
dev.off()



```


```{r}
library(precrec)
library(gridExtra)
top_x_table_k562 <- top_n(table_k562, 35, diff_k562)[,c(4,5,6)]
scores <- join_scores(top_x_table_k562[,1],top_x_table_k562[,2])
labels <- join_labels(top_x_table_k562[,3],top_x_table_k562[,3])
curve_data <- mmdata(scores, labels,modnames = c("panda", "egret"))
curves <- evalmod(curve_data)
#autoplot(curves)
#curves
#auc(curves)

rocP <- roc(top_x_table_k562$chipseq_k562, top_x_table_k562$panda, auc = TRUE, direction = "<")
rocE <- roc(top_x_table_k562$chipseq_k562, top_x_table_k562$egret_k562,auc = TRUE, direction = "<")
test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
pandaAUC <- test$estimate["AUC of roc1"]
egretAUC <- test$estimate["AUC of roc2"]
pval <- round(test$p.value,digits = 4)

pandaROC <- as.data.frame(curves$rocs[[1]]$x)
colnames(pandaROC) <- c("x")
pandaROC$y <- curves$rocs[[1]]$y
pandaROC$Method <- "PANDA"

egretROC <- as.data.frame(curves$rocs[[2]]$x)
colnames(egretROC) <- c("x")
egretROC$y <- curves$rocs[[2]]$y
egretROC$Method <- "EGRET"

roc_data <- rbind(pandaROC, egretROC)
roccurves <- ggplot(roc_data, aes(x=x, y=y, color=Method, group = Method)) + geom_line() +  theme_bw(base_size = 15) + scale_color_manual(values=pallet) + xlim(0,1) + ylim(0,1) + geom_abline(intercept = 0, slope = 1, linetype = 2) + labs(title="ROC curves - K562", x ="1 - Specificity", y = "Sensitivity") +annotate("text", x = 0.2, y = 0.8, label = paste0("Delong p: ",pval))


pandaPR <- as.data.frame(curves$prcs[[1]]$x)
colnames(pandaPR) <- c("x")
pandaPR$y <- curves$prcs[[1]]$y
pandaPR$Method <- "PANDA"

egretPR <- as.data.frame(curves$prcs[[2]]$x)
colnames(egretPR) <- c("x")
egretPR$y <- curves$prcs[[2]]$y
egretPR$Method <- "EGRET"

pr_data <- rbind(pandaPR, egretPR)
prcurves <- ggplot(pr_data, aes(x=x, y=y, color=Method, group = Method)) + geom_line() +  theme_bw(base_size = 15) + scale_color_manual(values=pallet) + xlim(0,1) + ylim(0,1) + labs(title="PR curves - K562", x ="Recall", y = "Precision") 
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend<-g_legend(prcurves)


p3 <- grid.arrange(roccurves + theme(legend.position="hidden"), prcurves + theme(legend.position="right"), nrow=1, widths = c(4,5.5))
pdf("k562_top35_curves.pdf",width = 8, height = 3)
plot(p3)
dev.off()
```



# Now show that difference scores are predictive of disruption with AUCs
```{r}
color <- "#D5006A"
top_x_table_gm12878 <- top_n(table_gm12878, 35, diff_gm12878)
top_x_table_gm12878$disrupt <- 1 - top_x_table_gm12878$chipseq
curve <- evalmod(scores = top_x_table_gm12878$diff_gm12878, labels = top_x_table_gm12878$disrupt, posclass = 1)
autoplot(curve)

roc <- as.data.frame(curve$rocs[[1]]$x)
colnames(roc) <- c("x")
roc$y <- curve$rocs[[1]]$y


pr <- as.data.frame(curve$prcs[[1]]$x)
colnames(pr) <- c("x")
pr$y <- curve$prcs[[1]]$y

roccurves <- ggplot(roc, aes(x=x, y=y, color = color)) + geom_line() +  theme_bw() + scale_color_manual(values=c(color)) + xlim(0,1) + ylim(0,1) + geom_abline(intercept = 0, slope = 1, linetype = 2) + labs(title="ROC-disruption (GM12878)", x ="1 - Specificity", y = "Sensitivity")

prcurves <- ggplot(pr, aes(x=x, y=y, color=color)) + geom_line() +  theme_bw() + scale_color_manual(values=c(color)) + xlim(0,1) + ylim(0,1) + labs(title="PR-disruption (GM12878)", x ="Recall", y = "Precision") 
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend<-g_legend(prcurves)


p3 <- grid.arrange(arrangeGrob(roccurves + theme(legend.position="none"), prcurves + theme(legend.position="none"), nrow=1, widths = c(4,4)), newpage = FALSE)
pdf("gm12878_top35_disruptionprediction_curves.pdf",width = 6, height = 3)
plot(p3)
dev.off()

```



```{r}
color <- "#D5006A"
top_x_table_k562 <- top_n(table_k562, 35, diff_k562)
top_x_table_k562$disrupt <- 1 - top_x_table_k562$chipseq
curve <- evalmod(scores = top_x_table_k562$diff_k562, labels = top_x_table_k562$disrupt, posclass = 1)
autoplot(curve)

roc <- as.data.frame(curve$rocs[[1]]$x)
colnames(roc) <- c("x")
roc$y <- curve$rocs[[1]]$y


pr <- as.data.frame(curve$prcs[[1]]$x)
colnames(pr) <- c("x")
pr$y <- curve$prcs[[1]]$y

roccurves <- ggplot(roc, aes(x=x, y=y, color = color)) + geom_line() +  theme_bw() + scale_color_manual(values=color) + xlim(0,1) + ylim(0,1) + geom_abline(intercept = 0, slope = 1, linetype = 2) + labs(title="ROC-disruption (K562)", x ="1 - Specificity", y = "Sensitivity")

prcurves <- ggplot(pr, aes(x=x, y=y, color=color)) + geom_line() +  theme_bw() + scale_color_manual(values=color) + xlim(0,1) + ylim(0,1) + labs(title="PR-disruption (K562)", x ="Recall", y = "Precision") 
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend<-g_legend(prcurves)


p3 <- grid.arrange(arrangeGrob(roccurves + theme(legend.position="none"), prcurves + theme(legend.position="none"), nrow=1, widths = c(4,4)), newpage = FALSE)
pdf("k562_top35_disruptionprediction_curves.pdf",width = 6, height = 3)
plot(p3)
dev.off()
```