---
title: "Final EGRET v1 validation: GM12878 with mouse"
author: "Deborah Weighill"
date: "June 19, 2020"
output:
  html_notebook:
    toc: true
    theme: cosmo
    df_print: paged
---


# Overview
This script validates the PANDA and EGRET networks against ChIP-seq data.


# Libraries, tags and inputs
Load libraries, set tag

```{r}
library(rtracklayer)
library(tidyr)
library(pROC)
library(reshape)
library(PRROC)
library(dplyr)
library(precrec)
library(ggplot2)
tag <- "finalEgret_v1_GM12878_06192020_withMouse"
chipseq <- "../../ca_egret/ca_egret_v3.1_LOO/validation/GM12878_remap2018_all_macs2_hg19_v1_2.bed"
```

# Make Panda/EGRET edge table

Load all of the PANDA/EGRET regnets

```{r echo=FALSE,results='hide'}
library(reshape2)

load("../outputs/finalEgret_v1_na12878_eQTLs_06182020_panda.RData")
net <- melt(regnet)
colnames(net) = c("tf","gene","score")
net$id <- paste0(net$tf,net$gene)

regnet_edge_table <- data.frame(net$id)
colnames(regnet_edge_table) <- c("id")
regnet_edge_table$tf <- as.vector(net$tf)
regnet_edge_table$gene <- as.vector(net$gene)
regnet_edge_table$panda <- as.vector(net$score)

#list of the file names of the rest of the networks
files <- c("finalEgret_v1_na12878_withMouse_06192020_egret.RData")

#nicknames of the networks
names <- c("egret_GM12878")



for (p in c(1:1)) {
  filename <- paste0("../outputs/",files[p])
  netname <- names[p]
  print(netname)
  load(filename)
  net <- melt(regnet)
  colnames(net) = c("tf","gene","score")
  net$id <- paste0(net$tf,net$gene)
  if(all.equal(as.vector(regnet_edge_table$id),as.vector(net$id))){
    print(p)
    regnet_edge_table$tempID <- as.vector(net$score)
    #print(colnames(regnet_edge_table))
    colnames(regnet_edge_table)[colnames(regnet_edge_table) == "tempID"] <- netname
  }
}
filename <- paste0("../outputs/edge_table_",tag,".RData")
save(regnet_edge_table, file = filename)
```

# Load ChIP-seq data
We load the chipseq data and make a ground truth regulatory network for validation.

```{r echo=FALSE,results='hide'}
# read in the gene annotation file with gene ranges.
genes <- read.table("../annotation/ensembl_genes_from_ucsc.txt", header = TRUE,sep = "\t")
genes$promotorLeft <- ifelse(genes$strand == "+", (genes$txStart + 1 - 750), (genes$txEnd-250))
genes$promotorRight <- ifelse(genes$strand == "+", (genes$txStart + 1 + 250), (genes$txEnd+750))
gr_genes_promotor <- GRanges(seqnames = genes$chrom, ranges = IRanges(start = genes$promotorLeft, end = genes$promotorRight), strand = NULL, mcols = genes[,13:15])

# chipseq chipseq data
gr_chipseq <- import(chipseq)
overlaps <- data.frame(findOverlaps(gr_chipseq, gr_genes_promotor))
overlaps$tf_name <- gr_chipseq$name[overlaps$queryHits]
overlaps$gene_name <- gr_genes_promotor$mcols.name2[overlaps$subjectHits]
validation_regnet <- separate(overlaps, tf_name, c("exp", "tf", "cell line"), "\\.")
chipseq_regnet <- distinct(validation_regnet[,c("tf","gene_name")])
chipseq_regnet$id <- paste0(chipseq_regnet$tf,chipseq_regnet$gene)
```

# Overlap networks with ChIP-seq data
We now filter down the panda/egret networks to only include those TFs which are in the ChIP-seq data (our validation set).
```{r echo=FALSE,results='hide'}
filtered_net_table <- regnet_edge_table[(regnet_edge_table$tf %in% chipseq_regnet$tf) ,]
filtered_net_table$chipseq <- ifelse(filtered_net_table$id %in% chipseq_regnet$id,1,0)
```

# Validate entire networks

Plot ROC/PR curves for each PANDA and EGRET nework, as well as table of AUCs and PRs
```{r}
library(pROC)
library(precrec)
library(ggplot2)
# genotype validation
for (i in c(4:5)){
  specs <- colnames(filtered_net_table)[i]
  curve <- evalmod(scores = filtered_net_table[,i], labels = as.factor(filtered_net_table$chipseq),modnames = c(specs))
  autoplot(curve)
  aucs <- auc(curve)
  print(aucs)
}
```



# Validate highest difference edges with chipseq
## GM12878
```{r echo=FALSE,results='hide'}
#calculate differences
#nicknames of the networks

filtered_net_table$diff_egret <- abs(filtered_net_table$egret_GM12878 - filtered_net_table$panda)

validation_table <- filtered_net_table
filename <- paste0("../outputs/validation_table_",tag,".RData")
save(validation_table, file = filename)
```

```{r,results='hide',fig.show=TRUE}
names <- c("egret_GM12878")
diff_cols <- c(7)
egret_net_nums <- c(5)
panda_net_nums <- c(4)
chipseq_col <- 6
get_roc_for_diff_thresh <- function(egretNum, thresh) {
  diffcolnum <- diff_cols[egretNum]
  egretcol <- egret_net_nums[egretNum]
  panda_col <- panda_net_nums[egretNum]
  
  top_diff_table <- validation_table[(validation_table[,c(diffcolnum)]>thresh),c(panda_col,egretcol,chipseq_col)]
  scores <- join_scores(top_diff_table[,1],top_diff_table[,2])
  labels <- join_labels(top_diff_table[,3],top_diff_table[,3])
  specs <- paste0(names[egretNum],", thresh: ",thresh)
  
  rocP <- roc(top_diff_table$chipseq, top_diff_table[,1], auc = TRUE, direction = "<")
  rocE <- roc(top_diff_table$chipseq, top_diff_table[,2],auc = TRUE, direction = "<")
  test <- roc.test(rocP, rocE, alternative = "less", method = "delong" )
  pandaAUC <- test$estimate["AUC of roc1"]
  egretAUC <- test$estimate["AUC of roc2"]
  pval <- test$p.value
  print(paste(egretNum,thresh,pval))
  
  curve_data <- mmdata(scores, labels,modnames = c("panda", specs), dsids = c(1, 2))
  curves <- evalmod(curve_data)
  nn <- attr(curves$rocs,"data_info")$nn[1]
  np <- attr(curves$rocs,"data_info")$np[1]
  result <- c(egretNum, names[egretNum], thresh, nn, np, pandaAUC, egretAUC, pval)
  return(result)
  autoplot(curves)
}

egret_priors <- c(1)
thresholds <- c(0.1, 0.15, 0.2, 0.25, 0.5, 0.75,1,2,3)

egret_auc_pvals<- data.frame()

for (p in egret_priors){
  for (t in thresholds){
    result <- get_roc_for_diff_thresh(p,t)
    df <- data.frame(t(result), stringsAsFactors = FALSE)
    egret_auc_pvals <- rbind(egret_auc_pvals,df)
  }
}

filename <- paste0("../outputs/AUC_pvals_",tag,".txt")
colnames(egret_auc_pvals) <- c("egretNum", "name", "thresh", "nn", "np", "pandaAUC", "egretAUC", "pval")
write.table(egret_auc_pvals, file = filename, sep = "\t", col.names = TRUE, quote=FALSE)
egret_auc_pvals
```

```{r, fig.show=TRUE, echo=FALSE,results='hide'}
names <- c("egret_GM12878")
diff_cols <- c(7)
egret_net_nums <- c(5)
panda_net_nums <- c(4)
chipseq_col <- 6

get_roc_for_diff_thresh <- function(egretNum, thresh) {
  diffcolnum <- diff_cols[egretNum]
  egretcol <- egret_net_nums[egretNum]
  panda_col <- panda_net_nums[egretNum]
  
  top_diff_table <- validation_table[(validation_table[,c(diffcolnum)]>thresh),c(panda_col,egretcol,chipseq_col)]
  scores <- join_scores(top_diff_table[,1],top_diff_table[,2])
  labels <- join_labels(top_diff_table[,3],top_diff_table[,3])
  specs <- paste0(names[egretNum],", thresh: ",thresh)
  curve_data <- mmdata(scores, labels,modnames = c("panda", specs), dsids = c(1, 2))
  curves <- evalmod(curve_data)
  aucs <- auc(curves)
  print(aucs)
  autoplot(curves)
}


egret_priors <- c(1)
#thresholds <- c(0.5)
thresholds <- c(0.05,0.1,0.25,0.5,0.75,1)


for (p in egret_priors){
  for (t in thresholds){
    get_roc_for_diff_thresh(p,t)
  }
}
```